<!DOCTYPE html><html lang="en" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1"><title>2021-05-14-tadaA-report | yang's blog</title><meta name="description" content="估参算法改进 将sapply全部改为Rcpp进行循环，整体的估参速度提升很多，如5组annotation联合估参时，改进前需要7h，而改进后只需要1.2h   多个参数估参时，optim method 由默认的Nelder-Mead改为BFGS，避免陷入局部最优   prior由原来固定的0.05，改为一个可变的参数，和annotations的effect size一起由optim估参决定  AS"><meta name="author" content=""><meta name="copyright" content=""><meta name="format-detection" content="telephone=no"><link rel="shortcut icon" href="/project/img/favicon.png"><link rel="canonical" href="https://yfu1116.github.io/project/2021-05-14-tadaA-report/"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//fonts.googleapis.com" crossorigin="crossorigin"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><meta property="og:type" content="article"><meta property="og:title" content="2021-05-14-tadaA-report"><meta property="og:url" content="https://yfu1116.github.io/project/2021-05-14-tadaA-report/"><meta property="og:site_name" content="yang's blog"><meta property="og:description" content="估参算法改进 将sapply全部改为Rcpp进行循环，整体的估参速度提升很多，如5组annotation联合估参时，改进前需要7h，而改进后只需要1.2h   多个参数估参时，optim method 由默认的Nelder-Mead改为BFGS，避免陷入局部最优   prior由原来固定的0.05，改为一个可变的参数，和annotations的effect size一起由optim估参决定  AS"><meta property="og:image" content="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg"><meta property="article:published_time" content="2021-05-14T11:04:08.506Z"><meta property="article:modified_time" content="2021-05-14T11:04:19.675Z"><meta name="twitter:card" content="summary"><script>var activateDarkMode = function () {
  document.documentElement.setAttribute('data-theme', 'dark')
  if (document.querySelector('meta[name="theme-color"]') !== null) {
    document.querySelector('meta[name="theme-color"]').setAttribute('content', '#000')
  }
}
var activateLightMode = function () {
  document.documentElement.setAttribute('data-theme', 'light')
  if (document.querySelector('meta[name="theme-color"]') !== null) {
    document.querySelector('meta[name="theme-color"]').setAttribute('content', '#fff')
  }
}

var getCookies = function (name) {
  const value = `; ${document.cookie}`
  const parts = value.split(`; ${name}=`)
  if (parts.length === 2) return parts.pop().split(';').shift()
}

var autoChangeMode = 'false'
var t = getCookies('theme')
if (autoChangeMode === '1') {
  var isDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches
  var isLightMode = window.matchMedia('(prefers-color-scheme: light)').matches
  var isNotSpecified = window.matchMedia('(prefers-color-scheme: no-preference)').matches
  var hasNoSupport = !isDarkMode && !isLightMode && !isNotSpecified

  if (t === undefined) {
    if (isLightMode) activateLightMode()
    else if (isDarkMode) activateDarkMode()
    else if (isNotSpecified || hasNoSupport) {
      console.log('You specified no preference for a color scheme or your browser does not support it. I Schedule dark mode during night time.')
      var now = new Date()
      var hour = now.getHours()
      var isNight = hour <= 6 || hour >= 18
      isNight ? activateDarkMode() : activateLightMode()
    }
    window.matchMedia('(prefers-color-scheme: dark)').addListener(function (e) {
      if (Cookies.get('theme') === undefined) {
        e.matches ? activateDarkMode() : activateLightMode()
      }
    })
  } else if (t === 'light') activateLightMode()
  else activateDarkMode()
} else if (autoChangeMode === '2') {
  now = new Date()
  hour = now.getHours()
  isNight = hour <= 6 || hour >= 18
  if (t === undefined) isNight ? activateDarkMode() : activateLightMode()
  else if (t === 'light') activateLightMode()
  else activateDarkMode()
} else {
  if (t === 'dark') activateDarkMode()
  else if (t === 'light') activateLightMode()
}</script><link rel="stylesheet" href="/project/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.css"><link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Titillium+Web&amp;display=swap"><script>var GLOBAL_CONFIG = { 
  root: '/project/',
  hexoversion: '4.0.0',
  algolia: undefined,
  localSearch: undefined,
  translate: undefined,
  noticeOutdate: undefined,
  copy: {
    success: 'Copy successfully',
    error: 'Copy error',
    noSupport: 'The browser does not support'
  },
  bookmark: {
    message_prev: 'Press',
    message_next: 'to bookmark this page'
  },
  runtime_unit: 'days',
  runtime: false,
  copyright: undefined,
  ClickShowText: undefined,
  medium_zoom: false,
  fancybox: true,
  Snackbar: undefined,
  justifiedGallery: {
    js: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/js/jquery.justifiedGallery.min.js',
    css: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/css/justifiedGallery.min.css'
  },
  baiduPush: false,
  highlightCopy: true,
  highlightLang: true,
  isPhotoFigcaption: false,
  islazyload: true,
  isanchor: false    
}</script><script id="config_change">var GLOBAL_CONFIG_SITE = { 
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isSidebar: true,
  postUpdate: '2021-05-14 19:04:19'
}</script><noscript><style>
#nav {
  opacity: 1
}
.justified-gallery img {
  opacity: 1
}
</style></noscript></head><body><div id="mobile-sidebar"><div id="menu_mask"></div><div id="mobile-sidebar-menus"><div class="mobile_author_icon"><img class="avatar-img" data-lazy-src="/project/img/avatar.png" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="mobile_post_data"><div class="mobile_data_item is-center"><div class="mobile_data_link"><a href="/project/archives/"><div class="headline">Articles</div><div class="length_num">71</div></a></div></div></div><hr/></div></div><div id="body-wrap"><div id="sidebar"><i class="fas fa-arrow-right on" id="toggle-sidebar"></i><div class="sidebar-toc"><div class="sidebar-toc__title">Catalog</div><div class="sidebar-toc__progress"><span class="progress-notice">You've read</span><span class="progress-num">0</span><span class="progress-percentage">%</span><div class="sidebar-toc__progress-bar">     </div></div><div class="sidebar-toc__content"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#估参算法改进"><span class="toc-number">1.</span> <span class="toc-text">估参算法改进</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#ASD"><span class="toc-number">2.</span> <span class="toc-text">ASD</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#SNV-model"><span class="toc-number">2.1.</span> <span class="toc-text">SNV model</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#effect-size-of-functional-annotations"><span class="toc-number">2.1.1.</span> <span class="toc-text">effect size of functional annotations</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#functional-analysis"><span class="toc-number">2.1.2.</span> <span class="toc-text">functional analysis</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#indel-model"><span class="toc-number">2.2.</span> <span class="toc-text">indel model</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#SNV-model-indel-model"><span class="toc-number">2.3.</span> <span class="toc-text">SNV model + indel model</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#potentially-damaging-DNMs"><span class="toc-number">2.4.</span> <span class="toc-text">potentially damaging DNMs</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#CHD"><span class="toc-number">3.</span> <span class="toc-text">CHD</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#effect-size-of-functional-annotations-1"><span class="toc-number">3.1.</span> <span class="toc-text">effect size of functional annotations</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#risk-genes"><span class="toc-number">3.2.</span> <span class="toc-text">risk genes</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Functional-analysis-for-these-novel-genes"><span class="toc-number">3.3.</span> <span class="toc-text">Functional analysis for these novel genes</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#potentially-damaging-DNMs-1"><span class="toc-number">3.4.</span> <span class="toc-text">potentially damaging DNMs</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#next-plan"><span class="toc-number">4.</span> <span class="toc-text">next plan</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#question"><span class="toc-number">5.</span> <span class="toc-text">question</span></a></li></ol></div></div></div><header class="post-bg" id="page-header" style="background-image: url(https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg)"><nav id="nav"><span class="pull-left" id="blog_name"><a class="blog_title" id="site-name" href="/project/">yang's blog</a></span><span class="pull-right menus"><span class="toggle-menu close"><a class="site-page"><i class="fas fa-bars fa-fw"></i></a></span></span></nav><div id="post-info"><div id="post-title"><div class="posttitle">2021-05-14-tadaA-report</div></div><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">Created</span><time class="post-meta-date-created" datetime="2021-05-14T11:04:08.506Z" title="Created 2021-05-14 19:04:08">2021-05-14</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">Updated</span><time class="post-meta-date-updated" datetime="2021-05-14T11:04:19.675Z" title="Updated 2021-05-14 19:04:19">2021-05-14</time></span></div><div class="meta-secondline"> <span class="post-meta-separator">|</span><span class="post-meta-pv-cv"><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">Post View:</span><span id="busuanzi_value_page_pv"></span></span></div></div></div></header><main class="layout_post" id="content-inner"><article id="post"><div class="post-content" id="article-container"><h1 id="估参算法改进"><a href="#估参算法改进" class="headerlink" title="估参算法改进"></a>估参算法改进</h1><ul>
<li>将sapply全部改为Rcpp进行循环，整体的估参速度提升很多，如5组annotation联合估参时，改进前需要7h，而改进后只需要1.2h</li>
</ul>
<ul>
<li>多个参数估参时，optim method 由默认的<code>Nelder-Mead</code>改为<code>BFGS</code>，避免陷入局部最优</li>
</ul>
<ul>
<li>prior由原来固定的0.05，改为一个可变的参数，和annotations的effect size一起由optim估参决定</li>
</ul>
<h1 id="ASD"><a href="#ASD" class="headerlink" title="ASD"></a>ASD</h1><h2 id="SNV-model"><a href="#SNV-model" class="headerlink" title="SNV model"></a>SNV model</h2><h3 id="effect-size-of-functional-annotations"><a href="#effect-size-of-functional-annotations" class="headerlink" title="effect size of functional annotations"></a>effect size of functional annotations</h3><p>number of the affected individuals:  6430</p>
<p>separately estimated rr of all annotations:<a href="https://yfu1116.github.io/project/2021-05-11-rr-all-annota-ASD-CHD/">https://yfu1116.github.io/project/2021-05-11-rr-all-annota-ASD-CHD/</a></p>
<p><strong>feature selection:</strong>   固定prior为0.05，用optim对每个annotation单独估参，选择置信区间在0以上的进行joint estimation</p>
<p><code>joint_logRR</code>: </p>
<ul>
<li><p>jointly estimate logRR and prior using <code>optim(method = &quot;BFGS&quot;)</code>, where logRR is a vector, and prior is a scalar</p>
</li>
<li><p>result: <strong>prior = 0.118159391983715</strong></p>
</li>
</ul>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">readRDS(&quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/2021-05-11_selected16_joint_separate_rr.rds&quot;)</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.frame: 16 × 5</caption>
<thead>
    <tr><th scope="col">joint_logRR</th><th scope="col">annota</th><th scope="col">separate_logRR</th><th scope="col">upper_bound</th><th scope="col">lower_bound</th></tr>
    <tr><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;chr&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>0.7753066</td><td>hnRNPL binding regions                                                                                    </td><td>1.599488</td><td>3.083753</td><td>0.1152234</td></tr>
    <tr><td>0.2631773</td><td><span style="white-space:pre-wrap">coding constraint &gt;= 90                                                                                   </span></td><td>1.897444</td><td>2.103377</td><td>1.6915119</td></tr>
    <tr><td>0.8483160</td><td>dbSNP.RBP-Var: likely to affcted RNA-bind, RNA 2nd structure                                              </td><td>1.844203</td><td>2.278540</td><td>1.4098657</td></tr>
    <tr><td>0.6809386</td><td>RNA modifications, including m6a, m1A, m5C, and etc.(filter_low_RMVar_hb)                                 </td><td>1.312018</td><td>1.645988</td><td>0.9780486</td></tr>
    <tr><td>0.5315570</td><td><span style="white-space:pre-wrap">deepsea: mix ago_adult_brain.BA4 &amp; ago_adult_brain.Cingulate.gyrus                                        </span></td><td>2.483918</td><td>2.903837</td><td>2.0639991</td></tr>
    <tr><td>0.5140172</td><td>deepsea: mix elavl_Adult_brain.all_human_samples &amp; elavl_Adult_brain.BA9_Alzheimer &amp; elavl_Adult_brain.BA9</td><td>2.592861</td><td>3.006368</td><td>2.1793531</td></tr>
    <tr><td>0.2471636</td><td><span style="white-space:pre-wrap">1&lt;= mpc score &lt; 2                                                                                         </span></td><td>1.621601</td><td>1.835855</td><td>1.4073458</td></tr>
    <tr><td>0.7175324</td><td><span style="white-space:pre-wrap">mpc score &gt;= 2                                                                                            </span></td><td>2.321405</td><td>2.547951</td><td>2.0948598</td></tr>
    <tr><td>0.7615884</td><td>interaction-disrupting: mutations annotated as interface residues and probably damaging by PolyPhen-2     </td><td>2.600493</td><td>2.828070</td><td>2.3729150</td></tr>
    <tr><td>0.7574717</td><td>pathogenic missense: primateAI_MVP_mix                                                                    </td><td>1.757185</td><td>1.892405</td><td>1.6219658</td></tr>
    <tr><td>0.2998243</td><td>spidex_lower10pct                                                                                         </td><td>1.773391</td><td>1.967028</td><td>1.5797548</td></tr>
    <tr><td>0.3567299</td><td><span style="white-space:pre-wrap">spliceAI accept gain &gt;= 0.05                                                                              </span></td><td>1.997280</td><td>3.664953</td><td>0.3296074</td></tr>
    <tr><td>0.9058033</td><td><span style="white-space:pre-wrap">spliceAI donor gain &gt;= 0.05                                                                               </span></td><td>2.800307</td><td>3.476940</td><td>2.1236739</td></tr>
    <tr><td>1.1840759</td><td>ptv in [0,0.5)                                                                                            </td><td>2.081598</td><td>2.557930</td><td>1.6052671</td></tr>
    <tr><td>1.0524383</td><td>ptv in [0.5,0.995)                                                                                        </td><td>2.209710</td><td>2.964529</td><td>1.4548916</td></tr>
    <tr><td>2.5145196</td><td>ptv in [0.995,1)                                                                                          </td><td>3.493791</td><td>3.759486</td><td>3.2280951</td></tr>
</tbody>
</table>



<p><strong>heatmap of similarity bewteen two annotations</strong></p>
<p><code>x-axis</code>: count of overlap / count of annota1 mutations in our windows <code>(r1)</code>,   <code>y-axis</code>: count of overlap / count of annota2 mutations in our windows <code>(r2)</code></p>
<p>red denotes the ratio over 40%, and blue denotes the ratio less than 40%</p>
<p><img src= "/project/img/loading.gif" data-lazy-src="https://z3.ax1x.com/2021/05/13/gBAwa4.png" alt="gBAwa4.png"></p>
<p><code>r1</code>: count of overlap / count of annota1 mutations in our windows,   <code>r2</code>: count of overlap / count of annota2 mutations in our windows </p>
<p>以40%为分界线，两两相似度较高的annotation如下：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">library(data.table)</span><br><span class="line">readRDS(&quot;/storage11_7T/fuy/TADA-A/annotation/overlp/results/asd_chd_red.txt&quot;)</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.table: 9 × 4</caption>
<thead>
    <tr><th scope="col">annota1</th><th scope="col">annota2</th><th scope="col">r1</th><th scope="col">r2</th></tr>
    <tr><th scope="col">&lt;chr&gt;</th><th scope="col">&lt;chr&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>pathogenic missense: primateAI </td><td>mpc12                                                                         </td><td>0.4027</td><td>0.3804</td></tr>
    <tr><td>pathogenic missense: primateAI </td><td>MVP                                                                           </td><td>0.4337</td><td>0.2908</td></tr>
    <tr><td>pathogenic missense: primateAI </td><td>interaction-disrupting: interface residues and probably damaging by PolyPhen-2</td><td>0.1035</td><td>0.4279</td></tr>
    <tr><td>pathogenic missense: primateAI </td><td>mpc2                                                                          </td><td>0.2076</td><td>0.6311</td></tr>
    <tr><td>spidex                         </td><td>ptv.0-05                                                                      </td><td>0.1838</td><td>0.6232</td></tr>
    <tr><td>spidex                         </td><td>ptv.995                                                                       </td><td>0.0639</td><td>0.8053</td></tr>
    <tr><td>spidex                         </td><td>ptv.05-995                                                                    </td><td>0.0497</td><td>0.6226</td></tr>
    <tr><td>pathogenic missense: MVP       </td><td>interaction-disrupting: interface residues and probably damaging by PolyPhen-2</td><td>0.0904</td><td>0.5569</td></tr>
    <tr><td>pathogenic missense: MVP       </td><td>mpc2                                                                          </td><td>0.1080</td><td>0.4895</td></tr>
</tbody>
</table>



<h3 id="functional-analysis"><a href="#functional-analysis" class="headerlink" title="functional analysis"></a>functional analysis</h3><p>SNV model: 找出80个风险基因，其中8个基因中没有覆盖任何DNM，因此将其去除</p>
<h2 id="indel-model"><a href="#indel-model" class="headerlink" title="indel model"></a>indel model</h2><p>将SNV模型得到的基因致病的posterior作为indel模型的prior，用EM算法得到frameshift的effect size，details: <a href="https://yfu1116.github.io/project/2021-05-13-indel-model-ASD/">https://yfu1116.github.io/project/2021-05-13-indel-model-ASD/</a></p>
<p><strong>Results:</strong></p>
<p>RR(relative risk) = 3.64713076316748</p>
<p>logRR = 1.29394076617897</p>
<p>跟SNV的结果相比，indel model新增了17个新基因，这些基因的功能富集在neuron projection morphogenesis, Neurodevelopmental Disorders, Autistic behavior等(<strong>details:</strong> <a href="https://yfu1116.github.io/project/2021-05-13_md_indel-snv_ASD_riskG_GO_DisGeNET_17/">https://yfu1116.github.io/project/2021-05-13_md_indel-snv_ASD_riskG_GO_DisGeNET_17/</a>))  ，一定程度上证明了indel model的可行性：</p>
<h2 id="SNV-model-indel-model"><a href="#SNV-model-indel-model" class="headerlink" title="SNV model + indel model"></a>SNV model + indel model</h2><p>相对于cell那篇文章里找出的102个风险基因，我们的SNV+indel model 找出了62个与其一致的基因，和27个novel ASD risk genes，这些novel risk genes的功能富集如下：</p>
<p><strong>novel risk gene list:</strong></p>
<p><strong>GO</strong></p>
<p><img src= "/project/img/loading.gif" data-lazy-src="https://z3.ax1x.com/2021/05/10/gNAU8U.png" alt="gNAU8U.png"></p>
<p><strong>DisGeNET (gene-disease association)</strong></p>
<p><img src= "/project/img/loading.gif" data-lazy-src="https://z3.ax1x.com/2021/05/10/gNEQJK.png" alt="gNEQJK.png"></p>
<h2 id="potentially-damaging-DNMs"><a href="#potentially-damaging-DNMs" class="headerlink" title="potentially damaging DNMs"></a>potentially damaging DNMs</h2><p>DNMs with functional annotations in novel risk genes :<a href="https://github.com/yfu1116/project/blob/master/files/ASD_annota_merge_info.xlsx?raw=true" target="_blank" rel="noopener">https://github.com/yfu1116/project/blob/master/files/ASD_annota_merge_info.xlsx?raw=true</a></p>
<p>打算重点看synonymous / constrained transcriptional or post-transcriptional regulatory variants</p>
<h1 id="CHD"><a href="#CHD" class="headerlink" title="CHD"></a>CHD</h1><h2 id="effect-size-of-functional-annotations-1"><a href="#effect-size-of-functional-annotations-1" class="headerlink" title="effect size of functional annotations"></a>effect size of functional annotations</h2><p>number of the affected individuals:  2871</p>
<p>separately estimated rr of all annotations: <a href="https://yfu1116.github.io/project/2021-05-11-rr-all-annota-ASD-CHD/">https://yfu1116.github.io/project/2021-05-11-rr-all-annota-ASD-CHD/</a></p>
<p><strong>feature selection:</strong>   固定prior为0.05，用optim对每个annotation单独估参，选择置信区间在0以上的进行joint estimation</p>
<p><code>joint_logRR</code>: </p>
<ul>
<li><p>estimate logRR and prior using <code>optim(method = &quot;BFGS&quot;)</code>, where logRR is a vector, and prior is a scalar</p>
</li>
<li><p>result: <strong>prior = 0.0293059350146621</strong></p>
</li>
</ul>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">readRDS(&quot;/storage11_7T/fuy/TADA-A/CHD/rds/sig_0512_all_annota_rr.rds&quot;)</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.frame: 16 × 5</caption>
<thead>
    <tr><th></th><th scope="col">joint_logRR</th><th scope="col">annota</th><th scope="col">separately_logRR</th><th scope="col">upper_bound</th><th scope="col">lower_bound</th></tr>
    <tr><th></th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;chr&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><th scope="row">3</th><td> 0.11161697</td><td><span style="white-space:pre-wrap">coding constraint &gt;= 90                                                                                   </span></td><td>1.687606</td><td>2.192012</td><td> 1.18320074</td></tr>
    <tr><th scope="row">5</th><td> 0.70350768</td><td>RADAR: top5% variants in the post-transcriptional regulome of RBP                                         </td><td>1.814882</td><td>2.339178</td><td> 1.29058489</td></tr>
    <tr><th scope="row">6</th><td> 1.48950949</td><td>dbSNP.RBP-Var: likely to affcted RNA-bind, RNA 2nd structure                                              </td><td>2.194627</td><td>2.734925</td><td> 1.65432873</td></tr>
    <tr><th scope="row">10</th><td> 1.02046920</td><td><span style="white-space:pre-wrap">deepsea: mix ago_adult_brain.BA4 &amp; ago_adult_brain.Cingulate.gyrus                                        </span></td><td>2.430548</td><td>3.074458</td><td> 1.78663714</td></tr>
    <tr><th scope="row">11</th><td>-0.18538841</td><td>deepsea: mix elavl_Adult_brain.all_human_samples &amp; elavl_Adult_brain.BA9_Alzheimer &amp; elavl_Adult_brain.BA9</td><td>1.738795</td><td>3.155253</td><td> 0.32233776</td></tr>
    <tr><th scope="row">12</th><td> 0.32493815</td><td><span style="white-space:pre-wrap">1&lt;= mpc score &lt; 2                                                                                         </span></td><td>1.813293</td><td>2.139593</td><td> 1.48699171</td></tr>
    <tr><th scope="row">13</th><td> 0.61086386</td><td><span style="white-space:pre-wrap">mpc score &gt;= 2                                                                                            </span></td><td>2.058641</td><td>2.554155</td><td> 1.56312683</td></tr>
    <tr><th scope="row">14</th><td> 0.03448242</td><td>interaction-disrupting: mutations annotated as interface residues and probably damaging by PolyPhen-2     </td><td>2.307627</td><td>2.803729</td><td> 1.81152465</td></tr>
    <tr><th scope="row">15</th><td> 1.75734201</td><td>pathogenic missense: primateAI_MVP_mix                                                                    </td><td>2.035133</td><td>2.225165</td><td> 1.84509985</td></tr>
    <tr><th scope="row">16</th><td> 0.63166845</td><td>spidex_lower10pct                                                                                         </td><td>2.030793</td><td>2.296126</td><td> 1.76545964</td></tr>
    <tr><th scope="row">17</th><td> 0.96439229</td><td><span style="white-space:pre-wrap">spliceAI accept gain &gt;= 0.05                                                                              </span></td><td>2.944004</td><td>4.195226</td><td> 1.69278194</td></tr>
    <tr><th scope="row">18</th><td> 0.19183110</td><td><span style="white-space:pre-wrap">spliceAI donor gain &gt;= 0.05                                                                               </span></td><td>2.472156</td><td>3.889113</td><td> 1.05519861</td></tr>
    <tr><th scope="row">20</th><td> 0.63851827</td><td><span style="white-space:pre-wrap">spliceAI donor loss &gt;= 0.05                                                                               </span></td><td>2.156331</td><td>4.376120</td><td>-0.06345714</td></tr>
    <tr><th scope="row">21</th><td> 1.12710657</td><td>ptv in [0,0.5)                                                                                            </td><td>2.283768</td><td>2.944003</td><td> 1.62353425</td></tr>
    <tr><th scope="row">22</th><td> 2.61386398</td><td>ptv in [0.5,995)                                                                                          </td><td>3.039657</td><td>3.769662</td><td> 2.30965246</td></tr>
    <tr><th scope="row">23</th><td> 3.02368867</td><td>ptv in [0.995,1)                                                                                          </td><td>3.701728</td><td>4.160876</td><td> 3.24258001</td></tr>
</tbody>
</table>



<h2 id="risk-genes"><a href="#risk-genes" class="headerlink" title="risk genes"></a>risk genes</h2><p><strong>known</strong> CHD risk genelist is from the paper: <a href="https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-019-0709-8" target="_blank" rel="noopener">https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-019-0709-8</a></p>
<p><strong>SNV model:</strong> 20 CHD risk genes are identified, of which 12 are novel genes</p>
<p><strong>INDEL model:</strong> effect size of frameshift = 6.29013716769851, <strong>2 new risk genes</strong> are identified(<strong>“CIT”</strong>,  <strong>“KDM5B”</strong>)</p>
<p><strong>SNV + indel model:</strong> 21 CHD risk genes are identified, of which 13 are novel genes. Details:</p>
<h2 id="Functional-analysis-for-these-novel-genes"><a href="#Functional-analysis-for-these-novel-genes" class="headerlink" title="Functional analysis for these novel genes"></a>Functional analysis for these novel genes</h2><p><strong>GO</strong> </p>
<p><img src= "/project/img/loading.gif" data-lazy-src="https://z3.ax1x.com/2021/05/13/gBUQPA.png" alt="gBUQPA.png"></p>
<p><strong>DisGeNET (gene-disease association)</strong></p>
<p><img src= "/project/img/loading.gif" data-lazy-src="https://z3.ax1x.com/2021/05/13/gBU0Gn.png" alt="gBU0Gn.png"></p>
<h2 id="potentially-damaging-DNMs-1"><a href="#potentially-damaging-DNMs-1" class="headerlink" title="potentially damaging DNMs"></a>potentially damaging DNMs</h2><p>DNMs with functional annotations in novel risk genes :<a href="https://github.com/yfu1116/project/blob/master/files/CHD_annota_merge_info.xlsx?raw=true" target="_blank" rel="noopener">https://github.com/yfu1116/project/blob/master/files/CHD_annota_merge_info.xlsx?raw=true</a></p>
<h1 id="next-plan"><a href="#next-plan" class="headerlink" title="next plan"></a>next plan</h1><p>find potential transcriptional or post-transcriptional variants in the etiology of ASD</p>
<h1 id="question"><a href="#question" class="headerlink" title="question"></a>question</h1><p>1) frameshift effect size偏小的原因</p>
<p>2) 当一个变异有多重注释，如(protein-truncating (score: 0-0.05) and RNA modifying)，如何确认哪种作用机制的影响大？</p>
</div><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">Author: </span><span class="post-copyright-info"><a href="mailto:undefined" target="_blank" rel="noopener"></a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">Link: </span><span class="post-copyright-info"><a href="https://yfu1116.github.io/project/2021-05-14-tadaA-report/">https://yfu1116.github.io/project/2021-05-14-tadaA-report/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">Copyright Notice: </span><span class="post-copyright-info">All articles in this blog are licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank" rel="noopener">CC BY-NC-SA 4.0</a> unless stating additionally.</span></div></div><div class="tag_share"><div class="post-meta__tag-list"></div><div class="post_share"><div class="social-share" data-image="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" data-sites="github,email,facebook,twitter,weixin,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js/dist/css/share.min.css"><script src="https://cdn.jsdelivr.net/npm/social-share.js/dist/js/social-share.min.js" defer></script></div></div><nav class="pagination-post" id="pagination"><div class="prev-post pull-left"><a href="/project/2021-05-20-fix-pi-VS-estim-pi-effect-size-num-enrich/"><img class="prev-cover" data-lazy-src="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" onerror="onerror=null;src='/project/img/404.jpg'"><div class="pagination-info"><div class="label">Previous Post</div><div class="prev_info">2021-05-20-fix-pi-VS-estim-pi-effect-size-num-enrich</div></div></a></div><div class="next-post pull-right"><a href="/project/2021-05-13-indel+snv-novel-CHD-GO/"><img class="next-cover" data-lazy-src="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" onerror="onerror=null;src='/project/img/404.jpg'"><div class="pagination-info"><div class="label">Next Post</div><div class="next_info">2021-05-13-indel+snv-novel-CHD-GO</div></div></a></div></nav></article></main><footer id="footer" data-type="color"><div id="footer-wrap"><div class="copyright">&copy;2020 - 2021 By </div><div class="framework-info"><span>Framework </span><a href="https://hexo.io" target="_blank" rel="noopener">Hexo</a><span class="footer-separator">|</span><span>Theme </span><a href="https://github.com/jerryc127/hexo-theme-butterfly" target="_blank" rel="noopener">Butterfly</a></div></div></footer></div><section id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="Read Mode"><i class="fas fa-book-open"></i></button><button id="font_plus" type="button" title="Increase Font Size"><i class="fas fa-plus"></i></button><button id="font_minus" type="button" title="Decrease Font Size"><i class="fas fa-minus"></i></button><button id="darkmode" type="button" title="Switch Between Light And Dark Mode"><i class="fas fa-adjust"></i></button></div><div id="rightside-config-show"><button id="rightside_config" type="button" title="Setting"><i class="fas fa-cog"></i></button><button class="close" id="mobile-toc-button" type="button" title="Table Of Contents"><i class="fas fa-list-ul"></i></button><button id="go-up" type="button" title="Back To Top"><i class="fas fa-arrow-up"></i></button></div></section><div><script src="https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js"></script><script src="/project/js/utils.js"></script><script src="/project/js/main.js"></script><script src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.js"></script><script src="https://cdn.jsdelivr.net/npm/instant.page/instantpage.min.js" type="module" defer></script><script src="https://cdn.jsdelivr.net/npm/vanilla-lazyload/dist/lazyload.iife.min.js" async></script><div class="js-pjax"><script>if (!window.MathJax) {
  window.MathJax = {
    loader: {
      source: {
        '[tex]/amsCd': '[tex]/amscd'
      }
    },
    tex: {
      inlineMath: [ ['$','$'], ["\\(","\\)"]],
      tags: 'ams'
    },
    options: {
      renderActions: {
        findScript: [10, doc => {
          for (const node of document.querySelectorAll('script[type^="math/tex"]')) {
            const display = !!node.type.match(/; *mode=display/)
            const math = new doc.options.MathItem(node.textContent, doc.inputJax[0], display)
            const text = document.createTextNode('')
            node.parentNode.replaceChild(text, node)
            math.start = {node: text, delim: '', n: 0}
            math.end = {node: text, delim: '', n: 0}
            doc.math.push(math)
          }
        }, '']
      }
    }
  }
  
  var script = document.createElement('script')
  script.src = 'https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js'
  script.id = 'MathJax-script'
  script.async = true
  document.head.appendChild(script)
} else {
  MathJax.startup.document.state(0)
  MathJax.texReset()
  MathJax.typeset()
}</script><script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div></div></body></html>