<!DOCTYPE html><html lang="en" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1"><title>2021-03-18-tadaA-report | yang's blog</title><meta name="description" content="Notes: Data structure array  EM rectify Optim  Annotation 占总数的比值，是否过于conserved, overlap heatmap Estimation of indel rateSNV_rate: 2N * summation over all possible mutation rates of bases in window i Y"><meta name="author" content=""><meta name="copyright" content=""><meta name="format-detection" content="telephone=no"><link rel="shortcut icon" href="/project/img/favicon.png"><link rel="canonical" href="https://yfu1116.github.io/project/2021-03-18-tadaA-report/"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//fonts.googleapis.com" crossorigin="crossorigin"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><meta property="og:type" content="article"><meta property="og:title" content="2021-03-18-tadaA-report"><meta property="og:url" content="https://yfu1116.github.io/project/2021-03-18-tadaA-report/"><meta property="og:site_name" content="yang's blog"><meta property="og:description" content="Notes: Data structure array  EM rectify Optim  Annotation 占总数的比值，是否过于conserved, overlap heatmap Estimation of indel rateSNV_rate: 2N * summation over all possible mutation rates of bases in window i Y"><meta property="og:image" content="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg"><meta property="article:published_time" content="2021-03-19T08:03:34.292Z"><meta property="article:modified_time" content="2021-03-19T08:03:55.826Z"><meta name="twitter:card" content="summary"><script>var activateDarkMode = function () {
  document.documentElement.setAttribute('data-theme', 'dark')
  if (document.querySelector('meta[name="theme-color"]') !== null) {
    document.querySelector('meta[name="theme-color"]').setAttribute('content', '#000')
  }
}
var activateLightMode = function () {
  document.documentElement.setAttribute('data-theme', 'light')
  if (document.querySelector('meta[name="theme-color"]') !== null) {
    document.querySelector('meta[name="theme-color"]').setAttribute('content', '#fff')
  }
}

var getCookies = function (name) {
  const value = `; ${document.cookie}`
  const parts = value.split(`; ${name}=`)
  if (parts.length === 2) return parts.pop().split(';').shift()
}

var autoChangeMode = 'false'
var t = getCookies('theme')
if (autoChangeMode === '1') {
  var isDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches
  var isLightMode = window.matchMedia('(prefers-color-scheme: light)').matches
  var isNotSpecified = window.matchMedia('(prefers-color-scheme: no-preference)').matches
  var hasNoSupport = !isDarkMode && !isLightMode && !isNotSpecified

  if (t === undefined) {
    if (isLightMode) activateLightMode()
    else if (isDarkMode) activateDarkMode()
    else if (isNotSpecified || hasNoSupport) {
      console.log('You specified no preference for a color scheme or your browser does not support it. I Schedule dark mode during night time.')
      var now = new Date()
      var hour = now.getHours()
      var isNight = hour <= 6 || hour >= 18
      isNight ? activateDarkMode() : activateLightMode()
    }
    window.matchMedia('(prefers-color-scheme: dark)').addListener(function (e) {
      if (Cookies.get('theme') === undefined) {
        e.matches ? activateDarkMode() : activateLightMode()
      }
    })
  } else if (t === 'light') activateLightMode()
  else activateDarkMode()
} else if (autoChangeMode === '2') {
  now = new Date()
  hour = now.getHours()
  isNight = hour <= 6 || hour >= 18
  if (t === undefined) isNight ? activateDarkMode() : activateLightMode()
  else if (t === 'light') activateLightMode()
  else activateDarkMode()
} else {
  if (t === 'dark') activateDarkMode()
  else if (t === 'light') activateLightMode()
}</script><link rel="stylesheet" href="/project/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.css"><link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Titillium+Web&amp;display=swap"><script>var GLOBAL_CONFIG = { 
  root: '/project/',
  hexoversion: '4.0.0',
  algolia: undefined,
  localSearch: undefined,
  translate: undefined,
  noticeOutdate: undefined,
  copy: {
    success: 'Copy successfully',
    error: 'Copy error',
    noSupport: 'The browser does not support'
  },
  bookmark: {
    message_prev: 'Press',
    message_next: 'to bookmark this page'
  },
  runtime_unit: 'days',
  runtime: false,
  copyright: undefined,
  ClickShowText: undefined,
  medium_zoom: false,
  fancybox: true,
  Snackbar: undefined,
  justifiedGallery: {
    js: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/js/jquery.justifiedGallery.min.js',
    css: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/css/justifiedGallery.min.css'
  },
  baiduPush: false,
  highlightCopy: true,
  highlightLang: true,
  isPhotoFigcaption: false,
  islazyload: true,
  isanchor: false    
}</script><script id="config_change">var GLOBAL_CONFIG_SITE = { 
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isSidebar: true,
  postUpdate: '2021-03-19 16:03:55'
}</script><noscript><style>
#nav {
  opacity: 1
}
.justified-gallery img {
  opacity: 1
}
</style></noscript></head><body><div id="mobile-sidebar"><div id="menu_mask"></div><div id="mobile-sidebar-menus"><div class="mobile_author_icon"><img class="avatar-img" data-lazy-src="/project/img/avatar.png" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="mobile_post_data"><div class="mobile_data_item is-center"><div class="mobile_data_link"><a href="/project/archives/"><div class="headline">Articles</div><div class="length_num">36</div></a></div></div></div><hr/></div></div><div id="body-wrap"><div id="sidebar"><i class="fas fa-arrow-right on" id="toggle-sidebar"></i><div class="sidebar-toc"><div class="sidebar-toc__title">Catalog</div><div class="sidebar-toc__progress"><span class="progress-notice">You've read</span><span class="progress-num">0</span><span class="progress-percentage">%</span><div class="sidebar-toc__progress-bar">     </div></div><div class="sidebar-toc__content"><ol class="toc"><li class="toc-item toc-level-3"><a class="toc-link" href="#Estimation-of-indel-rate"><span class="toc-number">1.</span> <span class="toc-text">Estimation of indel rate</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#frameshift-rate"><span class="toc-number">1.1.</span> <span class="toc-text">frameshift rate</span></a></li></ol></li><li class="toc-item toc-level-3"><a class="toc-link" href="#Estimation-of-frameshift-effect-size"><span class="toc-number">2.</span> <span class="toc-text">Estimation of frameshift effect size</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#SNV-model"><span class="toc-number">3.</span> <span class="toc-text">SNV model</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#New-annotations"><span class="toc-number">4.</span> <span class="toc-text">New annotations</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#CHD-data"><span class="toc-number">5.</span> <span class="toc-text">CHD data</span></a></li></ol></div></div></div><header class="post-bg" id="page-header" style="background-image: url(https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg)"><nav id="nav"><span class="pull-left" id="blog_name"><a class="blog_title" id="site-name" href="/project/">yang's blog</a></span><span class="pull-right menus"><span class="toggle-menu close"><a class="site-page"><i class="fas fa-bars fa-fw"></i></a></span></span></nav><div id="post-info"><div id="post-title"><div class="posttitle">2021-03-18-tadaA-report</div></div><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">Created</span><time class="post-meta-date-created" datetime="2021-03-19T08:03:34.292Z" title="Created 2021-03-19 16:03:34">2021-03-19</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">Updated</span><time class="post-meta-date-updated" datetime="2021-03-19T08:03:55.826Z" title="Updated 2021-03-19 16:03:55">2021-03-19</time></span></div><div class="meta-secondline"> <span class="post-meta-separator">|</span><span class="post-meta-pv-cv"><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">Post View:</span><span id="busuanzi_value_page_pv"></span></span></div></div></div></header><main class="layout_post" id="content-inner"><article id="post"><div class="post-content" id="article-container"><p><strong>Notes:</strong></p>
<p>Data structure array </p>
<p>EM rectify</p>
<p>Optim </p>
<p>Annotation 占总数的比值，是否过于conserved, overlap heatmap</p>
<h3 id="Estimation-of-indel-rate"><a href="#Estimation-of-indel-rate" class="headerlink" title="Estimation of indel rate"></a>Estimation of indel rate</h3><p>SNV_rate: 2N * summation over all possible mutation rates of bases in window i</p>
<p>Y ~ possion($\mu$)</p>
<p>Y ~ possion($e^{\beta_0 + \beta_1 * SNV_rate + \beta_2 * GC_content}$)</p>
<p>log(obs indel count) ~ $\beta_0$ + $\beta_1$ * SNV_rate + $\beta_2$ * GC_content</p>
<p>$\mu$ indel rates over 2N individuals.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">library(data.table)</span><br><span class="line">options(digits=22)</span><br><span class="line"></span><br><span class="line">df = read.table(&quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/indel_rate/gene_level_GC_SNV_rate_indel_ct_len.txt&quot;,header=T)</span><br><span class="line">nrow(df)</span><br><span class="line">head(df,3)</span><br></pre></td></tr></table></figure>


<p>17478</p>
<table>
<caption>A data.frame: 3 × 5</caption>
<thead>
    <tr><th></th><th scope="col">genename</th><th scope="col">GC_content</th><th scope="col">SNV_rate</th><th scope="col">indel_count</th><th scope="col">length</th></tr>
    <tr><th></th><th scope="col">&lt;fct&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;int&gt;</th><th scope="col">&lt;int&gt;</th></tr>
</thead>
<tbody>
    <tr><th scope="row">1</th><td>OR4F5 </td><td>0.4281050000000000133049</td><td>0.00001156033999999999950243</td><td>0</td><td> 918</td></tr>
    <tr><th scope="row">2</th><td>SAMD11</td><td>0.6826832583463340320407</td><td>0.00007795386489999999845429</td><td>0</td><td>3205</td></tr>
    <tr><th scope="row">3</th><td>NOC2L </td><td>0.5827767403863509665385</td><td>0.00009993501699999999762520</td><td>0</td><td>5539</td></tr>
</tbody>
</table>




<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">N=rep(6430,nrow(df))</span><br><span class="line"></span><br><span class="line">df$SNV_2N = 2 * N * df$SNV_rate</span><br><span class="line"></span><br><span class="line">res=glm(indel_count~SNV_2N + GC_content,family=poisson(link=&quot;log&quot;),data=df) </span><br><span class="line">print(summary(res))</span><br><span class="line"></span><br><span class="line">coef=as.numeric(coef(res)) </span><br><span class="line">df$INDEL_RATE=exp(coef[1] + coef[2]*df$SNV_2N + coef[3]*df$GC_content) ##### b</span><br><span class="line"></span><br><span class="line"># fwrite(df[,c(1,7,5)],&apos;/storage11_7T/fuy/TADA-A/cell_WES/DNM/indel_rate/INDEL&apos;,sep=&quot;\t&quot;,col.names=F)</span><br></pre></td></tr></table></figure>


<pre><code>Call:
glm(formula = indel_count ~ SNV_2N + GC_content, family = poisson(link = &quot;log&quot;), 
    data = df)

Deviance Residuals: 
                   Min                      1Q                  Median  
-5.3747723046756075149  -0.2721079831257477433  -0.2543498238112971155  
                    3Q                     Max  
-0.2414373113538021975   7.6654372768663812110  

Coefficients:
                           Estimate              Std. Error
(Intercept) -3.55875256114883642411  0.24395625412901122964
SNV_2N       0.32595843840840044159  0.01404195081335555921
GC_content  -0.26301045582160398340  0.47999476989334100008
                            z value Pr(&gt;|z|)    
(Intercept) -14.5876699999999992485  &lt; 2e-16 ***
SNV_2N       23.2131900000000008788  &lt; 2e-16 ***
GC_content   -0.5479399999999999826  0.58373    
---
Signif. codes:  
  0 ‘***’ 0.001000000000000000020817 ‘**’ 0.01000000000000000020817 ‘*’
  0.05000000000000000277556 ‘.’ 0.1000000000000000055511 ‘ ’ 1

(Dispersion parameter for poisson family taken to be 1)

    Null deviance: 4441.4194643153414290  on 17477  degrees of freedom
Residual deviance: 4263.1231826492557957  on 17475  degrees of freedom
AIC: 5387.1954605381806687

Number of Fisher Scoring iterations: 14</code></pre><h4 id="frameshift-rate"><a href="#frameshift-rate" class="headerlink" title="frameshift rate"></a>frameshift rate</h4><p>$\frac{the , number , of ,frameshift}{the , number , of , indel}$<br>in unaffected siblings.</p>
<p>fs_fraction = 0.744186</p>
<h3 id="Estimation-of-frameshift-effect-size"><a href="#Estimation-of-frameshift-effect-size" class="headerlink" title="Estimation of frameshift effect size"></a>Estimation of frameshift effect size</h3><ul>
<li>EM</li>
</ul>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br></pre></td><td class="code"><pre><span class="line">setwd(&quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/indel_rate/test&quot;)</span><br><span class="line">options(scipen=200)</span><br><span class="line">options(digits=22)</span><br><span class="line">gama= 2 ### initial gamma</span><br><span class="line">N = 6430 ### sample size</span><br><span class="line">fs_fraction = 0.744186 ### fraction of frameshift</span><br><span class="line"></span><br><span class="line">Mydata=read.table(&apos;17478_gene_obs_count_risk_prior.txt&apos;,header=T)</span><br><span class="line">Mydata1=read.table(&apos;prior&apos;,header=T)</span><br><span class="line">IR=read.table(&apos;/storage11_7T/fuy/TADA-A/cell_WES/DNM/indel_rate/INDEL&apos;)</span><br><span class="line">colnames(IR) = c(&quot;genename&quot;,&quot;sum_indel_rate&quot;,&quot;length&quot;)</span><br><span class="line"></span><br><span class="line">mg = merge(Mydata,IR,by=&quot;genename&quot;)</span><br><span class="line"></span><br><span class="line">mg2 = merge(mg,Mydata1,by=&quot;genename&quot;)[,-3]</span><br><span class="line">colnames(mg2)[2] = &quot;observed_frameshift_count&quot;</span><br><span class="line"></span><br><span class="line"># p = mg2$risk</span><br><span class="line"># p = rep(.05,nrow(mg2))</span><br><span class="line"># p = runif(nrow(mg2),min=1e-10,max=1)</span><br><span class="line"></span><br><span class="line">repeat&#123;</span><br><span class="line">  mydata3 = p</span><br><span class="line"></span><br><span class="line">  mydata4 = mg2$observed_frameshift_count          </span><br><span class="line">#   mydata5 = fs_fraction*2*N*mg2$sum_indel_rate</span><br><span class="line">  mydata5 = fs_fraction * mg2$sum_indel_rate</span><br><span class="line">    </span><br><span class="line">  p=(mydata3*dpois(mydata4,gama*mydata5,log=FALSE))/((1-mydata3)*dpois(mydata4,mydata5,log=FALSE)+</span><br><span class="line">                                                          mydata3*dpois(mydata4,gama*mydata5,log=FALSE)) #代入公式</span><br><span class="line"></span><br><span class="line">  tes=gama</span><br><span class="line">  gama=sum(p*mydata4)/sum(p*mydata5)</span><br><span class="line">  if(abs(gama-tes)/tes &lt; 1e-8) break</span><br><span class="line">  </span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">q0 = 1-p  ### risk probability of non-risk genes</span><br><span class="line"></span><br><span class="line">mg2$post = p</span><br><span class="line">mg2$q0 = q0</span><br><span class="line"></span><br><span class="line">mg2 = mg2[order(mg2$q0),]</span><br><span class="line"></span><br><span class="line">FDR = c()</span><br><span class="line">for (i in 1:nrow(mg2)) FDR[i] &lt;- sum(mg2$q0[1:i]) / i</span><br><span class="line">mg2$FDR = FDR</span><br><span class="line"></span><br><span class="line">mg2$FDR2 =  mg2$q0 * nrow(mg2) / rank(mg2$q0)  # mg2$FDR3 = p.adjust(mg2$q0,method=&quot;fdr&quot;,n=nrow(mg2))</span><br><span class="line"></span><br><span class="line">nrow(mg2[mg2$FDR&lt;0.1,])</span><br><span class="line"></span><br><span class="line">gama</span><br><span class="line">log(gama)</span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">readRDS(&quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/indel_rate/indel_EM.rds&quot;)</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.frame: 2 × 4</caption>
<thead>
    <tr><th scope="col">n_genes</th><th scope="col">gama</th><th scope="col">log_gm</th><th scope="col">desc</th></tr>
    <tr><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;fct&gt;</th></tr>
</thead>
<tbody>
    <tr><td>136</td><td> 3.742353998933570125018</td><td>1.319714824909679951048</td><td>update gama        </td></tr>
    <tr><td>435</td><td>36.893193501840499948230</td><td>3.608027076087319873210</td><td>update gama and pi0</td></tr>
</tbody>
</table>



<ul>
<li><p>Q:</p>
<ul>
<li><p>difference between FDR and FDR2</p>
</li>
<li><p>若更新pi0，则初始pi0几乎无影响</p>
</li>
</ul>
</li>
</ul>
<ul>
<li>validate gamma by simulation</li>
</ul>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">mg2[1:n_risk_genes,]$sim_fs_ct = rpois(n_risk_genes,</span><br><span class="line">                                       fs_fraction * mg2[1:n_risk_genes,]$sum_indel_rate * sim_gama)</span><br><span class="line">    </span><br><span class="line">mg2[(n_risk_genes + 1):nrow(mg2),]$sim_fs_ct = rpois(nrow(mg2) - n_risk_genes, </span><br><span class="line">                                                     fs_fraction * mg2[(n_risk_genes + 1):nrow(mg2),]$sum_indel_rate)</span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">100 rounds </span><br><span class="line">n risk genes = all genes</span><br><span class="line">res_gama(74) &gt; sim_gama (50)</span><br></pre></td></tr></table></figure>

<h3 id="SNV-model"><a href="#SNV-model" class="headerlink" title="SNV model"></a>SNV model</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">pi0 = rep(0.05,n_genes)</span><br><span class="line">update gama only(NaN)</span><br></pre></td></tr></table></figure>

<h3 id="New-annotations"><a href="#New-annotations" class="headerlink" title="New annotations"></a>New annotations</h3><ul>
<li>spliceAI<ul>
<li>DS_AG    Delta score (acceptor gain)</li>
<li>DS_AL    Delta score (acceptor loss)</li>
<li>DS_DG    Delta score (donor gain)</li>
<li>DS_DL    Delta score (donor loss)</li>
</ul>
</li>
</ul>
<ul>
<li>PPI<ul>
<li>Interactome INSIDER (probably damaging) &amp; MPC &gt; 2 </li>
</ul>
</li>
<li>trap score (no responding)</li>
<li>deepsea clustering </li>
</ul>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">setwd(&quot;/storage11_7T/fuy/TADA-A/annotation/spliceai&quot;)</span><br><span class="line">readRDS(&quot;spliceai_sep_RR4_annota.rds&quot;)</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.frame: 4 × 4</caption>
<thead>
    <tr><th scope="col">logRR</th><th scope="col">lower_bound</th><th scope="col">upper_bound</th><th scope="col">annota</th></tr>
    <tr><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;chr&gt;</th></tr>
</thead>
<tbody>
    <tr><td> 2.0294776266614920068321</td><td>  1.1561569801227173925895</td><td> 2.902798273200266621075</td><td>DS_AG (acceptor gain)</td></tr>
    <tr><td>-0.9999999671218416930074</td><td>-12.3062343578646924413533</td><td>10.306234423621010165562</td><td>DS_AL (acceptor loss)</td></tr>
    <tr><td> 2.0682421448743815162175</td><td>  1.2534159904270909535740</td><td> 2.883068299321672078861</td><td>DS_DG (donor gain)   </td></tr>
    <tr><td> 0.3651119660030998637090</td><td> -0.5284987397323961388906</td><td> 1.258722671738595977331</td><td>DS_DL (donor loss)   </td></tr>
</tbody>
</table>




<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">readRDS(&quot;/storage11_7T/fuy/TADA-A/annotation/PPI/PPI_rr.rds&quot;)</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.frame: 1 × 4</caption>
<thead>
    <tr><th scope="col">logRR</th><th scope="col">lower_bound</th><th scope="col">upper_bound</th><th scope="col">annota</th></tr>
    <tr><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;chr&gt;</th></tr>
</thead>
<tbody>
    <tr><td>2.939515000000000100044</td><td>2.570340999999999986869</td><td>3.308688000000000073442</td><td>Interactome INSIDER (probably damaging) &amp; MPC &gt; 2</td></tr>
</tbody>
</table>



<h3 id="CHD-data"><a href="#CHD-data" class="headerlink" title="CHD data"></a>CHD data</h3><ul>
<li>2871 CHD probands from <a href="https://pubmed.ncbi.nlm.nih.gov/28991257/" target="_blank" rel="noopener">Contribution of rare inherited and de novo variants in 2,871 congenital heart disease probands</a></li>
</ul>
</div><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">Author: </span><span class="post-copyright-info"><a href="mailto:undefined" target="_blank" rel="noopener"></a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">Link: </span><span class="post-copyright-info"><a href="https://yfu1116.github.io/project/2021-03-18-tadaA-report/">https://yfu1116.github.io/project/2021-03-18-tadaA-report/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">Copyright Notice: </span><span class="post-copyright-info">All articles in this blog are licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank" rel="noopener">CC BY-NC-SA 4.0</a> unless stating additionally.</span></div></div><div class="tag_share"><div class="post-meta__tag-list"></div><div class="post_share"><div class="social-share" data-image="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" data-sites="github,email,facebook,twitter,weixin,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js/dist/css/share.min.css"><script src="https://cdn.jsdelivr.net/npm/social-share.js/dist/js/social-share.min.js" defer></script></div></div><nav class="pagination-post" id="pagination"><div class="prev-post pull-left"><a href="/project/2021-03-22-simulation/"><img class="prev-cover" data-lazy-src="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" onerror="onerror=null;src='/project/img/404.jpg'"><div class="pagination-info"><div class="label">Previous Post</div><div class="prev_info">2021-03-22-simulation</div></div></a></div><div class="next-post pull-right"><a href="/project/2021-01-13-tadaA-report/"><img class="next-cover" data-lazy-src="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" onerror="onerror=null;src='/project/img/404.jpg'"><div class="pagination-info"><div class="label">Next Post</div><div class="next_info">2021-01-13-tadaA-report</div></div></a></div></nav></article></main><footer id="footer" data-type="color"><div id="footer-wrap"><div class="copyright">&copy;2020 - 2021 By </div><div class="framework-info"><span>Framework </span><a href="https://hexo.io" target="_blank" rel="noopener">Hexo</a><span class="footer-separator">|</span><span>Theme </span><a href="https://github.com/jerryc127/hexo-theme-butterfly" target="_blank" rel="noopener">Butterfly</a></div></div></footer></div><section id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="Read Mode"><i class="fas fa-book-open"></i></button><button id="font_plus" type="button" title="Increase Font Size"><i class="fas fa-plus"></i></button><button id="font_minus" type="button" title="Decrease Font Size"><i class="fas fa-minus"></i></button><button id="darkmode" type="button" title="Switch Between Light And Dark Mode"><i class="fas fa-adjust"></i></button></div><div id="rightside-config-show"><button id="rightside_config" type="button" title="Setting"><i class="fas fa-cog"></i></button><button class="close" id="mobile-toc-button" type="button" title="Table Of Contents"><i class="fas fa-list-ul"></i></button><button id="go-up" type="button" title="Back To Top"><i class="fas fa-arrow-up"></i></button></div></section><div><script src="https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js"></script><script src="/project/js/utils.js"></script><script src="/project/js/main.js"></script><script src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.js"></script><script src="https://cdn.jsdelivr.net/npm/instant.page/instantpage.min.js" type="module" defer></script><script src="https://cdn.jsdelivr.net/npm/vanilla-lazyload/dist/lazyload.iife.min.js" async></script><div class="js-pjax"><script>if (!window.MathJax) {
  window.MathJax = {
    loader: {
      source: {
        '[tex]/amsCd': '[tex]/amscd'
      }
    },
    tex: {
      inlineMath: [ ['$','$'], ["\\(","\\)"]],
      tags: 'ams'
    },
    options: {
      renderActions: {
        findScript: [10, doc => {
          for (const node of document.querySelectorAll('script[type^="math/tex"]')) {
            const display = !!node.type.match(/; *mode=display/)
            const math = new doc.options.MathItem(node.textContent, doc.inputJax[0], display)
            const text = document.createTextNode('')
            node.parentNode.replaceChild(text, node)
            math.start = {node: text, delim: '', n: 0}
            math.end = {node: text, delim: '', n: 0}
            doc.math.push(math)
          }
        }, '']
      }
    }
  }
  
  var script = document.createElement('script')
  script.src = 'https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js'
  script.id = 'MathJax-script'
  script.async = true
  document.head.appendChild(script)
} else {
  MathJax.startup.document.state(0)
  MathJax.texReset()
  MathJax.typeset()
}</script><script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div></div></body></html>