<!DOCTYPE html><html lang="en" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1"><title>2021-01-13-tadaA-report | yang's blog</title><meta name="description" content="DNM data we used in the analysis1readRDS(&quot;&#x2F;storage11_7T&#x2F;fuy&#x2F;TADA-A&#x2F;cell_WES&#x2F;DNM&#x2F;DNM_descrip.rds&quot;)    A data.frame: 2 × 3      n_samplen_DNMdescrip     &lt;dbl&gt;&lt;dbl&gt;&lt;chr&gt;"><meta name="author" content=""><meta name="copyright" content=""><meta name="format-detection" content="telephone=no"><link rel="shortcut icon" href="/project/img/favicon.png"><link rel="canonical" href="https://yfu1116.github.io/project/2021-01-13-tadaA-report/"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//fonts.googleapis.com" crossorigin="crossorigin"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><meta property="og:type" content="article"><meta property="og:title" content="2021-01-13-tadaA-report"><meta property="og:url" content="https://yfu1116.github.io/project/2021-01-13-tadaA-report/"><meta property="og:site_name" content="yang's blog"><meta property="og:description" content="DNM data we used in the analysis1readRDS(&quot;&#x2F;storage11_7T&#x2F;fuy&#x2F;TADA-A&#x2F;cell_WES&#x2F;DNM&#x2F;DNM_descrip.rds&quot;)    A data.frame: 2 × 3      n_samplen_DNMdescrip     &lt;dbl&gt;&lt;dbl&gt;&lt;chr&gt;"><meta property="og:image" content="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg"><meta property="article:published_time" content="2021-01-21T01:16:17.088Z"><meta property="article:modified_time" content="2021-01-21T01:16:47.006Z"><meta name="twitter:card" content="summary"><script>var activateDarkMode = function () {
  document.documentElement.setAttribute('data-theme', 'dark')
  if (document.querySelector('meta[name="theme-color"]') !== null) {
    document.querySelector('meta[name="theme-color"]').setAttribute('content', '#000')
  }
}
var activateLightMode = function () {
  document.documentElement.setAttribute('data-theme', 'light')
  if (document.querySelector('meta[name="theme-color"]') !== null) {
    document.querySelector('meta[name="theme-color"]').setAttribute('content', '#fff')
  }
}

var getCookies = function (name) {
  const value = `; ${document.cookie}`
  const parts = value.split(`; ${name}=`)
  if (parts.length === 2) return parts.pop().split(';').shift()
}

var autoChangeMode = 'false'
var t = getCookies('theme')
if (autoChangeMode === '1') {
  var isDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches
  var isLightMode = window.matchMedia('(prefers-color-scheme: light)').matches
  var isNotSpecified = window.matchMedia('(prefers-color-scheme: no-preference)').matches
  var hasNoSupport = !isDarkMode && !isLightMode && !isNotSpecified

  if (t === undefined) {
    if (isLightMode) activateLightMode()
    else if (isDarkMode) activateDarkMode()
    else if (isNotSpecified || hasNoSupport) {
      console.log('You specified no preference for a color scheme or your browser does not support it. I Schedule dark mode during night time.')
      var now = new Date()
      var hour = now.getHours()
      var isNight = hour <= 6 || hour >= 18
      isNight ? activateDarkMode() : activateLightMode()
    }
    window.matchMedia('(prefers-color-scheme: dark)').addListener(function (e) {
      if (Cookies.get('theme') === undefined) {
        e.matches ? activateDarkMode() : activateLightMode()
      }
    })
  } else if (t === 'light') activateLightMode()
  else activateDarkMode()
} else if (autoChangeMode === '2') {
  now = new Date()
  hour = now.getHours()
  isNight = hour <= 6 || hour >= 18
  if (t === undefined) isNight ? activateDarkMode() : activateLightMode()
  else if (t === 'light') activateLightMode()
  else activateDarkMode()
} else {
  if (t === 'dark') activateDarkMode()
  else if (t === 'light') activateLightMode()
}</script><link rel="stylesheet" href="/project/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.css"><link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Titillium+Web&amp;display=swap"><script>var GLOBAL_CONFIG = { 
  root: '/project/',
  hexoversion: '4.0.0',
  algolia: undefined,
  localSearch: undefined,
  translate: undefined,
  noticeOutdate: undefined,
  copy: {
    success: 'Copy successfully',
    error: 'Copy error',
    noSupport: 'The browser does not support'
  },
  bookmark: {
    message_prev: 'Press',
    message_next: 'to bookmark this page'
  },
  runtime_unit: 'days',
  runtime: false,
  copyright: undefined,
  ClickShowText: undefined,
  medium_zoom: false,
  fancybox: true,
  Snackbar: undefined,
  justifiedGallery: {
    js: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/js/jquery.justifiedGallery.min.js',
    css: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/css/justifiedGallery.min.css'
  },
  baiduPush: false,
  highlightCopy: true,
  highlightLang: true,
  isPhotoFigcaption: false,
  islazyload: true,
  isanchor: false    
}</script><script id="config_change">var GLOBAL_CONFIG_SITE = { 
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isSidebar: true,
  postUpdate: '2021-01-21 09:16:47'
}</script><noscript><style>
#nav {
  opacity: 1
}
.justified-gallery img {
  opacity: 1
}
</style></noscript></head><body><div id="mobile-sidebar"><div id="menu_mask"></div><div id="mobile-sidebar-menus"><div class="mobile_author_icon"><img class="avatar-img" data-lazy-src="/project/img/avatar.png" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="mobile_post_data"><div class="mobile_data_item is-center"><div class="mobile_data_link"><a href="/project/archives/"><div class="headline">Articles</div><div class="length_num">71</div></a></div></div></div><hr/></div></div><div id="body-wrap"><div id="sidebar"><i class="fas fa-arrow-right on" id="toggle-sidebar"></i><div class="sidebar-toc"><div class="sidebar-toc__title">Catalog</div><div class="sidebar-toc__progress"><span class="progress-notice">You've read</span><span class="progress-num">0</span><span class="progress-percentage">%</span><div class="sidebar-toc__progress-bar">     </div></div><div class="sidebar-toc__content"><ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#DNM-data-we-used-in-the-analysis"><span class="toc-number">1.</span> <span class="toc-text">DNM data we used in the analysis</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Annotation-categories-from-the-paper-in-Cell"><span class="toc-number">2.</span> <span class="toc-text">Annotation categories from the paper in Cell</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#RR-and-burden"><span class="toc-number">2.1.</span> <span class="toc-text">RR and burden</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Other-annotations"><span class="toc-number">3.</span> <span class="toc-text">Other annotations</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#RR"><span class="toc-number">3.1.</span> <span class="toc-text">RR</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#DeepSEA"><span class="toc-number">3.1.1.</span> <span class="toc-text">DeepSEA</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Annotations-apart-from-DeepSEA"><span class="toc-number">3.1.2.</span> <span class="toc-text">Annotations apart from DeepSEA</span></a></li></ol></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#risk-genes"><span class="toc-number">4.</span> <span class="toc-text">risk genes</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Enrichment-analysis"><span class="toc-number">5.</span> <span class="toc-text">Enrichment analysis</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#candidate-variants"><span class="toc-number">6.</span> <span class="toc-text">candidate variants</span></a></li></ol></div></div></div><header class="post-bg" id="page-header" style="background-image: url(https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg)"><nav id="nav"><span class="pull-left" id="blog_name"><a class="blog_title" id="site-name" href="/project/">yang's blog</a></span><span class="pull-right menus"><span class="toggle-menu close"><a class="site-page"><i class="fas fa-bars fa-fw"></i></a></span></span></nav><div id="post-info"><div id="post-title"><div class="posttitle">2021-01-13-tadaA-report</div></div><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">Created</span><time class="post-meta-date-created" datetime="2021-01-21T01:16:17.088Z" title="Created 2021-01-21 09:16:17">2021-01-21</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">Updated</span><time class="post-meta-date-updated" datetime="2021-01-21T01:16:47.006Z" title="Updated 2021-01-21 09:16:47">2021-01-21</time></span></div><div class="meta-secondline"> <span class="post-meta-separator">|</span><span class="post-meta-pv-cv"><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">Post View:</span><span id="busuanzi_value_page_pv"></span></span></div></div></div></header><main class="layout_post" id="content-inner"><article id="post"><div class="post-content" id="article-container"><h2 id="DNM-data-we-used-in-the-analysis"><a href="#DNM-data-we-used-in-the-analysis" class="headerlink" title="DNM data we used in the analysis"></a>DNM data we used in the analysis</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">readRDS(&quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/DNM_descrip.rds&quot;)</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.frame: 2 × 3</caption>
<thead>
    <tr><th></th><th scope="col">n_sample</th><th scope="col">n_DNM</th><th scope="col">descrip</th></tr>
    <tr><th></th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;chr&gt;</th></tr>
</thead>
<tbody>
    <tr><th scope="row">affected</th><td>6430</td><td>6788</td><td>coding autosomal SNV in 17478 LoF-intolerant genes</td></tr>
    <tr><th scope="row">sibling</th><td>2179</td><td>2149</td><td>coding autosomal SNV in 17478 LoF-intolerant genes</td></tr>
</tbody>
</table>



<h2 id="Annotation-categories-from-the-paper-in-Cell"><a href="#Annotation-categories-from-the-paper-in-Cell" class="headerlink" title="Annotation categories from the paper in Cell"></a>Annotation categories from the paper in Cell</h2><p><code>annota (VEP)</code>:  annotation catagories from the paper in Cell</p>
<p><code>burden (aff / sib)</code> $= \frac{ASD , nonsynonymous , SNVs \quad / \quad ASD , synonymous , SNVs}{control , ASD , nonsynonymous , SNVs \quad / \quad control , ASD , synonymous , SNVs}$</p>
<p><code>logRR_paper</code> $= 1 , + , \frac{burden_paper , (aff , / , sib) , -  , 1 }{0.05}$</p>
<p><code>sep</code>: the left columns are derived from the paper, the right columns are calculated from the DNM data we used. The estimation is performed after calibration.</p>
<p><code>burden (obs / bg)</code> $= \frac{observed  }{background}$</p>
<p><code>logRR_formula</code> $= 1 , + , \frac{burden , (obs , / , bg) , -  , 1 }{0.05}$</p>
<p><code>separate_logRR</code>: separately estimated RR</p>
<p><code>joint_logRR</code>: jointly estimated RR of annotations 1-5, which were used in that paper.</p>
<h3 id="RR-and-burden"><a href="#RR-and-burden" class="headerlink" title="RR and burden"></a>RR and burden</h3><p>In TADA-A model, we calibrated mutation rate using the number of synonymous mutations. A uniform prior of 0.05 was used to estimate RR and to calculate posterior probability.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">readRDS(&quot;cell_annota_tadaA_bd_RR.rds&quot;)</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.frame: 7 × 9</caption>
<thead>
    <tr><th scope="col">annota (VEP)</th><th scope="col">burden_paper (aff / sib)</th><th scope="col">logRR_paper</th><th scope="col">sep</th><th scope="col">burden (aff / sib)</th><th scope="col">burden (obs/bg)</th><th scope="col">logRR_formula</th><th scope="col">separate_logRR</th><th scope="col">joint_logRR</th></tr>
    <tr><th scope="col">&lt;fct&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;chr&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>PTV_Highest(pLI=0.995-1) </td><td>3.5194598</td><td> 3.9394280</td><td>|</td><td>6.430804</td><td>3.621622</td><td>3.978418</td><td>3.493790881</td><td>3.3180267</td></tr>
    <tr><td>PTV_Middle(pLI=0.5-0.995)</td><td>1.3462415</td><td> 2.0700008</td><td>|</td><td>1.649148</td><td>1.459459</td><td>2.321327</td><td>2.209709635</td><td>0.3995045</td></tr>
    <tr><td>PTV_Lowest(pLI=0-0.5)    </td><td>0.9917312</td><td>-0.1807738</td><td>|</td><td>1.195162</td><td>1.445312</td><td>2.293166</td><td>2.081597693</td><td>1.9613477</td></tr>
    <tr><td>Missense_Highest(MPC≥2) </td><td>2.0541788</td><td> 3.0948341</td><td>|</td><td>2.079337</td><td>1.567686</td><td>2.513957</td><td>2.321866609</td><td>2.2135460</td></tr>
    <tr><td>Missense_Middle(MPC=1-2) </td><td>1.1572499</td><td> 1.4219022</td><td>|</td><td>1.183534</td><td>1.222964</td><td>1.697317</td><td>1.623570149</td><td>1.4426293</td></tr>
    <tr><td><span style="white-space:pre-wrap">Missense_Lowest(MPC&lt;1)   </span></td><td>0.9768610</td><td>-0.6213461</td><td>|</td><td>0.995654</td><td>1.219199</td><td>1.683427</td><td>1.254158652</td><td><span style="white-space:pre-wrap">       NA</span></td></tr>
    <tr><td>Synonymous               </td><td>1.0000000</td><td> 0.0000000</td><td>|</td><td>1.008760</td><td>1.000000</td><td>0.000000</td><td>0.003533577</td><td>       NA</td></tr>
</tbody>
</table>



<p>For <code>PTV</code> categories, the differences of the results between the paper and ours are partly caused by the loss of INDEL in our model, and the discrepancy of coding regions. </p>
<p>For <code>MPC&gt;=2</code>, it may mainly caused by the model.</p>
<h2 id="Other-annotations"><a href="#Other-annotations" class="headerlink" title="Other annotations"></a>Other annotations</h2><ul>
<li>We derived a set of priors as <code>1–FDR</code> from that paper, then used the top 1000 genes for estimating relative risks.</li>
</ul>
<h3 id="RR"><a href="#RR" class="headerlink" title="RR"></a>RR</h3><h4 id="DeepSEA"><a href="#DeepSEA" class="headerlink" title="DeepSEA"></a>DeepSEA</h4><p>The results of 217 DeepSEA categories (top 5% mutations) are similar, which can be downloaded <a href="https://github.com/yfu1116/yfu1116.github.io/blob/master/2021-01-141-FDR_all_rbp_top1000gene_separate_RR.xlsx" target="_blank" rel="noopener">here</a>.</p>
<p>we selected 5 significant brain-related categories to refit the model. </p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ab[,-5] # estimate separately</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.frame: 5 × 4</caption>
<thead>
    <tr><th></th><th scope="col">separate_logRR</th><th scope="col">lower_bound</th><th scope="col">upper_bound</th><th scope="col">annota</th></tr>
    <tr><th></th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;chr&gt;</th></tr>
</thead>
<tbody>
    <tr><th scope="row">9</th><td>1.752452</td><td>1.3571405</td><td>2.147764</td><td>ago_adult_brain.BA4.hg19                </td></tr>
    <tr><th scope="row">10</th><td>1.074426</td><td>0.5180078</td><td>1.630844</td><td>ago_adult_brain.Cingulate.gyrus.hg19    </td></tr>
    <tr><th scope="row">11</th><td>1.479920</td><td>1.0291704</td><td>1.930669</td><td>elavl_Adult_brain.all_human_samples.hg19</td></tr>
    <tr><th scope="row">12</th><td>1.427915</td><td>0.9616153</td><td>1.894214</td><td>elavl_Adult_brain.BA9_Alzheimer.hg19    </td></tr>
    <tr><th scope="row">13</th><td>1.442728</td><td>0.9821886</td><td>1.903266</td><td>elavl_Adult_brain.BA9.hg19              </td></tr>
</tbody>
</table>



<h4 id="Annotations-apart-from-DeepSEA"><a href="#Annotations-apart-from-DeepSEA" class="headerlink" title="Annotations apart from DeepSEA"></a>Annotations apart from DeepSEA</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">rbind(a,b) # estimate separately</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.frame: 9 × 4</caption>
<thead>
    <tr><th scope="col">logRR</th><th scope="col">lower_bound</th><th scope="col">upper_bound</th><th scope="col">annota</th></tr>
    <tr><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;chr&gt;</th></tr>
</thead>
<tbody>
    <tr><td> 1.7849791</td><td> 1.6493587</td><td>1.9205995</td><td>coding constraint &gt;90</td></tr>
    <tr><td> 1.4374067</td><td> 0.6965003</td><td>2.1783131</td><td>CLIPdb               </td></tr>
    <tr><td> 1.3173839</td><td> 1.0880712</td><td>1.5466965</td><td>RADAR_RBP top5%      </td></tr>
    <tr><td> 0.9380841</td><td> 0.4834046</td><td>1.3927636</td><td>RBP-VarDB            </td></tr>
    <tr><td> 1.1453607</td><td> 0.5565725</td><td>1.7341488</td><td>ribosnitch           </td></tr>
    <tr><td> 1.9569585</td><td> 1.8443240</td><td>2.0695929</td><td>MVP                  </td></tr>
    <tr><td> 1.7428031</td><td> 1.6302765</td><td>1.8553297</td><td>primateAI            </td></tr>
    <tr><td> 1.8734926</td><td> 1.7359787</td><td>2.0110065</td><td>spidex               </td></tr>
    <tr><td>-1.0000000</td><td>-2.3013881</td><td>0.3013882</td><td>CADD                 </td></tr>
</tbody>
</table>




<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">a  ### joint estimation of all significant annotations</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.frame: 13 × 6</caption>
<thead>
    <tr><th scope="col">separate_logRR</th><th scope="col">lower_bound</th><th scope="col">upper_bound</th><th scope="col">annota</th><th scope="col">joint_RR</th><th scope="col">idx</th></tr>
    <tr><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;chr&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;int&gt;</th></tr>
</thead>
<tbody>
    <tr><td>1.7849791</td><td>1.6493587</td><td>1.920600</td><td><span style="white-space:pre-wrap">coding constraint &gt;90                   </span></td><td> 0.5993900</td><td> 1</td></tr>
    <tr><td>1.4374067</td><td>0.6965003</td><td>2.178313</td><td>CLIPdb                                  </td><td>-0.4573503</td><td> 2</td></tr>
    <tr><td>1.3173839</td><td>1.0880712</td><td>1.546697</td><td>RADAR_RBP top5%                         </td><td> 0.2315051</td><td> 3</td></tr>
    <tr><td>0.9380841</td><td>0.4834046</td><td>1.392764</td><td>RBP-VarDB                               </td><td> 0.4038981</td><td> 4</td></tr>
    <tr><td>1.1453607</td><td>0.5565725</td><td>1.734149</td><td>ribosnitch                              </td><td>-0.3492912</td><td> 5</td></tr>
    <tr><td>1.9569585</td><td>1.8443240</td><td>2.069593</td><td>MVP                                     </td><td> 1.0807593</td><td> 6</td></tr>
    <tr><td>1.7428031</td><td>1.6302765</td><td>1.855330</td><td>primateAI                               </td><td> 0.6242487</td><td> 7</td></tr>
    <tr><td>1.8734926</td><td>1.7359787</td><td>2.011007</td><td>spidex                                  </td><td> 1.1574875</td><td> 8</td></tr>
    <tr><td>1.7524525</td><td>1.3571405</td><td>2.147764</td><td>ago_adult_brain.BA4.hg19                </td><td> 0.5180716</td><td> 9</td></tr>
    <tr><td>1.0744256</td><td>0.5180078</td><td>1.630844</td><td>ago_adult_brain.Cingulate.gyrus.hg19    </td><td>-0.7293009</td><td>10</td></tr>
    <tr><td>1.4799195</td><td>1.0291704</td><td>1.930669</td><td>elavl_Adult_brain.all_human_samples.hg19</td><td>-0.1916567</td><td>11</td></tr>
    <tr><td>1.4279148</td><td>0.9616153</td><td>1.894214</td><td>elavl_Adult_brain.BA9_Alzheimer.hg19    </td><td>-0.9256815</td><td>12</td></tr>
    <tr><td>1.4427275</td><td>0.9821886</td><td>1.903266</td><td>elavl_Adult_brain.BA9.hg19              </td><td> 0.3635234</td><td>13</td></tr>
</tbody>
</table>



<h2 id="risk-genes"><a href="#risk-genes" class="headerlink" title="risk genes"></a>risk genes</h2><p><code>P</code>: annotations from the paper</p>
<p><code>O</code>: annotations from the above other sources</p>
<p><code>misA</code>: BFs of <code>mpc&gt;=2</code> from the paper</p>
<p><code>misB</code>: BFs of <code>2&gt;mpc&gt;=1</code> from the paper</p>
<p><code>dn ptv</code>: BFs of <code>denovo ptv</code> from the paper</p>
<p><code>cc ptv</code>: BFs of <code>case-control ptv</code> from the paper</p>
<p><code>novel_gene</code>: number of genes out of the list of <code>102 risk genes</code></p>
<p>The paper identified <code>102 risk genes</code> with q-value &lt; 0.1 when <code>de novo and case-control</code> data was used, and <code>65 genes</code> when only <code>de novo</code> data was used.</p>
<p>Using TADA-A model, we identified 47 genes when only <code>de novo SNVs</code> were considered, and 66 genes when a set of <code>BFs of case-control PTVs</code> from the paper were recruited.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"> ### compare the results of the paper with ours</span><br><span class="line">mg</span><br><span class="line">df[1:2,-8] ### details of TADA-A</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.frame: 4 × 5</caption>
<thead>
    <tr><th scope="col">Annota</th><th scope="col">Dataset</th><th scope="col">Model</th><th scope="col">other_info</th><th scope="col">n_risk_genes</th></tr>
    <tr><th scope="col">&lt;fct&gt;</th><th scope="col">&lt;fct&gt;</th><th scope="col">&lt;fct&gt;</th><th scope="col">&lt;fct&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>P</td><td>denovo SNVs/INDELs               </td><td>TADA+(the paper)</td><td>no           </td><td> 65</td></tr>
    <tr><td>P</td><td>denovo &amp; case-control SNVs/INDELs</td><td>TADA+(the paper)</td><td><span style="white-space:pre-wrap">no           </span></td><td>102</td></tr>
    <tr><td>P</td><td>denovo SNVs                      </td><td>TADA-A          </td><td>no           </td><td> 47</td></tr>
    <tr><td>P</td><td>denovo SNVs                      </td><td>TADA-A          </td><td>BFs of cc ptv</td><td> 66</td></tr>
</tbody>
</table>




<table>
<caption>A data.frame: 2 × 7</caption>
<thead>
    <tr><th></th><th scope="col">Annota</th><th scope="col">prior_for_estimate_RR</th><th scope="col">gene_for_estimate_RR</th><th scope="col">prior_for_cal_posterior</th><th scope="col">addtional_BFs_for_posterior</th><th scope="col">num_gene_identified</th><th scope="col">novel_gene</th></tr>
    <tr><th></th><th scope="col">&lt;fct&gt;</th><th scope="col">&lt;fct&gt;</th><th scope="col">&lt;fct&gt;</th><th scope="col">&lt;fct&gt;</th><th scope="col">&lt;fct&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><th scope="row">1</th><td>P</td><td>uniform</td><td>all</td><td>uniform</td><td>no    </td><td>47</td><td>10</td></tr>
    <tr><th scope="row">2</th><td>P</td><td>uniform</td><td>all</td><td>uniform</td><td>cc ptv</td><td>66</td><td>14</td></tr>
</tbody>
</table>



<p>Putting all selected annotations in TADA-A, we tested four scenarios as follows.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">### four scenarios</span><br><span class="line">df2</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.frame: 4 × 8</caption>
<thead>
    <tr><th></th><th scope="col">Annota</th><th scope="col">prior_for_estimate_RR</th><th scope="col">gene_for_estimate_RR</th><th scope="col">prior_for_cal_posterior</th><th scope="col">addtional_BFs_for_posterior</th><th scope="col">num_gene_identified</th><th scope="col">novel_gene</th><th scope="col">scenario</th></tr>
    <tr><th></th><th scope="col">&lt;fct&gt;</th><th scope="col">&lt;fct&gt;</th><th scope="col">&lt;fct&gt;</th><th scope="col">&lt;fct&gt;</th><th scope="col">&lt;fct&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;int&gt;</th></tr>
</thead>
<tbody>
    <tr><th scope="row">3</th><td>P+O</td><td>uniform             </td><td>all    </td><td>uniform</td><td>no                     </td><td> 60</td><td> 23</td><td>3</td></tr>
    <tr><th scope="row">4</th><td>P+O</td><td>uniform             </td><td>all    </td><td>uniform</td><td>cc ptv                 </td><td> 85</td><td> 33</td><td>4</td></tr>
    <tr><th scope="row">5</th><td>O  </td><td>1-FDR from the paper</td><td>top1000</td><td>uniform</td><td>misA*misB*dn ptv*cc ptv</td><td>217</td><td>123</td><td>5</td></tr>
    <tr><th scope="row">6</th><td>O  </td><td>1-FDR from the paper</td><td>top1000</td><td>uniform</td><td>no                     </td><td> 29</td><td> 13</td><td>6</td></tr>
</tbody>
</table>



<p>We took the result of scenario 3 for further analysis.</p>
<h2 id="Enrichment-analysis"><a href="#Enrichment-analysis" class="headerlink" title="Enrichment analysis"></a>Enrichment analysis</h2><ul>
<li>GWAS</li>
<li>functional gene lists</li>
</ul>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">readRDS(&quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/report/1-13_brain_gwas_EA.rds&quot;)</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.frame: 6 × 4</caption>
<thead>
    <tr><th></th><th scope="col">traits</th><th scope="col">enrich</th><th scope="col">pVal</th><th scope="col">qVal</th></tr>
    <tr><th></th><th scope="col">&lt;fct&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><th scope="row">2</th><td>Schizophrenia                           </td><td>2.271464</td><td>0.001156453</td><td>0.006938721</td></tr>
    <tr><th scope="row">4</th><td>Attention deficit hyperactivity disorder</td><td>3.711331</td><td>0.011461353</td><td>0.034384059</td></tr>
    <tr><th scope="row">5</th><td>Educational attainment                  </td><td>1.774402</td><td>0.020703425</td><td>0.041406851</td></tr>
    <tr><th scope="row">1</th><td>Autism spectrum disorder                </td><td>2.376599</td><td>0.088642274</td><td>0.132963411</td></tr>
    <tr><th scope="row">3</th><td>Major depressive disorder               </td><td>0.000000</td><td>1.000000000</td><td>1.000000000</td></tr>
    <tr><th scope="row">6</th><td>Height adjusted BMI                     </td><td>0.000000</td><td>1.000000000</td><td>1.000000000</td></tr>
</tbody>
</table>




<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">a</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.frame: 10 × 8</caption>
<thead>
    <tr><th></th><th scope="col">Geneset</th><th scope="col">enrich</th><th scope="col">pVal</th><th scope="col">n_novel_g</th><th scope="col">n_novel_g_geneset</th><th scope="col">n_geneset</th><th scope="col">n_total_genes</th><th scope="col">qVal</th></tr>
    <tr><th></th><th scope="col">&lt;chr&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;int&gt;</th><th scope="col">&lt;int&gt;</th><th scope="col">&lt;int&gt;</th><th scope="col">&lt;int&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><th scope="row">1</th><td>haploinsufficiency_including_all_without_ncscore     </td><td> 2.6695</td><td>0.000003</td><td>123</td><td>26</td><td>1384</td><td>17478</td><td>0.000010000</td></tr>
    <tr><th scope="row">2</th><td>top5%_brainspan_exp.gene                             </td><td> 1.5099</td><td>0.142765</td><td>123</td><td> 9</td><td> 847</td><td>17478</td><td>0.142765000</td></tr>
    <tr><th scope="row">3</th><td>Petrovski_plosgen_RVIS_score_top5_pct                </td><td> 3.3414</td><td>0.000004</td><td>123</td><td>19</td><td> 808</td><td>17478</td><td>0.000010000</td></tr>
    <tr><th scope="row">4</th><td>160210_GO_brain_genes                                </td><td> 1.3777</td><td>0.044349</td><td>123</td><td>28</td><td>2888</td><td>17478</td><td>0.049276667</td></tr>
    <tr><th scope="row">5</th><td>AutismKB                                             </td><td> 5.6388</td><td>0.001972</td><td>123</td><td> 5</td><td> 126</td><td>17478</td><td>0.003286667</td></tr>
    <tr><th scope="row">7</th><td>sfrai_genes_high_confidence_new_170717               </td><td>12.9180</td><td>0.010350</td><td>123</td><td> 2</td><td>  22</td><td>17478</td><td>0.014785714</td></tr>
    <tr><th scope="row">9</th><td>Darnell_Cell_2011_FMRP_targets                       </td><td> 4.3345</td><td>0.000000</td><td>123</td><td>23</td><td> 754</td><td>17478</td><td>0.000000000</td></tr>
    <tr><th scope="row">10</th><td>DAWN_new_q0.05                                       </td><td> 7.0944</td><td>0.000000</td><td>123</td><td>34</td><td> 681</td><td>17478</td><td>0.000000000</td></tr>
    <tr><th scope="row">11</th><td>Kenny_MP_2014_brain_functional                       </td><td> 3.4999</td><td>0.014426</td><td>123</td><td> 5</td><td> 203</td><td>17478</td><td>0.018032500</td></tr>
    <tr><th scope="row">12</th><td>Irimia_cell_2014_neuron_specific_alternative_splicing</td><td> 2.2705</td><td>0.000111</td><td>123</td><td>24</td><td>1502</td><td>17478</td><td>0.000222000</td></tr>
</tbody>
</table>



<h2 id="candidate-variants"><a href="#candidate-variants" class="headerlink" title="candidate variants"></a>candidate variants</h2><ul>
<li>DNMs in risk genes,<br>which are simultaneously annotated as DeepSEA catagories and coding constraint sequences or synonymous mutations, are candidates variants for further analysis.</li>
</ul>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">readRDS(&quot;2021-01-14_1-FDR_s5_RBP_6788SNV_sf_other_annota_pre_prior_1-13_res.rds&quot;)</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.frame: 19 × 9</caption>
<thead>
    <tr><th scope="col">idx</th><th scope="col">DeepSEA_RBP</th><th scope="col">genename</th><th scope="col">chr</th><th scope="col">start</th><th scope="col">end</th><th scope="col">REF</th><th scope="col">ALT</th><th scope="col">other_annota</th></tr>
    <tr><th scope="col">&lt;int&gt;</th><th scope="col">&lt;chr&gt;</th><th scope="col">&lt;chr&gt;</th><th scope="col">&lt;chr&gt;</th><th scope="col">&lt;int&gt;</th><th scope="col">&lt;int&gt;</th><th scope="col">&lt;chr&gt;</th><th scope="col">&lt;chr&gt;</th><th scope="col">&lt;chr&gt;</th></tr>
</thead>
<tbody>
    <tr><td>12</td><td>elavl_Adult_brain.BA9.hg19              </td><td>TFAP4  </td><td>chr16</td><td>  4312613</td><td>  4312614</td><td>G</td><td>A</td><td>ccr90, syn</td></tr>
    <tr><td> 9</td><td>ago_adult_brain.Cingulate.gyrus.hg19    </td><td>KDM6B  </td><td>chr17</td><td>  7755276</td><td>  7755277</td><td>G</td><td>A</td><td>ccr90, syn</td></tr>
    <tr><td>10</td><td>elavl_Adult_brain.all_human_samples.hg19</td><td>MEF2D  </td><td>chr1 </td><td>156452416</td><td>156452417</td><td>G</td><td>A</td><td>ccr90, syn</td></tr>
    <tr><td> 9</td><td>ago_adult_brain.Cingulate.gyrus.hg19    </td><td>FBN1   </td><td>chr15</td><td> 48766840</td><td> 48766841</td><td>T</td><td>C</td><td>ccr90, syn</td></tr>
    <tr><td>12</td><td>elavl_Adult_brain.BA9.hg19              </td><td>MYO9B  </td><td>chr19</td><td> 17318043</td><td> 17318044</td><td>A</td><td>G</td><td>ccr90, syn</td></tr>
    <tr><td> 8</td><td>ago_adult_brain.BA4.hg19                </td><td>DENND4A</td><td>chr15</td><td> 66048520</td><td> 66048521</td><td>A</td><td>G</td><td>ccr90, syn</td></tr>
    <tr><td>10</td><td>elavl_Adult_brain.all_human_samples.hg19</td><td>DENND4A</td><td>chr15</td><td> 66048520</td><td> 66048521</td><td>A</td><td>G</td><td>ccr90, syn</td></tr>
    <tr><td> 8</td><td>ago_adult_brain.BA4.hg19                </td><td>SHANK3 </td><td>chr22</td><td> 51117093</td><td> 51117094</td><td>C</td><td>G</td><td>ccr90, syn</td></tr>
    <tr><td> 8</td><td>ago_adult_brain.BA4.hg19                </td><td>SCN1A  </td><td>chr2 </td><td>166848858</td><td>166848859</td><td>C</td><td>G</td><td>ccr90, syn</td></tr>
    <tr><td>11</td><td>elavl_Adult_brain.BA9_Alzheimer.hg19    </td><td>SCN2A  </td><td>chr2 </td><td>166231425</td><td>166231426</td><td>A</td><td>T</td><td>ccr90, syn</td></tr>
    <tr><td> 8</td><td>ago_adult_brain.BA4.hg19                </td><td>TRAF7  </td><td>chr16</td><td>  2226079</td><td>  2226080</td><td>C</td><td>T</td><td>ccr90, syn</td></tr>
    <tr><td>12</td><td>elavl_Adult_brain.BA9.hg19              </td><td>GRIN2B </td><td>chr12</td><td> 13724821</td><td> 13724822</td><td>C</td><td>T</td><td>ccr90, syn</td></tr>
    <tr><td>12</td><td>elavl_Adult_brain.BA9.hg19              </td><td>GRIN2B </td><td>chr12</td><td> 13768559</td><td> 13768560</td><td>C</td><td>T</td><td>ccr90, syn</td></tr>
    <tr><td>12</td><td>elavl_Adult_brain.BA9.hg19              </td><td>FOXP1  </td><td>chr3 </td><td> 71021816</td><td> 71021817</td><td>C</td><td>T</td><td>ccr90, syn</td></tr>
    <tr><td> 8</td><td>ago_adult_brain.BA4.hg19                </td><td>ZC3H14 </td><td>chr14</td><td> 89034438</td><td> 89034439</td><td>C</td><td>T</td><td>ccr90, syn</td></tr>
    <tr><td> 8</td><td>ago_adult_brain.BA4.hg19                </td><td>MED13L </td><td>chr12</td><td>116424951</td><td>116424952</td><td>C</td><td>T</td><td>ccr90, syn</td></tr>
    <tr><td>12</td><td>elavl_Adult_brain.BA9.hg19              </td><td>SCN2A  </td><td>chr2 </td><td>166237635</td><td>166237636</td><td>C</td><td>T</td><td>ccr90, syn</td></tr>
    <tr><td>12</td><td>elavl_Adult_brain.BA9.hg19              </td><td>ZBTB20 </td><td>chr3 </td><td>114070427</td><td>114070428</td><td>G</td><td>T</td><td>ccr90, syn</td></tr>
    <tr><td>10</td><td>elavl_Adult_brain.all_human_samples.hg19</td><td>TBR1   </td><td>chr2 </td><td>162274306</td><td>162274307</td><td>G</td><td>T</td><td>ccr90, syn</td></tr>
</tbody>
</table>


</div><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">Author: </span><span class="post-copyright-info"><a href="mailto:undefined" target="_blank" rel="noopener"></a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">Link: </span><span class="post-copyright-info"><a href="https://yfu1116.github.io/project/2021-01-13-tadaA-report/">https://yfu1116.github.io/project/2021-01-13-tadaA-report/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">Copyright Notice: </span><span class="post-copyright-info">All articles in this blog are licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank" rel="noopener">CC BY-NC-SA 4.0</a> unless stating additionally.</span></div></div><div class="tag_share"><div class="post-meta__tag-list"></div><div class="post_share"><div class="social-share" data-image="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" data-sites="github,email,facebook,twitter,weixin,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js/dist/css/share.min.css"><script src="https://cdn.jsdelivr.net/npm/social-share.js/dist/js/social-share.min.js" defer></script></div></div><nav class="pagination-post" id="pagination"><div class="prev-post pull-left"><a href="/project/2021-03-18-tadaA-report/"><img class="prev-cover" data-lazy-src="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" onerror="onerror=null;src='/project/img/404.jpg'"><div class="pagination-info"><div class="label">Previous Post</div><div class="prev_info">2021-03-18-tadaA-report</div></div></a></div><div class="next-post pull-right"><a href="/project/201126-tadaA-report/"><img class="next-cover" data-lazy-src="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" onerror="onerror=null;src='/project/img/404.jpg'"><div class="pagination-info"><div class="label">Next Post</div><div class="next_info">201126-tadaA-report</div></div></a></div></nav></article></main><footer id="footer" data-type="color"><div id="footer-wrap"><div class="copyright">&copy;2020 - 2021 By </div><div class="framework-info"><span>Framework </span><a href="https://hexo.io" target="_blank" rel="noopener">Hexo</a><span class="footer-separator">|</span><span>Theme </span><a href="https://github.com/jerryc127/hexo-theme-butterfly" target="_blank" rel="noopener">Butterfly</a></div></div></footer></div><section id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="Read Mode"><i class="fas fa-book-open"></i></button><button id="font_plus" type="button" title="Increase Font Size"><i class="fas fa-plus"></i></button><button id="font_minus" type="button" title="Decrease Font Size"><i class="fas fa-minus"></i></button><button id="darkmode" type="button" title="Switch Between Light And Dark Mode"><i class="fas fa-adjust"></i></button></div><div id="rightside-config-show"><button id="rightside_config" type="button" title="Setting"><i class="fas fa-cog"></i></button><button class="close" id="mobile-toc-button" type="button" title="Table Of Contents"><i class="fas fa-list-ul"></i></button><button id="go-up" type="button" title="Back To Top"><i class="fas fa-arrow-up"></i></button></div></section><div><script src="https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js"></script><script src="/project/js/utils.js"></script><script src="/project/js/main.js"></script><script src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.js"></script><script src="https://cdn.jsdelivr.net/npm/instant.page/instantpage.min.js" type="module" defer></script><script src="https://cdn.jsdelivr.net/npm/vanilla-lazyload/dist/lazyload.iife.min.js" async></script><div class="js-pjax"><script>if (!window.MathJax) {
  window.MathJax = {
    loader: {
      source: {
        '[tex]/amsCd': '[tex]/amscd'
      }
    },
    tex: {
      inlineMath: [ ['$','$'], ["\\(","\\)"]],
      tags: 'ams'
    },
    options: {
      renderActions: {
        findScript: [10, doc => {
          for (const node of document.querySelectorAll('script[type^="math/tex"]')) {
            const display = !!node.type.match(/; *mode=display/)
            const math = new doc.options.MathItem(node.textContent, doc.inputJax[0], display)
            const text = document.createTextNode('')
            node.parentNode.replaceChild(text, node)
            math.start = {node: text, delim: '', n: 0}
            math.end = {node: text, delim: '', n: 0}
            doc.math.push(math)
          }
        }, '']
      }
    }
  }
  
  var script = document.createElement('script')
  script.src = 'https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js'
  script.id = 'MathJax-script'
  script.async = true
  document.head.appendChild(script)
} else {
  MathJax.startup.document.state(0)
  MathJax.texReset()
  MathJax.typeset()
}</script><script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div></div></body></html>