<!DOCTYPE html><html lang="en" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1"><title>2021-04-07-PPI-annotation | yang's blog</title><meta name="description" content="PPI-MPC212library(tadaA)setwd(&quot;&#x2F;storage11_7T&#x2F;fuy&#x2F;TADA-A&#x2F;annotation&quot;)   12345678910111213141516171819202122232425262728293031323334system.time(compact_data &lt;- TADA_A_read_info_by_chunks(mu"><meta name="author" content=""><meta name="copyright" content=""><meta name="format-detection" content="telephone=no"><link rel="shortcut icon" href="/project/img/favicon.png"><link rel="canonical" href="https://yfu1116.github.io/project/2021-04-07-PPI-annotation/"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//fonts.googleapis.com" crossorigin="crossorigin"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><meta property="og:type" content="article"><meta property="og:title" content="2021-04-07-PPI-annotation"><meta property="og:url" content="https://yfu1116.github.io/project/2021-04-07-PPI-annotation/"><meta property="og:site_name" content="yang's blog"><meta property="og:description" content="PPI-MPC212library(tadaA)setwd(&quot;&#x2F;storage11_7T&#x2F;fuy&#x2F;TADA-A&#x2F;annotation&quot;)   12345678910111213141516171819202122232425262728293031323334system.time(compact_data &lt;- TADA_A_read_info_by_chunks(mu"><meta property="og:image" content="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg"><meta property="article:published_time" content="2021-04-07T03:19:17.659Z"><meta property="article:modified_time" content="2021-04-07T03:19:44.859Z"><meta name="twitter:card" content="summary"><script>var activateDarkMode = function () {
  document.documentElement.setAttribute('data-theme', 'dark')
  if (document.querySelector('meta[name="theme-color"]') !== null) {
    document.querySelector('meta[name="theme-color"]').setAttribute('content', '#000')
  }
}
var activateLightMode = function () {
  document.documentElement.setAttribute('data-theme', 'light')
  if (document.querySelector('meta[name="theme-color"]') !== null) {
    document.querySelector('meta[name="theme-color"]').setAttribute('content', '#fff')
  }
}

var getCookies = function (name) {
  const value = `; ${document.cookie}`
  const parts = value.split(`; ${name}=`)
  if (parts.length === 2) return parts.pop().split(';').shift()
}

var autoChangeMode = 'false'
var t = getCookies('theme')
if (autoChangeMode === '1') {
  var isDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches
  var isLightMode = window.matchMedia('(prefers-color-scheme: light)').matches
  var isNotSpecified = window.matchMedia('(prefers-color-scheme: no-preference)').matches
  var hasNoSupport = !isDarkMode && !isLightMode && !isNotSpecified

  if (t === undefined) {
    if (isLightMode) activateLightMode()
    else if (isDarkMode) activateDarkMode()
    else if (isNotSpecified || hasNoSupport) {
      console.log('You specified no preference for a color scheme or your browser does not support it. I Schedule dark mode during night time.')
      var now = new Date()
      var hour = now.getHours()
      var isNight = hour <= 6 || hour >= 18
      isNight ? activateDarkMode() : activateLightMode()
    }
    window.matchMedia('(prefers-color-scheme: dark)').addListener(function (e) {
      if (Cookies.get('theme') === undefined) {
        e.matches ? activateDarkMode() : activateLightMode()
      }
    })
  } else if (t === 'light') activateLightMode()
  else activateDarkMode()
} else if (autoChangeMode === '2') {
  now = new Date()
  hour = now.getHours()
  isNight = hour <= 6 || hour >= 18
  if (t === undefined) isNight ? activateDarkMode() : activateLightMode()
  else if (t === 'light') activateLightMode()
  else activateDarkMode()
} else {
  if (t === 'dark') activateDarkMode()
  else if (t === 'light') activateLightMode()
}</script><link rel="stylesheet" href="/project/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.css"><link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Titillium+Web&amp;display=swap"><script>var GLOBAL_CONFIG = { 
  root: '/project/',
  hexoversion: '4.0.0',
  algolia: undefined,
  localSearch: undefined,
  translate: undefined,
  noticeOutdate: undefined,
  copy: {
    success: 'Copy successfully',
    error: 'Copy error',
    noSupport: 'The browser does not support'
  },
  bookmark: {
    message_prev: 'Press',
    message_next: 'to bookmark this page'
  },
  runtime_unit: 'days',
  runtime: false,
  copyright: undefined,
  ClickShowText: undefined,
  medium_zoom: false,
  fancybox: true,
  Snackbar: undefined,
  justifiedGallery: {
    js: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/js/jquery.justifiedGallery.min.js',
    css: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/css/justifiedGallery.min.css'
  },
  baiduPush: false,
  highlightCopy: true,
  highlightLang: true,
  isPhotoFigcaption: false,
  islazyload: true,
  isanchor: false    
}</script><script id="config_change">var GLOBAL_CONFIG_SITE = { 
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isSidebar: true,
  postUpdate: '2021-04-07 11:19:44'
}</script><noscript><style>
#nav {
  opacity: 1
}
.justified-gallery img {
  opacity: 1
}
</style></noscript></head><body><div id="mobile-sidebar"><div id="menu_mask"></div><div id="mobile-sidebar-menus"><div class="mobile_author_icon"><img class="avatar-img" data-lazy-src="/project/img/avatar.png" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="mobile_post_data"><div class="mobile_data_item is-center"><div class="mobile_data_link"><a href="/project/archives/"><div class="headline">Articles</div><div class="length_num">64</div></a></div></div></div><hr/></div></div><div id="body-wrap"><div id="sidebar"><i class="fas fa-arrow-right on" id="toggle-sidebar"></i><div class="sidebar-toc"><div class="sidebar-toc__title">Catalog</div><div class="sidebar-toc__progress"><span class="progress-notice">You've read</span><span class="progress-num">0</span><span class="progress-percentage">%</span><div class="sidebar-toc__progress-bar">     </div></div><div class="sidebar-toc__content"><ol class="toc"><li class="toc-item toc-level-3"><a class="toc-link" href="#PPI-MPC2"><span class="toc-number">1.</span> <span class="toc-text">PPI-MPC2</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#PPI-only"><span class="toc-number">2.</span> <span class="toc-text">PPI only</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#PPH2-PPI-PPH2"><span class="toc-number">3.</span> <span class="toc-text">PPH2_PPI-PPH2</span></a></li></ol></div></div></div><header class="post-bg" id="page-header" style="background-image: url(https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg)"><nav id="nav"><span class="pull-left" id="blog_name"><a class="blog_title" id="site-name" href="/project/">yang's blog</a></span><span class="pull-right menus"><span class="toggle-menu close"><a class="site-page"><i class="fas fa-bars fa-fw"></i></a></span></span></nav><div id="post-info"><div id="post-title"><div class="posttitle">2021-04-07-PPI-annotation</div></div><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">Created</span><time class="post-meta-date-created" datetime="2021-04-07T03:19:17.659Z" title="Created 2021-04-07 11:19:17">2021-04-07</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">Updated</span><time class="post-meta-date-updated" datetime="2021-04-07T03:19:44.859Z" title="Updated 2021-04-07 11:19:44">2021-04-07</time></span></div><div class="meta-secondline"> <span class="post-meta-separator">|</span><span class="post-meta-pv-cv"><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">Post View:</span><span id="busuanzi_value_page_pv"></span></span></div></div></div></header><main class="layout_post" id="content-inner"><article id="post"><div class="post-content" id="article-container"><h3 id="PPI-MPC2"><a href="#PPI-MPC2" class="headerlink" title="PPI-MPC2"></a>PPI-MPC2</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">library(tadaA)</span><br><span class="line">setwd(&quot;/storage11_7T/fuy/TADA-A/annotation&quot;)</span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br></pre></td><td class="code"><pre><span class="line">system.time(compact_data &lt;- TADA_A_read_info_by_chunks(</span><br><span class="line">mut_files = &quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/affected/6788snv.affected.cd.auto.no_pli_rm.allele.bed&quot;,</span><br><span class="line">    </span><br><span class="line">window_file = &quot;/storage11_7T/fuy/TADA-A/db/MS_data/windows_partition/17478gene_hg19.87.auto.exon.wd.bed&quot; ,</span><br><span class="line">    </span><br><span class="line">mutrate_scaling_files = c(&quot;/storage11_7T/fuy/TADA-A/cell_WES/sf_1792_d_2058.1_cd_uniform_scaling_factors.txt&quot;),</span><br><span class="line"> </span><br><span class="line">sample_sizes = 6430, </span><br><span class="line">    </span><br><span class="line">gene_prior_file = &quot;/storage11_7T/fuy/TADA-A/db/MS_data/prior/new_uniform_gene_prior.txt&quot;,</span><br><span class="line">      </span><br><span class="line">nonAS_noncoding_annotations = NA, </span><br><span class="line"></span><br><span class="line">AS_noncoding_annotations = list(</span><br><span class="line"></span><br><span class="line">c(&quot;/storage11_7T/fuy/TADA-A/annotation/PPI/MPC2.hg19.interactome.altA.bed&quot;,</span><br><span class="line"> &quot;/storage11_7T/fuy/TADA-A/annotation/PPI/MPC2.hg19.interactome.altC.bed&quot;,</span><br><span class="line"> &quot;/storage11_7T/fuy/TADA-A/annotation/PPI/MPC2.hg19.interactome.altG.bed&quot;,</span><br><span class="line"> &quot;/storage11_7T/fuy/TADA-A/annotation/PPI/MPC2.hg19.interactome.altT.bed&quot;)    </span><br><span class="line">   </span><br><span class="line">),                    </span><br><span class="line"></span><br><span class="line">report_proportion = 1, #1000/17484,</span><br><span class="line">chunk = 1,</span><br><span class="line">node_n = 1,</span><br><span class="line"></span><br><span class="line">mutrate_ref_files = c(&quot;/storage11_7T/fuy/TADA-A/db/MS_data/mutrate/merge/supple3/plus_supple3_uq.17478gene.st.mg.A.bw&quot;,</span><br><span class="line">                     &quot;/storage11_7T/fuy/TADA-A/db/MS_data/mutrate/merge/supple3/plus_supple3_uq.17478gene.st.mg.C.bw&quot;,</span><br><span class="line">                     &quot;/storage11_7T/fuy/TADA-A/db/MS_data/mutrate/merge/supple3/plus_supple3_uq.17478gene.st.mg.G.bw&quot;,</span><br><span class="line">                     &quot;/storage11_7T/fuy/TADA-A/db/MS_data/mutrate/merge/supple3/plus_supple3_uq.17478gene.st.mg.T.bw&quot;)</span><br><span class="line"></span><br><span class="line">))</span><br><span class="line"></span><br><span class="line">saveRDS(compact_data,paste0(&quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/&quot;,Sys.Date(),&quot;_PPI_6788SNV_sf_uni_prior_compact.rds&quot;))</span><br></pre></td></tr></table></figure>

<pre><code>[1] &quot;Made a tmp folder for tmp files.&quot;
[1] &quot;Read in window files.&quot;
[1] &quot;Read in mutrate scaling file /storage11_7T/fuy/TADA-A/cell_WES/sf_1792_d_2058.1_cd_uniform_scaling_factors.txt&quot;
[1] &quot;Read in gene prior file.&quot;
[1] &quot;Chose the top 100% genes for parameter estimation.&quot;
[1] &quot;Patitioned genomic window to base-level coordinates at Round 1. Time consumed: 640.778s.&quot;
[1] &quot;Obtained un-calibrated base-level mutation rate for alt alleles. Time consumed: 2078.17.s&quot;
[1] &quot;Added allele specific annotations 1 .Time consumed: 1751.883s.&quot;
[1] &quot;Begin collapsing data...&quot;
[1] &quot;Read in mutation file /storage11_7T/fuy/TADA-A/cell_WES/DNM/affected/6788snv.affected.cd.auto.no_pli_rm.allele.bed.&quot;
[1] &quot;Collapsed data based on annotations for Study 1 .Time consumed: 1813.595.s&quot;
[1] &quot;Temp files cleaned and data recording finished!&quot;



     user    system   elapsed 
10148.569   275.272  6318.571 </code></pre><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">RR = TADA_A_RR_estimate(data = compact_data$base_info, selected_annotations = c(1), </span><br><span class="line"></span><br><span class="line">          gene_prior_file = &quot;/storage11_7T/fuy/TADA-A/db/MS_data/prior/new_uniform_gene_prior.txt&quot;,              </span><br><span class="line">                        optimization_iteration = 2000)$rr_report </span><br><span class="line">RR</span><br></pre></td></tr></table></figure>

<pre><code>[1] &quot;Read in gene prior file. Time consumed: 0.022s.&quot;
[1] &quot;Finished optimization. Time consumed: 46.834.s&quot;
[1] &quot;Got confidence intervals of RR estimates.&quot;
[1] &quot;Finished RR estimation!&quot;</code></pre><table>
<caption>A data.frame: 1 × 3</caption>
<thead>
    <tr><th scope="col">logRR</th><th scope="col">lower_bound</th><th scope="col">upper_bound</th></tr>
    <tr><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>2.939515</td><td>2.570341</td><td>3.308688</td></tr>
</tbody>
</table>




<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">df = data.frame(logRR=2.939515,lower_bound=2.570341,upper_bound=3.308688)</span><br><span class="line">saveRDS(df,&quot;/storage11_7T/fuy/TADA-A/annotation/PPI/PPI_rr.rds&quot;)</span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">options(scipen=200)</span><br><span class="line">tm = proc.time()</span><br><span class="line">BF = TADA_A_get_BFs(data = compact_data$base_info,</span><br><span class="line">               selected_annotations = c(1),</span><br><span class="line">               rr = RR$logRR , </span><br><span class="line">               additional_BF_file = c(&quot;/storage11_7T/fuy/TADA-A/annotation/cd_BF/uniform_cd_BFs.txt&quot;), </span><br><span class="line">#                additional_BF_file = c(&quot;/storage11_7T/fuy/TADA-A/annotation/cd_BF/cc_only_uniform_cd_BFs.txt&quot;),</span><br><span class="line">#                 additional_BF_file = c(&quot;/storage11_7T/fuy/TADA-A/annotation/cd_BF/17484_pp_all_BF.txt&quot;),</span><br><span class="line">               TADA_p0 = 0.94)</span><br><span class="line"></span><br><span class="line">g_BF = BF$gene_BF_table</span><br><span class="line">g_BF = g_BF[order(g_BF$FDR_all),]</span><br><span class="line">g_BF2 = g_BF[g_BF$FDR_all &lt;= 0.1,]</span><br><span class="line">proc.time() - tm</span><br><span class="line">nrow(g_BF2)</span><br></pre></td></tr></table></figure>

<pre><code>[1] &quot;Read in additional BF file /storage11_7T/fuy/TADA-A/annotation/cd_BF/uniform_cd_BFs.txt.&quot;
[1] &quot;Flagged genes that don&apos;t have any bases with any informative annotation.&quot;
[1] &quot;Got genenames without bases that have informative rr features.&quot;
[1] &quot;Got logBF_noncoding.&quot;
[1] &quot;Added coding and non-coding logBF. Time consumed: 2.865s.&quot;
[1] &quot;Finished!&quot;



   user  system elapsed 
  3.707   1.133   4.098 </code></pre><p>7</p>
<ul>
<li>2021/4/2<h3 id="PPI-only"><a href="#PPI-only" class="headerlink" title="PPI only"></a>PPI only</h3></li>
</ul>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">library(tadaA)</span><br><span class="line">setwd(&quot;/storage11_7T/fuy/TADA-A/annotation&quot;)</span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br></pre></td><td class="code"><pre><span class="line">system.time(compact_data &lt;- TADA_A_read_info_by_chunks(</span><br><span class="line">mut_files = &quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/affected/6788snv.affected.cd.auto.no_pli_rm.allele.bed&quot;,</span><br><span class="line">    </span><br><span class="line">window_file = &quot;/storage11_7T/fuy/TADA-A/db/MS_data/windows_partition/17478gene_hg19.87.auto.exon.wd.bed&quot; ,</span><br><span class="line">    </span><br><span class="line">mutrate_scaling_files = c(&quot;/storage11_7T/fuy/TADA-A/cell_WES/sf_1792_d_2058.1_cd_uniform_scaling_factors.txt&quot;),</span><br><span class="line"> </span><br><span class="line">sample_sizes = 6430, </span><br><span class="line">    </span><br><span class="line">gene_prior_file = &quot;/storage11_7T/fuy/TADA-A/db/MS_data/prior/new_uniform_gene_prior.txt&quot;,</span><br><span class="line">      </span><br><span class="line">nonAS_noncoding_annotations = c(&quot;/storage11_7T/fuy/TADA-A/annotation/PPI/hg19.mapped.bed&quot;) , </span><br><span class="line"></span><br><span class="line">AS_noncoding_annotations = NA,</span><br><span class="line"></span><br><span class="line">report_proportion = 1, #1000/17484,</span><br><span class="line">chunk = 1,</span><br><span class="line">node_n = 1,</span><br><span class="line"></span><br><span class="line">mutrate_ref_files = c(&quot;/storage11_7T/fuy/TADA-A/db/MS_data/mutrate/merge/supple3/plus_supple3_uq.17478gene.st.mg.A.bw&quot;,</span><br><span class="line">                     &quot;/storage11_7T/fuy/TADA-A/db/MS_data/mutrate/merge/supple3/plus_supple3_uq.17478gene.st.mg.C.bw&quot;,</span><br><span class="line">                     &quot;/storage11_7T/fuy/TADA-A/db/MS_data/mutrate/merge/supple3/plus_supple3_uq.17478gene.st.mg.G.bw&quot;,</span><br><span class="line">                     &quot;/storage11_7T/fuy/TADA-A/db/MS_data/mutrate/merge/supple3/plus_supple3_uq.17478gene.st.mg.T.bw&quot;)</span><br><span class="line"></span><br><span class="line">))</span><br><span class="line"></span><br><span class="line">saveRDS(compact_data,paste0(&quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/&quot;,Sys.Date(),&quot;_PPI_6788SNV_sf_uni_prior_compact.rds&quot;))</span><br></pre></td></tr></table></figure>

<pre><code>[1] &quot;Made a tmp folder for tmp files.&quot;
[1] &quot;Read in window files.&quot;
[1] &quot;Read in mutrate scaling file /storage11_7T/fuy/TADA-A/cell_WES/sf_1792_d_2058.1_cd_uniform_scaling_factors.txt&quot;
[1] &quot;Read in gene prior file.&quot;
[1] &quot;Chose the top 100% genes for parameter estimation.&quot;
[1] &quot;Patitioned genomic window to base-level coordinates at Round 1. Time consumed: 642.339s.&quot;
[1] &quot;Obtained un-calibrated base-level mutation rate for alt alleles. Time consumed: 2066.244.s&quot;
[1] &quot;Added non-allele specific annotations 1. Time consumed: 576.144s.&quot;
[1] &quot;Begin collapsing data...&quot;
[1] &quot;Read in mutation file /storage11_7T/fuy/TADA-A/cell_WES/DNM/affected/6788snv.affected.cd.auto.no_pli_rm.allele.bed.&quot;
[1] &quot;Collapsed data based on annotations for Study 1 .Time consumed: 1976.235.s&quot;
[1] &quot;Temp files cleaned and data recording finished!&quot;



    user   system  elapsed 
8261.069  225.750 5295.197 </code></pre><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">library(Rcpp)</span><br><span class="line">sourceCpp(file=&quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/ll_sum.cpp&quot;)</span><br><span class="line">sourceCpp(file=&quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/multi_annota.cpp&quot;)</span><br><span class="line">sourceCpp(file=&quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/post.cpp&quot;)</span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br></pre></td><td class="code"><pre><span class="line"># compact_data = readRDS(&quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/2021-01-19cp6_all_gene_PO_6788SNV_sf_uni_prior_compact.rds&quot;)</span><br><span class="line">data=compact_data$base_info</span><br><span class="line">selected_annotations=c(1)</span><br><span class="line">gene_prior_file = &quot;/storage11_7T/fuy/TADA-A/db/MS_data/prior/new_uniform_gene_prior.txt&quot;</span><br><span class="line">optimization_iteration = 2000</span><br><span class="line"></span><br><span class="line">gene_prior = fread(gene_prior_file)</span><br><span class="line">gene_prior = gene_prior[order(gene_prior$genename),]</span><br><span class="line"></span><br><span class="line">fr &lt;- function(x)&#123;</span><br><span class="line">all_rr = x</span><br><span class="line"></span><br><span class="line">logP_Zg0 = sumall0(data)</span><br><span class="line">logP_Zg1 = sumall1(data,selected_annotations,all_rr)</span><br><span class="line"></span><br><span class="line">logP_table&lt;-data.table(logP_Zg1 = logP_Zg1, logP_Zg0 = logP_Zg0, genename = names(data))</span><br><span class="line">logP_table &lt;- logP_table[gene_prior, on = &quot;genename&quot;]</span><br><span class="line">logP_table &lt;- logP_table[complete.cases(logP_table)]</span><br><span class="line">idx = match(unique(logP_table$genename),logP_table$genename)</span><br><span class="line">idx2 = c(idx,nrow(logP_table) + 1)</span><br><span class="line">pr = logP_table[idx,]$prior</span><br><span class="line">ll_sum1 &lt;- ll_sum(idx2,logP_table$logP_Zg1,logP_table$logP_Zg0,pr)</span><br><span class="line">ll_sum1 </span><br><span class="line">    &#125;</span><br><span class="line"></span><br><span class="line">mle &lt;- optim(rep(0.1, 1), fr, method = &quot;Brent&quot;, lower = -1, upper = 10,</span><br><span class="line">                 control=list(&quot;fnscale&quot;=-1, &quot;maxit&quot; = optimization_iteration), hessian = TRUE)</span><br><span class="line"></span><br><span class="line">fisher_info &lt;- solve(-mle$hessian)</span><br><span class="line">prop_sigma &lt;- sqrt(c(diag(fisher_info)))</span><br><span class="line">rr_estimate &lt;- c(mle$par)</span><br><span class="line">upper&lt;-rr_estimate+1.96*prop_sigma</span><br><span class="line">lower&lt;-rr_estimate-1.96*prop_sigma</span><br><span class="line"></span><br><span class="line">rr_report &lt;-data.frame(logRR = rr_estimate, lower_bound = lower, upper_bound = upper)</span><br><span class="line">rr_report</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.frame: 1 × 3</caption>
<thead>
    <tr><th scope="col">logRR</th><th scope="col">lower_bound</th><th scope="col">upper_bound</th></tr>
    <tr><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>0.01709766</td><td>0.01149644</td><td>0.02269889</td></tr>
</tbody>
</table>




<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">rr_report$annota = &quot;PPI_only&quot;</span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">saveRDS(rr_report,&quot;/storage11_7T/fuy/TADA-A/annotation/PPI/PPI_only.rds&quot;)</span><br></pre></td></tr></table></figure>

<h3 id="PPH2-PPI-PPH2"><a href="#PPH2-PPI-PPH2" class="headerlink" title="PPH2_PPI-PPH2"></a>PPH2_PPI-PPH2</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br></pre></td><td class="code"><pre><span class="line">system.time(compact_data &lt;- TADA_A_read_info_by_chunks(</span><br><span class="line">mut_files = &quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/affected/6788snv.affected.cd.auto.no_pli_rm.allele.bed&quot;,</span><br><span class="line">    </span><br><span class="line">window_file = &quot;/storage11_7T/fuy/TADA-A/db/MS_data/windows_partition/17478gene_hg19.87.auto.exon.wd.bed&quot; ,</span><br><span class="line">    </span><br><span class="line">mutrate_scaling_files = c(&quot;/storage11_7T/fuy/TADA-A/cell_WES/sf_1792_d_2058.1_cd_uniform_scaling_factors.txt&quot;),</span><br><span class="line"> </span><br><span class="line">sample_sizes = 6430, </span><br><span class="line">    </span><br><span class="line">gene_prior_file = &quot;/storage11_7T/fuy/TADA-A/db/MS_data/prior/new_uniform_gene_prior.txt&quot;,</span><br><span class="line">      </span><br><span class="line">nonAS_noncoding_annotations = NA , </span><br><span class="line"></span><br><span class="line">AS_noncoding_annotations = list(</span><br><span class="line">c(</span><br><span class="line">&quot;/storage11_7T/fuy/TADA-A/annotation/dbNSFP/wd/uq.wd.PPH2_D.HDIV.A.bed&quot;,</span><br><span class="line">&quot;/storage11_7T/fuy/TADA-A/annotation/dbNSFP/wd/uq.wd.PPH2_D.HDIV.C.bed&quot;,</span><br><span class="line">&quot;/storage11_7T/fuy/TADA-A/annotation/dbNSFP/wd/uq.wd.PPH2_D.HDIV.G.bed&quot;,</span><br><span class="line">&quot;/storage11_7T/fuy/TADA-A/annotation/dbNSFP/wd/uq.wd.PPH2_D.HDIV.T.bed&quot; ) ,</span><br><span class="line"></span><br><span class="line">c(</span><br><span class="line">&quot;/storage11_7T/fuy/TADA-A/annotation/dbNSFP/wd/uq.wd.PPH2_D.HVAR.A.bed&quot;,</span><br><span class="line">&quot;/storage11_7T/fuy/TADA-A/annotation/dbNSFP/wd/uq.wd.PPH2_D.HVAR.C.bed&quot;,</span><br><span class="line">&quot;/storage11_7T/fuy/TADA-A/annotation/dbNSFP/wd/uq.wd.PPH2_D.HVAR.G.bed&quot;,</span><br><span class="line">&quot;/storage11_7T/fuy/TADA-A/annotation/dbNSFP/wd/uq.wd.PPH2_D.HVAR.T.bed&quot; ) ,</span><br><span class="line"></span><br><span class="line">c(</span><br><span class="line">&quot;/storage11_7T/fuy/TADA-A/annotation/PPI/PPH2-PPI/wd/uq.wd.PPI.PPH2_D.HDIV.A.bed&quot;,</span><br><span class="line">&quot;/storage11_7T/fuy/TADA-A/annotation/PPI/PPH2-PPI/wd/uq.wd.PPI.PPH2_D.HDIV.C.bed&quot;,</span><br><span class="line">&quot;/storage11_7T/fuy/TADA-A/annotation/PPI/PPH2-PPI/wd/uq.wd.PPI.PPH2_D.HDIV.G.bed&quot;,</span><br><span class="line">&quot;/storage11_7T/fuy/TADA-A/annotation/PPI/PPH2-PPI/wd/uq.wd.PPI.PPH2_D.HDIV.T.bed&quot;) ,</span><br><span class="line">    </span><br><span class="line">c(</span><br><span class="line">&quot;/storage11_7T/fuy/TADA-A/annotation/PPI/PPH2-PPI/wd/uq.wd.PPI.PPH2_D.HVAR.A.bed&quot;,</span><br><span class="line">&quot;/storage11_7T/fuy/TADA-A/annotation/PPI/PPH2-PPI/wd/uq.wd.PPI.PPH2_D.HVAR.C.bed&quot;,</span><br><span class="line">&quot;/storage11_7T/fuy/TADA-A/annotation/PPI/PPH2-PPI/wd/uq.wd.PPI.PPH2_D.HVAR.G.bed&quot;,</span><br><span class="line">&quot;/storage11_7T/fuy/TADA-A/annotation/PPI/PPH2-PPI/wd/uq.wd.PPI.PPH2_D.HVAR.T.bed&quot;) </span><br><span class="line">            </span><br><span class="line">    </span><br><span class="line">),</span><br><span class="line"></span><br><span class="line">report_proportion = 1, #1000/17484,</span><br><span class="line">chunk = 20,</span><br><span class="line">node_n = 2,</span><br><span class="line"></span><br><span class="line">mutrate_ref_files = c(&quot;/storage11_7T/fuy/TADA-A/db/MS_data/mutrate/merge/supple3/plus_supple3_uq.17478gene.st.mg.A.bw&quot;,</span><br><span class="line">                     &quot;/storage11_7T/fuy/TADA-A/db/MS_data/mutrate/merge/supple3/plus_supple3_uq.17478gene.st.mg.C.bw&quot;,</span><br><span class="line">                     &quot;/storage11_7T/fuy/TADA-A/db/MS_data/mutrate/merge/supple3/plus_supple3_uq.17478gene.st.mg.G.bw&quot;,</span><br><span class="line">                     &quot;/storage11_7T/fuy/TADA-A/db/MS_data/mutrate/merge/supple3/plus_supple3_uq.17478gene.st.mg.T.bw&quot;)</span><br><span class="line"></span><br><span class="line">))</span><br><span class="line"></span><br><span class="line">saveRDS(compact_data,paste0(&quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/&quot;,Sys.Date(),&quot;_PPH2_PPI-PPH2_6788SNV_sf_uni_prior_compact.rds&quot;))</span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">library(Rcpp)</span><br><span class="line">sourceCpp(file=&quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/ll_sum.cpp&quot;)</span><br><span class="line">sourceCpp(file=&quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/multi_annota.cpp&quot;)</span><br><span class="line">sourceCpp(file=&quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/post.cpp&quot;)</span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br></pre></td><td class="code"><pre><span class="line"># compact_data = readRDS(&quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/2021-01-19cp6_all_gene_PO_6788SNV_sf_uni_prior_compact.rds&quot;)</span><br><span class="line">data=compact_data$base_info</span><br><span class="line">selected_annotations=c(4)</span><br><span class="line">gene_prior_file = &quot;/storage11_7T/fuy/TADA-A/db/MS_data/prior/new_uniform_gene_prior.txt&quot;</span><br><span class="line">optimization_iteration = 2000</span><br><span class="line"></span><br><span class="line">gene_prior = fread(gene_prior_file)</span><br><span class="line">gene_prior = gene_prior[order(gene_prior$genename),]</span><br><span class="line"></span><br><span class="line">fr &lt;- function(x)&#123;</span><br><span class="line">all_rr = x</span><br><span class="line"></span><br><span class="line">logP_Zg0 = sumall0(data)</span><br><span class="line">logP_Zg1 = sumall1(data,selected_annotations,all_rr)</span><br><span class="line"></span><br><span class="line">logP_table&lt;-data.table(logP_Zg1 = logP_Zg1, logP_Zg0 = logP_Zg0, genename = names(data))</span><br><span class="line">logP_table &lt;- logP_table[gene_prior, on = &quot;genename&quot;]</span><br><span class="line">logP_table &lt;- logP_table[complete.cases(logP_table)]</span><br><span class="line">idx = match(unique(logP_table$genename),logP_table$genename)</span><br><span class="line">idx2 = c(idx,nrow(logP_table) + 1)</span><br><span class="line">pr = logP_table[idx,]$prior</span><br><span class="line">ll_sum1 &lt;- ll_sum(idx2,logP_table$logP_Zg1,logP_table$logP_Zg0,pr)</span><br><span class="line">ll_sum1 </span><br><span class="line">    &#125;</span><br><span class="line"></span><br><span class="line">mle &lt;- optim(rep(0.1, 1), fr, method = &quot;Brent&quot;, lower = -1, upper = 10,</span><br><span class="line">                 control=list(&quot;fnscale&quot;=-1, &quot;maxit&quot; = optimization_iteration), hessian = TRUE)</span><br><span class="line"></span><br><span class="line">fisher_info &lt;- solve(-mle$hessian)</span><br><span class="line">prop_sigma &lt;- sqrt(c(diag(fisher_info)))</span><br><span class="line">rr_estimate &lt;- c(mle$par)</span><br><span class="line">upper&lt;-rr_estimate+1.96*prop_sigma</span><br><span class="line">lower&lt;-rr_estimate-1.96*prop_sigma</span><br><span class="line"></span><br><span class="line">rr_report &lt;-data.frame(logRR = rr_estimate, lower_bound = lower, upper_bound = upper)</span><br><span class="line">rr_report</span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">rr_report$annota = &quot;PPI.PPH2_D.HVAR&quot;</span><br><span class="line">aaaa = rbind(aaa,rr_report)</span><br><span class="line">aaaa</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.frame: 4 × 4</caption>
<thead>
    <tr><th scope="col">logRR</th><th scope="col">lower_bound</th><th scope="col">upper_bound</th><th scope="col">annota</th></tr>
    <tr><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;chr&gt;</th></tr>
</thead>
<tbody>
    <tr><td>1.769077</td><td>1.642946</td><td>1.895207</td><td>PPH2_D.HDIV    </td></tr>
    <tr><td>1.902419</td><td>1.769754</td><td>2.035085</td><td>PPH2_D.HVAR    </td></tr>
    <tr><td>2.551475</td><td>2.345328</td><td>2.757622</td><td>PPI.PPH2_D.HDIV</td></tr>
    <tr><td>2.600493</td><td>2.372915</td><td>2.828070</td><td>PPI.PPH2_D.HVAR</td></tr>
</tbody>
</table>




<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">saveRDS(aaaa,&quot;/storage11_7T/fuy/TADA-A/annotation/PPI/2021-04-07_PPH2_PPI-PPH2_sep_RR.rds&quot;)</span><br></pre></td></tr></table></figure>
</div><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">Author: </span><span class="post-copyright-info"><a href="mailto:undefined" target="_blank" rel="noopener"></a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">Link: </span><span class="post-copyright-info"><a href="https://yfu1116.github.io/project/2021-04-07-PPI-annotation/">https://yfu1116.github.io/project/2021-04-07-PPI-annotation/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">Copyright Notice: </span><span class="post-copyright-info">All articles in this blog are licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank" rel="noopener">CC BY-NC-SA 4.0</a> unless stating additionally.</span></div></div><div class="tag_share"><div class="post-meta__tag-list"></div><div class="post_share"><div class="social-share" data-image="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" data-sites="github,email,facebook,twitter,weixin,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js/dist/css/share.min.css"><script src="https://cdn.jsdelivr.net/npm/social-share.js/dist/js/social-share.min.js" defer></script></div></div><nav class="pagination-post" id="pagination"><div class="prev-post pull-left"><a href="/project/2021-04-07-optim-multi-scalar-pi-gama/"><img class="prev-cover" data-lazy-src="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" onerror="onerror=null;src='/project/img/404.jpg'"><div class="pagination-info"><div class="label">Previous Post</div><div class="prev_info">2021-04-07-optim-multi-scalar-pi-gama</div></div></a></div><div class="next-post pull-right"><a href="/project/2021-03-26-multi-annota-RR-optim-test/"><img class="next-cover" data-lazy-src="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" onerror="onerror=null;src='/project/img/404.jpg'"><div class="pagination-info"><div class="label">Next Post</div><div class="next_info">2021-03-26-multi-annota-RR-optim-test</div></div></a></div></nav></article></main><footer id="footer" data-type="color"><div id="footer-wrap"><div class="copyright">&copy;2020 - 2021 By </div><div class="framework-info"><span>Framework </span><a href="https://hexo.io" target="_blank" rel="noopener">Hexo</a><span class="footer-separator">|</span><span>Theme </span><a href="https://github.com/jerryc127/hexo-theme-butterfly" target="_blank" rel="noopener">Butterfly</a></div></div></footer></div><section id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="Read Mode"><i class="fas fa-book-open"></i></button><button id="font_plus" type="button" title="Increase Font Size"><i class="fas fa-plus"></i></button><button id="font_minus" type="button" title="Decrease Font Size"><i class="fas fa-minus"></i></button><button id="darkmode" type="button" title="Switch Between Light And Dark Mode"><i class="fas fa-adjust"></i></button></div><div id="rightside-config-show"><button id="rightside_config" type="button" title="Setting"><i class="fas fa-cog"></i></button><button class="close" id="mobile-toc-button" type="button" title="Table Of Contents"><i class="fas fa-list-ul"></i></button><button id="go-up" type="button" title="Back To Top"><i class="fas fa-arrow-up"></i></button></div></section><div><script src="https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js"></script><script src="/project/js/utils.js"></script><script src="/project/js/main.js"></script><script src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.js"></script><script src="https://cdn.jsdelivr.net/npm/instant.page/instantpage.min.js" type="module" defer></script><script src="https://cdn.jsdelivr.net/npm/vanilla-lazyload/dist/lazyload.iife.min.js" async></script><div class="js-pjax"><script>if (!window.MathJax) {
  window.MathJax = {
    loader: {
      source: {
        '[tex]/amsCd': '[tex]/amscd'
      }
    },
    tex: {
      inlineMath: [ ['$','$'], ["\\(","\\)"]],
      tags: 'ams'
    },
    options: {
      renderActions: {
        findScript: [10, doc => {
          for (const node of document.querySelectorAll('script[type^="math/tex"]')) {
            const display = !!node.type.match(/; *mode=display/)
            const math = new doc.options.MathItem(node.textContent, doc.inputJax[0], display)
            const text = document.createTextNode('')
            node.parentNode.replaceChild(text, node)
            math.start = {node: text, delim: '', n: 0}
            math.end = {node: text, delim: '', n: 0}
            doc.math.push(math)
          }
        }, '']
      }
    }
  }
  
  var script = document.createElement('script')
  script.src = 'https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js'
  script.id = 'MathJax-script'
  script.async = true
  document.head.appendChild(script)
} else {
  MathJax.startup.document.state(0)
  MathJax.texReset()
  MathJax.typeset()
}</script><script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div></div></body></html>