<!DOCTYPE html><html lang="en" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1"><title>0923_TADA_cell_DNM_SNV_RADAR_RBP | yang's blog</title><meta name="description" content="1234567setwd(&quot;&#x2F;storage11_7T&#x2F;fuy&#x2F;TADA-A&quot;)library(data.table)f = fread(&quot;cell_DNM_allele.bed&quot;)f$l1 = nchar(f$V4)f$l2 = nchar(f$V5)f2 = f[f$l1 == 1 &amp; f$l2 ==1,]head(f2)    A data.t"><meta name="author" content=""><meta name="copyright" content=""><meta name="format-detection" content="telephone=no"><link rel="shortcut icon" href="/project/img/favicon.png"><link rel="canonical" href="https://yfu1116.github.io/project/0923_TADA_cell_DNM_SNV_RADAR_RBP/"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//fonts.googleapis.com" crossorigin="crossorigin"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><meta property="og:type" content="article"><meta property="og:title" content="0923_TADA_cell_DNM_SNV_RADAR_RBP"><meta property="og:url" content="https://yfu1116.github.io/project/0923_TADA_cell_DNM_SNV_RADAR_RBP/"><meta property="og:site_name" content="yang's blog"><meta property="og:description" content="1234567setwd(&quot;&#x2F;storage11_7T&#x2F;fuy&#x2F;TADA-A&quot;)library(data.table)f = fread(&quot;cell_DNM_allele.bed&quot;)f$l1 = nchar(f$V4)f$l2 = nchar(f$V5)f2 = f[f$l1 == 1 &amp; f$l2 ==1,]head(f2)    A data.t"><meta property="og:image" content="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg"><meta property="article:published_time" content="2020-09-23T02:33:14.946Z"><meta property="article:modified_time" content="2020-09-23T02:33:44.827Z"><meta name="twitter:card" content="summary"><script>var activateDarkMode = function () {
  document.documentElement.setAttribute('data-theme', 'dark')
  if (document.querySelector('meta[name="theme-color"]') !== null) {
    document.querySelector('meta[name="theme-color"]').setAttribute('content', '#000')
  }
}
var activateLightMode = function () {
  document.documentElement.setAttribute('data-theme', 'light')
  if (document.querySelector('meta[name="theme-color"]') !== null) {
    document.querySelector('meta[name="theme-color"]').setAttribute('content', '#fff')
  }
}

var getCookies = function (name) {
  const value = `; ${document.cookie}`
  const parts = value.split(`; ${name}=`)
  if (parts.length === 2) return parts.pop().split(';').shift()
}

var autoChangeMode = 'false'
var t = getCookies('theme')
if (autoChangeMode === '1') {
  var isDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches
  var isLightMode = window.matchMedia('(prefers-color-scheme: light)').matches
  var isNotSpecified = window.matchMedia('(prefers-color-scheme: no-preference)').matches
  var hasNoSupport = !isDarkMode && !isLightMode && !isNotSpecified

  if (t === undefined) {
    if (isLightMode) activateLightMode()
    else if (isDarkMode) activateDarkMode()
    else if (isNotSpecified || hasNoSupport) {
      console.log('You specified no preference for a color scheme or your browser does not support it. I Schedule dark mode during night time.')
      var now = new Date()
      var hour = now.getHours()
      var isNight = hour <= 6 || hour >= 18
      isNight ? activateDarkMode() : activateLightMode()
    }
    window.matchMedia('(prefers-color-scheme: dark)').addListener(function (e) {
      if (Cookies.get('theme') === undefined) {
        e.matches ? activateDarkMode() : activateLightMode()
      }
    })
  } else if (t === 'light') activateLightMode()
  else activateDarkMode()
} else if (autoChangeMode === '2') {
  now = new Date()
  hour = now.getHours()
  isNight = hour <= 6 || hour >= 18
  if (t === undefined) isNight ? activateDarkMode() : activateLightMode()
  else if (t === 'light') activateLightMode()
  else activateDarkMode()
} else {
  if (t === 'dark') activateDarkMode()
  else if (t === 'light') activateLightMode()
}</script><link rel="stylesheet" href="/project/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.css"><link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Titillium+Web&amp;display=swap"><script>var GLOBAL_CONFIG = { 
  root: '/project/',
  hexoversion: '4.0.0',
  algolia: undefined,
  localSearch: undefined,
  translate: undefined,
  noticeOutdate: undefined,
  copy: {
    success: 'Copy successfully',
    error: 'Copy error',
    noSupport: 'The browser does not support'
  },
  bookmark: {
    message_prev: 'Press',
    message_next: 'to bookmark this page'
  },
  runtime_unit: 'days',
  runtime: false,
  copyright: undefined,
  ClickShowText: undefined,
  medium_zoom: false,
  fancybox: true,
  Snackbar: undefined,
  justifiedGallery: {
    js: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/js/jquery.justifiedGallery.min.js',
    css: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/css/justifiedGallery.min.css'
  },
  baiduPush: false,
  highlightCopy: true,
  highlightLang: true,
  isPhotoFigcaption: false,
  islazyload: true,
  isanchor: false    
}</script><script id="config_change">var GLOBAL_CONFIG_SITE = { 
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isSidebar: false,
  postUpdate: '2020-09-23 10:33:44'
}</script><noscript><style>
#nav {
  opacity: 1
}
.justified-gallery img {
  opacity: 1
}
</style></noscript></head><body><div id="mobile-sidebar"><div id="menu_mask"></div><div id="mobile-sidebar-menus"><div class="mobile_author_icon"><img class="avatar-img" data-lazy-src="/project/img/avatar.png" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="mobile_post_data"><div class="mobile_data_item is-center"><div class="mobile_data_link"><a href="/project/archives/"><div class="headline">Articles</div><div class="length_num">67</div></a></div></div></div><hr/></div></div><div id="body-wrap"><header class="post-bg" id="page-header" style="background-image: url(https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg)"><nav id="nav"><span class="pull-left" id="blog_name"><a class="blog_title" id="site-name" href="/project/">yang's blog</a></span><span class="pull-right menus"><span class="toggle-menu close"><a class="site-page"><i class="fas fa-bars fa-fw"></i></a></span></span></nav><div id="post-info"><div id="post-title"><div class="posttitle">0923_TADA_cell_DNM_SNV_RADAR_RBP</div></div><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">Created</span><time class="post-meta-date-created" datetime="2020-09-23T02:33:14.946Z" title="Created 2020-09-23 10:33:14">2020-09-23</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">Updated</span><time class="post-meta-date-updated" datetime="2020-09-23T02:33:44.827Z" title="Updated 2020-09-23 10:33:44">2020-09-23</time></span></div><div class="meta-secondline"> <span class="post-meta-separator">|</span><span class="post-meta-pv-cv"><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">Post View:</span><span id="busuanzi_value_page_pv"></span></span></div></div></div></header><main class="layout_post" id="content-inner"><article id="post"><div class="post-content" id="article-container"><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">setwd(&quot;/storage11_7T/fuy/TADA-A&quot;)</span><br><span class="line">library(data.table)</span><br><span class="line">f = fread(&quot;cell_DNM_allele.bed&quot;)</span><br><span class="line">f$l1 = nchar(f$V4)</span><br><span class="line">f$l2 = nchar(f$V5)</span><br><span class="line">f2 = f[f$l1 == 1 &amp; f$l2 ==1,]</span><br><span class="line">head(f2)</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.table: 6 × 7</caption>
<thead>
    <tr><th scope="col">V1</th><th scope="col">V2</th><th scope="col">V3</th><th scope="col">V4</th><th scope="col">V5</th><th scope="col">l1</th><th scope="col">l2</th></tr>
    <tr><th scope="col">&lt;chr&gt;</th><th scope="col">&lt;int&gt;</th><th scope="col">&lt;int&gt;</th><th scope="col">&lt;chr&gt;</th><th scope="col">&lt;chr&gt;</th><th scope="col">&lt;int&gt;</th><th scope="col">&lt;int&gt;</th></tr>
</thead>
<tbody>
    <tr><td>chr1 </td><td> 94049573</td><td> 94049574</td><td>C</td><td>A</td><td>1</td><td>1</td></tr>
    <tr><td>chr10</td><td>114910882</td><td>114910883</td><td>G</td><td>A</td><td>1</td><td>1</td></tr>
    <tr><td>chr9 </td><td> 14150142</td><td> 14150143</td><td>C</td><td>T</td><td>1</td><td>1</td></tr>
    <tr><td>chr11</td><td> 66254813</td><td> 66254814</td><td>G</td><td>A</td><td>1</td><td>1</td></tr>
    <tr><td>chr10</td><td>114901075</td><td>114901076</td><td>G</td><td>A</td><td>1</td><td>1</td></tr>
    <tr><td>chr14</td><td> 93286181</td><td> 93286182</td><td>G</td><td>C</td><td>1</td><td>1</td></tr>
</tbody>
</table>




<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">fwrite(f2,&quot;cell_DNM_SNV_allele.bed&quot;,sep=&quot;\t&quot;,col.names=F)</span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">library(tadaA)</span><br><span class="line">setwd(&quot;/storage11_7T/fuy/TADA-A&quot;)</span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">TADA_A_adjust_mutrate(window_file = &apos;db/MS_data/windows_partition/cd_windows_with_div_score.bed&apos;,</span><br><span class="line">                      mut_file = &apos;cell_DNM_SNV.bed&apos;,</span><br><span class="line">                      genes = &quot;all&quot;,</span><br><span class="line">                      sample_size = 8609,</span><br><span class="line">                      mutrate_mode = &quot;regular&quot;,</span><br><span class="line">                      scaling_file_name = &quot;cell_DNM_SNV_scaling_factors.txt&quot;,</span><br><span class="line">                      scale_features = c(&quot;GC_content&quot;, &quot;div_score&quot;))</span><br></pre></td></tr></table></figure>

<pre><code>[1] &quot;Made a tmp folder for tmp files.&quot;
[1] &quot;Finished calculating mutation count within per genomic window. Time consumed: 2.897s&quot;
[1] &quot;Finished fitting mutation rate model and calculating scaling factors. Time consumed: 6.371s&quot;
[1] &quot;Removed temporary files.&quot;</code></pre><table>
<caption>A matrix: 3 × 4 of type dbl</caption>
<thead>
    <tr><th></th><th scope="col">Estimate</th><th scope="col">Std. Error</th><th scope="col">z value</th><th scope="col">Pr(&gt;|z|)</th></tr>
</thead>
<tbody>
    <tr><th scope="row">(Intercept)</th><td> 0.04822686</td><td>0.01119170</td><td>  4.309163</td><td>1.638737e-05</td></tr>
    <tr><th scope="row">GC_content</th><td>-0.17551081</td><td>0.01178763</td><td>-14.889404</td><td>3.861996e-50</td></tr>
    <tr><th scope="row">div_score</th><td> 0.04143530</td><td>0.01074956</td><td>  3.854604</td><td>1.159173e-04</td></tr>
</tbody>
</table>




<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br></pre></td><td class="code"><pre><span class="line">system.time(compact_data &lt;- TADA_A_read_info_by_chunks(</span><br><span class="line">mut_files = &quot;/storage11_7T/fuy/TADA-A/cell_DNM_SNV_allele.bed&quot;,</span><br><span class="line">    </span><br><span class="line">window_file =  &apos;db/MS_data/windows_partition/cd_windows_with_div_score.bed&apos;,</span><br><span class="line">    </span><br><span class="line">mutrate_scaling_files = c(&quot;/storage11_7T/fuy/TADA-A/cell_DNM_SNV_scaling_factors.txt&quot;),  </span><br><span class="line">    </span><br><span class="line">sample_sizes = 8609,</span><br><span class="line">    </span><br><span class="line">gene_prior_file = &quot;db/MS_data/prior/uniform_gene_prior.txt&quot;,</span><br><span class="line">    </span><br><span class="line">nonAS_noncoding_annotations =  NA,#c( </span><br><span class="line">#                                  &quot;/storage11_7T/fuy/TADA-A/annotation/ccr/ccrs.allchrom.gt90.bed&quot;),</span><br><span class="line">                                #&quot;/storage11_7T/fuy/TADA-A/annotation/ccr/chr_ccr_gt95_syn_rm.bed&quot;),</span><br><span class="line"></span><br><span class="line">AS_noncoding_annotations = list(</span><br><span class="line">####################### RADAR RBP ############################      </span><br><span class="line">c(&quot;/storage11_7T/fuy/TADA-A/annotation/RADAR_RBP/A.top005.bed&quot;,</span><br><span class="line"> &quot;/storage11_7T/fuy/TADA-A/annotation/RADAR_RBP/C.top005.bed&quot;,</span><br><span class="line"> &quot;/storage11_7T/fuy/TADA-A/annotation/RADAR_RBP/G.top005.bed&quot;,</span><br><span class="line"> &quot;/storage11_7T/fuy/TADA-A/annotation/RADAR_RBP/T.top005.bed&quot;)    </span><br><span class="line">    </span><br><span class="line">####################### WES denovo ptv ############################    </span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/pLI/denovo_pLI_snv_alt_A.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/denovo_pLI_snv_alt_C.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/denovo_pLI_snv_alt_G.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/denovo_pLI_snv_alt_A.bed&quot;),</span><br><span class="line">    </span><br><span class="line">####################### gnomad lof (PTV SNV: Nonsense, splice acceptor, and splice donor variants) ############################    </span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/gnomad/gnomad.v2.1.1.all_lofs_snv_alt_A.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/gnomad/gnomad.v2.1.1.all_lofs_snv_alt_C.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/gnomad/gnomad.v2.1.1.all_lofs_snv_alt_G.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/gnomad/gnomad.v2.1.1.all_lofs_snv_alt_T.bed&quot;), </span><br><span class="line">    </span><br><span class="line">####################### gnomad PTV tiers ############################</span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/pLI/pLI_nlt09_gnomad.v2.1.1.all_lofs_snv_alt_A.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/pLI_nlt09_gnomad.v2.1.1.all_lofs_snv_alt_C.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/pLI_nlt09_gnomad.v2.1.1.all_lofs_snv_alt_G.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/pLI_nlt09_gnomad.v2.1.1.all_lofs_snv_alt_T.bed&quot;)</span><br><span class="line">    </span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/pLI/pLI_05_995_gnomad.v2.1.1.all_lofs_snv_alt_A.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/pLI_05_995_gnomad.v2.1.1.all_lofs_snv_alt_C.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/pLI_05_995_gnomad.v2.1.1.all_lofs_snv_alt_G.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/pLI_05_995_gnomad.v2.1.1.all_lofs_snv_alt_T.bed&quot;),</span><br><span class="line">    </span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/pLI/pLI_lt05_gnomad.v2.1.1.all_lofs_snv_alt_A.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/pLI_lt05_gnomad.v2.1.1.all_lofs_snv_alt_C.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/pLI_lt05_gnomad.v2.1.1.all_lofs_snv_alt_G.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/pLI_lt05_gnomad.v2.1.1.all_lofs_snv_alt_T.bed&quot;),</span><br><span class="line">    </span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/pLI/pLI_nlt_995_gnomad.v2.1.1.all_lofs_snv_alt_A.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/pLI_nlt_995_gnomad.v2.1.1.all_lofs_snv_alt_C.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/pLI_nlt_995_gnomad.v2.1.1.all_lofs_snv_alt_G.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/pLI_nlt_995_gnomad.v2.1.1.all_lofs_snv_alt_T.bed&quot;),</span><br><span class="line">    </span><br><span class="line">####################### annova missense ############################</span><br><span class="line">#                       c(&quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/all_missense_alt_A.bed&quot;, </span><br><span class="line">#                           &quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/all_missense_alt_C.bed&quot;, </span><br><span class="line">#                           &quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/all_missense_alt_G.bed&quot;,</span><br><span class="line">#                           &quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/all_missense_alt_T.bed&quot;),</span><br><span class="line"></span><br><span class="line">####################### annova lof ############################</span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/all_lof_alt_A.bed&quot;, </span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/all_lof_alt_C.bed&quot;, </span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/all_lof_alt_G.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/all_lof_alt_T.bed&quot;),</span><br><span class="line"></span><br><span class="line">####################### annova syn ############################</span><br><span class="line"># #                               c(&quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/all_cd_syn_alt_A.bed&quot;,</span><br><span class="line"># #                                 &quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/all_cd_syn_alt_C.bed&quot;,</span><br><span class="line"># #                                 &quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/all_cd_syn_alt_G.bed&quot;,</span><br><span class="line"># #                                 &quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/all_cd_syn_alt_T.bed&quot;),</span><br><span class="line">    </span><br><span class="line">####################### CADD ############################</span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/CADD/whole_genome_SNVs_gt15_altA_within_10kb_and_promoter_no_utr.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/CADD/whole_genome_SNVs_gt15_altC_within_10kb_and_promoter_no_utr.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/CADD/whole_genome_SNVs_gt15_altG_within_10kb_and_promoter_no_utr.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/CADD/whole_genome_SNVs_gt15_altT_within_10kb_and_promoter_no_utr.bed&quot;),    </span><br><span class="line"></span><br><span class="line">####################### RBP ############################</span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/RBP-VarDB/RBP.all.bed.merge_overlap_hg19_refGenes_exons.gtf.lg.transc.fa.RNAsnpM3.bed.abspos.p0.1.altA.bed.merge_in_coding_windows.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/RBP-VarDB/RBP.all.bed.merge_overlap_hg19_refGenes_exons.gtf.lg.transc.fa.RNAsnpM3.bed.abspos.p0.1.altC.bed.merge_in_coding_windows.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/RBP-VarDB/RBP.all.bed.merge_overlap_hg19_refGenes_exons.gtf.lg.transc.fa.RNAsnpM3.bed.abspos.p0.1.altG.bed.merge_in_coding_windows.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/RBP-VarDB/RBP.all.bed.merge_overlap_hg19_refGenes_exons.gtf.lg.transc.fa.RNAsnpM3.bed.abspos.p0.1.altT.bed.merge_in_coding_windows.bed&quot;),</span><br><span class="line"></span><br><span class="line">###################### MVP ###########################</span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/MVP/chr_MVP_all_rare_missense_pathogen_rank_gt_75_alt_A.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/MVP/chr_MVP_all_rare_missense_pathogen_rank_gt_75_alt_C.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/MVP/chr_MVP_all_rare_missense_pathogen_rank_gt_75_alt_G.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/MVP/chr_MVP_all_rare_missense_pathogen_rank_gt_75_alt_T.bed&quot;),</span><br><span class="line">    </span><br><span class="line">############################ primateAI ################################</span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/primateAI/chr_primateAI_exome_mutation_pathogen_rank_gt_80_alt_A.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/primateAI/chr_primateAI_exome_mutation_pathogen_rank_gt_80_alt_C.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/primateAI/chr_primateAI_exome_mutation_pathogen_rank_gt_80_alt_G.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/primateAI/chr_primateAI_exome_mutation_pathogen_rank_gt_80_alt_T.bed&quot;),</span><br><span class="line"></span><br><span class="line">########################## spidex #############################</span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/db/MS_data/annota/spidex_public_noncommercial_v1_0.tab_alt_A_lower10pct.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/db/MS_data/annota/spidex_public_noncommercial_v1_0.tab_alt_C_lower10pct.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/db/MS_data/annota/spidex_public_noncommercial_v1_0.tab_alt_G_lower10pct.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/db/MS_data/annota/spidex_public_noncommercial_v1_0.tab_alt_T_lower10pct.bed&quot;),</span><br><span class="line"></span><br><span class="line">############## MPC ##################</span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/fordist_constraint_official_mpc_values_v2_MPC_gt2_altA.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/fordist_constraint_official_mpc_values_v2_MPC_gt2_altC.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/fordist_constraint_official_mpc_values_v2_MPC_gt2_altG.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/fordist_constraint_official_mpc_values_v2_MPC_gt2_altT.bed&quot;),</span><br><span class="line">    </span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/chr.MPC12_alt_A.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/chr.MPC12_alt_C.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/chr.MPC12_alt_G.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/chr.MPC12_alt_T.bed&quot;)</span><br><span class="line">    </span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/chr.MPC01_alt_A.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/chr.MPC01_alt_C.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/chr.MPC01_alt_G.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/chr.MPC01_alt_T.bed&quot;)</span><br><span class="line">    </span><br><span class="line">#################### ribosnitch ###########################</span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/ribosnitch/hg19_refGenes_exons.gtf.lg.transc.fa.RNAsnpM3.bed.abspos.p0.1.altA.bed.merge.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/ribosnitch/hg19_refGenes_exons.gtf.lg.transc.fa.RNAsnpM3.bed.abspos.p0.1.altC.bed.merge.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/ribosnitch/hg19_refGenes_exons.gtf.lg.transc.fa.RNAsnpM3.bed.abspos.p0.1.altG.bed.merge.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/ribosnitch/hg19_refGenes_exons.gtf.lg.transc.fa.RNAsnpM3.bed.abspos.p0.1.altT.bed.merge.bed&quot;),</span><br><span class="line">    </span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/ribosnitch/hg19_refGenes_exons.gtf.lg.transc.fa.RNAsnpM3.bed.abspos.p0.05.merged.altA_in_coding_windows.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/ribosnitch/hg19_refGenes_exons.gtf.lg.transc.fa.RNAsnpM3.bed.abspos.p0.05.merged.altC_in_coding_windows.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/ribosnitch/hg19_refGenes_exons.gtf.lg.transc.fa.RNAsnpM3.bed.abspos.p0.05.merged.altG_in_coding_windows.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/ribosnitch/hg19_refGenes_exons.gtf.lg.transc.fa.RNAsnpM3.bed.abspos.p0.05.merged.altT_in_coding_windows.bed&quot;),</span><br><span class="line">    </span><br><span class="line">###################### CLIPdb ############################</span><br><span class="line">#c(&quot;/storage11_7T/fuy/TADA-A/annotation/CLIPdb/human_combine.merged_hg19_refGenes_exons.gtf.lg.transc.fa.RNAsnpM3.bed.abspos.p0.05.merged.altA.bed&quot;,</span><br><span class="line">#&quot;/storage11_7T/fuy/TADA-A/annotation/CLIPdb/human_combine.merged_hg19_refGenes_exons.gtf.lg.transc.fa.RNAsnpM3.bed.abspos.p0.05.merged.altC.bed&quot;,</span><br><span class="line">#&quot;/storage11_7T/fuy/TADA-A/annotation/CLIPdb/human_combine.merged_hg19_refGenes_exons.gtf.lg.transc.fa.RNAsnpM3.bed.abspos.p0.05.merged.altG.bed&quot;,</span><br><span class="line">#&quot;/storage11_7T/fuy/TADA-A/annotation/CLIPdb/human_combine.merged_hg19_refGenes_exons.gtf.lg.transc.fa.RNAsnpM3.bed.abspos.p0.05.merged.altT.bed&quot;)</span><br><span class="line">    </span><br><span class="line">), </span><br><span class="line">                    </span><br><span class="line"></span><br><span class="line">report_proportion = 18665/18665,</span><br><span class="line">#chunk_partition_num =1,</span><br><span class="line">chunk = 2,</span><br><span class="line">node_n = 2,</span><br><span class="line">mutrate_ref_files = c(&quot;db/MS_data/mutrate/Example_windows_extended_1bp_for_getting_base_level_mutrate.bed.fasta.tri.alt_A.mutrate.bw&quot;,</span><br><span class="line">                   &quot;db/MS_data/mutrate/Example_windows_extended_1bp_for_getting_base_level_mutrate.bed.fasta.tri.alt_C.mutrate.bw&quot;,</span><br><span class="line">                   &quot;db/MS_data/mutrate/Example_windows_extended_1bp_for_getting_base_level_mutrate.bed.fasta.tri.alt_G.mutrate.bw&quot;,</span><br><span class="line">                   &quot;db/MS_data/mutrate/Example_windows_extended_1bp_for_getting_base_level_mutrate.bed.fasta.tri.alt_T.mutrate.bw&quot;)</span><br><span class="line">))</span><br><span class="line"></span><br><span class="line">saveRDS(compact_data,paste0(&quot;/storage11_7T/fuy/TADA-A/annotation/&quot;,Sys.Date(),&quot;_cell_DNM_SNV_uniform_prior_RADAR_compact.rds&quot;))</span><br></pre></td></tr></table></figure>


<pre><code> user   system  elapsed 
1.724    0.485 1252.766 </code></pre><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">tm = proc.time()</span><br><span class="line">RR = TADA_A_RR_estimate(data = compact_data$base_info, selected_annotations = c(1), </span><br><span class="line">         gene_prior_file = &quot;db/MS_data/prior/uniform_gene_prior.txt&quot;, optimization_iteration =2000)</span><br><span class="line">proc.time() - tm</span><br><span class="line">saveRDS(compact_data,paste0(&quot;/storage11_7T/fuy/TADA-A/annotation/&quot;,Sys.Date(),&quot;_cell_DNM_SNV_uniform_prior_RADAR_RR.rds&quot;))</span><br><span class="line"></span><br><span class="line">RR$rr_report</span><br></pre></td></tr></table></figure>

<pre><code>[1] &quot;Read in gene prior file. Time consumed: 0.004s.&quot;
[1] &quot;Finished optimization. Time consumed: 178.927.s&quot;
[1] &quot;Got confidence intervals of RR estimates.&quot;
[1] &quot;Finished RR estimation!&quot;



    user   system  elapsed 
1198.020    1.817  178.937 </code></pre><table>
<caption>A data.frame: 1 × 3</caption>
<thead>
    <tr><th scope="col">relative_risk</th><th scope="col">lower_bound</th><th scope="col">upper_bournd</th></tr>
    <tr><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>0.7688921</td><td>0.08826144</td><td>1.449523</td></tr>
</tbody>
</table>




<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">options(scipen=200)</span><br><span class="line">tm = proc.time()</span><br><span class="line">BF = TADA_A_get_BFs(data = compact_data$base_info,</span><br><span class="line">               selected_annotations = c(1), </span><br><span class="line">               rr = RR$rr_report$relative_risk, </span><br><span class="line">               additional_BF_file = c(&quot;/storage11_7T/fuy/TADA-A/annotation/cd_BF/uniform_cd_BFs.txt&quot;), </span><br><span class="line">               TADA_p0 = 0.94)</span><br><span class="line"></span><br><span class="line">g_BF = BF$gene_BF_table</span><br><span class="line">g_BF = g_BF[order(g_BF$FDR_all),]</span><br><span class="line">g_BF2 = g_BF[g_BF$FDR_all &lt;= 0.1,]</span><br><span class="line">proc.time() - tm</span><br><span class="line">nrow(g_BF2)</span><br></pre></td></tr></table></figure>

<pre><code>[1] &quot;Read in additional BF file /storage11_7T/fuy/TADA-A/annotation/cd_BF/uniform_cd_BFs.txt.&quot;
[1] &quot;Flagged genes that don&apos;t have any bases with any informative annotation.&quot;
[1] &quot;Got genenames without bases that have informative rr features.&quot;
[1] &quot;Got logBF_noncoding.&quot;
[1] &quot;Added coding and non-coding logBF. Time consumed: 2.476s.&quot;
[1] &quot;Finished!&quot;



   user  system elapsed 
  6.745   0.004   5.961 </code></pre><p>0</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">head(g_BF)</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.table: 6 × 9</caption>
<thead>
    <tr><th scope="col">genename</th><th scope="col">logBF_noncoding</th><th scope="col">logBF_coding</th><th scope="col">logBF_all</th><th scope="col">BF_noncoding</th><th scope="col">BF_coding</th><th scope="col">BF_all</th><th scope="col">FDR_coding</th><th scope="col">FDR_all</th></tr>
    <tr><th scope="col">&lt;chr&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>SETD5  </td><td>2.061843</td><td>0</td><td>2.061843</td><td>7.860447</td><td>1</td><td>7.860447</td><td>0.94</td><td>0.6658984</td></tr>
    <tr><td>SMARCC1</td><td>1.991872</td><td>0</td><td>1.991872</td><td>7.329241</td><td>1</td><td>7.329241</td><td>0.94</td><td>0.6735895</td></tr>
    <tr><td>PTMS   </td><td>1.504513</td><td>0</td><td>1.504513</td><td>4.501962</td><td>1</td><td>4.501962</td><td>0.94</td><td>0.7079876</td></tr>
    <tr><td>UBB    </td><td>1.448465</td><td>0</td><td>1.448465</td><td>4.256575</td><td>1</td><td>4.256575</td><td>0.94</td><td>0.7275785</td></tr>
    <tr><td>EZR    </td><td>1.420855</td><td>0</td><td>1.420855</td><td>4.140660</td><td>1</td><td>4.140660</td><td>0.94</td><td>0.7402535</td></tr>
    <tr><td>POLR2B </td><td>1.396861</td><td>0</td><td>1.396861</td><td>4.042490</td><td>1</td><td>4.042490</td><td>0.94</td><td>0.7493600</td></tr>
</tbody>
</table>


</div><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">Author: </span><span class="post-copyright-info"><a href="mailto:undefined" target="_blank" rel="noopener"></a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">Link: </span><span class="post-copyright-info"><a href="https://yfu1116.github.io/project/0923_TADA_cell_DNM_SNV_RADAR_RBP/">https://yfu1116.github.io/project/0923_TADA_cell_DNM_SNV_RADAR_RBP/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">Copyright Notice: </span><span class="post-copyright-info">All articles in this blog are licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank" rel="noopener">CC BY-NC-SA 4.0</a> unless stating additionally.</span></div></div><div class="tag_share"><div class="post-meta__tag-list"></div><div class="post_share"><div class="social-share" data-image="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" data-sites="github,email,facebook,twitter,weixin,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js/dist/css/share.min.css"><script src="https://cdn.jsdelivr.net/npm/social-share.js/dist/js/social-share.min.js" defer></script></div></div><nav class="pagination-post" id="pagination"><div class="prev-post pull-left"><a href="/project/0921-cell-DNM-RADAR-RBP/"><img class="prev-cover" data-lazy-src="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" onerror="onerror=null;src='/project/img/404.jpg'"><div class="pagination-info"><div class="label">Previous Post</div><div class="prev_info">0921-cell-DNM-RADAR-RBP</div></div></a></div><div class="next-post pull-right"><a href="/project/0920-tadaA-PTV-Mis/"><img class="next-cover" data-lazy-src="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" onerror="onerror=null;src='/project/img/404.jpg'"><div class="pagination-info"><div class="label">Next Post</div><div class="next_info">0920-tadaA-PTV-Mis</div></div></a></div></nav></article></main><footer id="footer" data-type="color"><div id="footer-wrap"><div class="copyright">&copy;2020 - 2021 By </div><div class="framework-info"><span>Framework </span><a href="https://hexo.io" target="_blank" rel="noopener">Hexo</a><span class="footer-separator">|</span><span>Theme </span><a href="https://github.com/jerryc127/hexo-theme-butterfly" target="_blank" rel="noopener">Butterfly</a></div></div></footer></div><section id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="Read Mode"><i class="fas fa-book-open"></i></button><button id="font_plus" type="button" title="Increase Font Size"><i class="fas fa-plus"></i></button><button id="font_minus" type="button" title="Decrease Font Size"><i class="fas fa-minus"></i></button><button id="darkmode" type="button" title="Switch Between Light And Dark Mode"><i class="fas fa-adjust"></i></button></div><div id="rightside-config-show"><button id="rightside_config" type="button" title="Setting"><i class="fas fa-cog"></i></button><button id="go-up" type="button" title="Back To Top"><i class="fas fa-arrow-up"></i></button></div></section><div><script src="https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js"></script><script src="/project/js/utils.js"></script><script src="/project/js/main.js"></script><script src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.js"></script><script src="https://cdn.jsdelivr.net/npm/instant.page/instantpage.min.js" type="module" defer></script><script src="https://cdn.jsdelivr.net/npm/vanilla-lazyload/dist/lazyload.iife.min.js" async></script><div class="js-pjax"><script>if (!window.MathJax) {
  window.MathJax = {
    loader: {
      source: {
        '[tex]/amsCd': '[tex]/amscd'
      }
    },
    tex: {
      inlineMath: [ ['$','$'], ["\\(","\\)"]],
      tags: 'ams'
    },
    options: {
      renderActions: {
        findScript: [10, doc => {
          for (const node of document.querySelectorAll('script[type^="math/tex"]')) {
            const display = !!node.type.match(/; *mode=display/)
            const math = new doc.options.MathItem(node.textContent, doc.inputJax[0], display)
            const text = document.createTextNode('')
            node.parentNode.replaceChild(text, node)
            math.start = {node: text, delim: '', n: 0}
            math.end = {node: text, delim: '', n: 0}
            doc.math.push(math)
          }
        }, '']
      }
    }
  }
  
  var script = document.createElement('script')
  script.src = 'https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js'
  script.id = 'MathJax-script'
  script.async = true
  document.head.appendChild(script)
} else {
  MathJax.startup.document.state(0)
  MathJax.texReset()
  MathJax.typeset()
}</script><script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div></div></body></html>