<!DOCTYPE html><html lang="en" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1"><title>1119-12166DNM-sf-6430-fam | yang's blog</title><meta name="description" content="6166 DNM  7131 DNM in coding windows ALT should be SNV while REF could be whatever. duplicated  list the de novo variants called in the 6,430 individuals with ASD and 2,179 individuals without ASD (a"><meta name="author" content=""><meta name="copyright" content=""><meta name="format-detection" content="telephone=no"><link rel="shortcut icon" href="/project/img/favicon.png"><link rel="canonical" href="https://yfu1116.github.io/project/1119-12166DNM-sf-6430-fam/"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//fonts.googleapis.com" crossorigin="crossorigin"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><meta property="og:type" content="article"><meta property="og:title" content="1119-12166DNM-sf-6430-fam"><meta property="og:url" content="https://yfu1116.github.io/project/1119-12166DNM-sf-6430-fam/"><meta property="og:site_name" content="yang's blog"><meta property="og:description" content="6166 DNM  7131 DNM in coding windows ALT should be SNV while REF could be whatever. duplicated  list the de novo variants called in the 6,430 individuals with ASD and 2,179 individuals without ASD (a"><meta property="og:image" content="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg"><meta property="article:published_time" content="2020-11-20T04:32:52.354Z"><meta property="article:modified_time" content="2020-11-20T04:33:12.749Z"><meta name="twitter:card" content="summary"><script>var activateDarkMode = function () {
  document.documentElement.setAttribute('data-theme', 'dark')
  if (document.querySelector('meta[name="theme-color"]') !== null) {
    document.querySelector('meta[name="theme-color"]').setAttribute('content', '#000')
  }
}
var activateLightMode = function () {
  document.documentElement.setAttribute('data-theme', 'light')
  if (document.querySelector('meta[name="theme-color"]') !== null) {
    document.querySelector('meta[name="theme-color"]').setAttribute('content', '#fff')
  }
}

var getCookies = function (name) {
  const value = `; ${document.cookie}`
  const parts = value.split(`; ${name}=`)
  if (parts.length === 2) return parts.pop().split(';').shift()
}

var autoChangeMode = 'false'
var t = getCookies('theme')
if (autoChangeMode === '1') {
  var isDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches
  var isLightMode = window.matchMedia('(prefers-color-scheme: light)').matches
  var isNotSpecified = window.matchMedia('(prefers-color-scheme: no-preference)').matches
  var hasNoSupport = !isDarkMode && !isLightMode && !isNotSpecified

  if (t === undefined) {
    if (isLightMode) activateLightMode()
    else if (isDarkMode) activateDarkMode()
    else if (isNotSpecified || hasNoSupport) {
      console.log('You specified no preference for a color scheme or your browser does not support it. I Schedule dark mode during night time.')
      var now = new Date()
      var hour = now.getHours()
      var isNight = hour <= 6 || hour >= 18
      isNight ? activateDarkMode() : activateLightMode()
    }
    window.matchMedia('(prefers-color-scheme: dark)').addListener(function (e) {
      if (Cookies.get('theme') === undefined) {
        e.matches ? activateDarkMode() : activateLightMode()
      }
    })
  } else if (t === 'light') activateLightMode()
  else activateDarkMode()
} else if (autoChangeMode === '2') {
  now = new Date()
  hour = now.getHours()
  isNight = hour <= 6 || hour >= 18
  if (t === undefined) isNight ? activateDarkMode() : activateLightMode()
  else if (t === 'light') activateLightMode()
  else activateDarkMode()
} else {
  if (t === 'dark') activateDarkMode()
  else if (t === 'light') activateLightMode()
}</script><link rel="stylesheet" href="/project/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.css"><link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Titillium+Web&amp;display=swap"><script>var GLOBAL_CONFIG = { 
  root: '/project/',
  hexoversion: '4.0.0',
  algolia: undefined,
  localSearch: undefined,
  translate: undefined,
  noticeOutdate: undefined,
  copy: {
    success: 'Copy successfully',
    error: 'Copy error',
    noSupport: 'The browser does not support'
  },
  bookmark: {
    message_prev: 'Press',
    message_next: 'to bookmark this page'
  },
  runtime_unit: 'days',
  runtime: false,
  copyright: undefined,
  ClickShowText: undefined,
  medium_zoom: false,
  fancybox: true,
  Snackbar: undefined,
  justifiedGallery: {
    js: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/js/jquery.justifiedGallery.min.js',
    css: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/css/justifiedGallery.min.css'
  },
  baiduPush: false,
  highlightCopy: true,
  highlightLang: true,
  isPhotoFigcaption: false,
  islazyload: true,
  isanchor: false    
}</script><script id="config_change">var GLOBAL_CONFIG_SITE = { 
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isSidebar: true,
  postUpdate: '2020-11-20 12:33:12'
}</script><noscript><style>
#nav {
  opacity: 1
}
.justified-gallery img {
  opacity: 1
}
</style></noscript></head><body><div id="mobile-sidebar"><div id="menu_mask"></div><div id="mobile-sidebar-menus"><div class="mobile_author_icon"><img class="avatar-img" data-lazy-src="/project/img/avatar.png" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="mobile_post_data"><div class="mobile_data_item is-center"><div class="mobile_data_link"><a href="/project/archives/"><div class="headline">Articles</div><div class="length_num">33</div></a></div></div></div><hr/></div></div><div id="body-wrap"><div id="sidebar"><i class="fas fa-arrow-right on" id="toggle-sidebar"></i><div class="sidebar-toc"><div class="sidebar-toc__title">Catalog</div><div class="sidebar-toc__progress"><span class="progress-notice">You've read</span><span class="progress-num">0</span><span class="progress-percentage">%</span><div class="sidebar-toc__progress-bar">     </div></div><div class="sidebar-toc__content"><ol class="toc"><li class="toc-item toc-level-3"><a class="toc-link" href="#MPC2"><span class="toc-number">1.</span> <span class="toc-text">MPC2</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#pli995"><span class="toc-number">2.</span> <span class="toc-text">pli995</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#autos-cd-syn-calibr"><span class="toc-number">3.</span> <span class="toc-text">autos cd syn calibr</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#auto-cd-syn-uncalibr"><span class="toc-number">4.</span> <span class="toc-text">auto cd syn uncalibr</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#MPC2-1"><span class="toc-number">5.</span> <span class="toc-text">MPC2</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#pli995-1"><span class="toc-number">6.</span> <span class="toc-text">pli995</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#calib-syn"><span class="toc-number">7.</span> <span class="toc-text">calib-syn</span></a></li></ol></div></div></div><header class="post-bg" id="page-header" style="background-image: url(https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg)"><nav id="nav"><span class="pull-left" id="blog_name"><a class="blog_title" id="site-name" href="/project/">yang's blog</a></span><span class="pull-right menus"><span class="toggle-menu close"><a class="site-page"><i class="fas fa-bars fa-fw"></i></a></span></span></nav><div id="post-info"><div id="post-title"><div class="posttitle">1119-12166DNM-sf-6430-fam</div></div><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">Created</span><time class="post-meta-date-created" datetime="2020-11-20T04:32:52.354Z" title="Created 2020-11-20 12:32:52">2020-11-20</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">Updated</span><time class="post-meta-date-updated" datetime="2020-11-20T04:33:12.749Z" title="Updated 2020-11-20 12:33:12">2020-11-20</time></span></div><div class="meta-secondline"> <span class="post-meta-separator">|</span><span class="post-meta-pv-cv"><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">Post View:</span><span id="busuanzi_value_page_pv"></span></span></div></div></div></header><main class="layout_post" id="content-inner"><article id="post"><div class="post-content" id="article-container"><p>6166 DNM</p>
<ul>
<li>7131 DNM in coding windows</li>
<li>ALT should be SNV while REF could be whatever.</li>
<li>duplicated</li>
</ul>
<p>list the de novo variants called in the 6,430 individuals with ASD and 2,179 individuals without ASD (a total of 15,789 de novo variants were identified covering 9,345 coding variants and 6,444 non-coding variants</p>
<p>From the family-based data, we identified 9,345 rare de novo variants in protein-coding exons (allele frequency ≤ 0.1% in our dataset and non-psychiatric subsets of reference databases): 63% of cases and 59% of unaffected siblings carried at least one such variant (4,073 of 6,430 and 1,294 of 2,179, respectively</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">f = fread(&quot;/storage11_7T/fuy/TADA-A/cell_WES/cd_uniform_scaling_factors.txt&quot;)</span><br><span class="line">colnames(f)[2] = &quot;z&quot;</span><br><span class="line">f$scaling_factor = f$z * (1484/1630)</span><br><span class="line">fwrite(f[,c(1,3)],&quot;/storage11_7T/fuy/TADA-A/cell_WES/cd_uniform_scaling_factors_1484_d_1630.txt&quot;,sep=&quot;\t&quot;)</span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">library(tadaA)</span><br><span class="line">setwd(&quot;/storage11_7T/fuy/TADA-A&quot;)</span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br></pre></td><td class="code"><pre><span class="line">system.time(compact_data &lt;- TADA_A_read_info_by_chunks(</span><br><span class="line">mut_files = &quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/12166DNM.affected.allele.bed&quot;,</span><br><span class="line">    </span><br><span class="line"># c(&quot;db/MS_data/DNM/Jiang_cases_DNM_with_allele_info.txt&quot;,</span><br><span class="line"># &quot;db/MS_data/DNM/Kong_cases_DNM_with_allele_info.txt&quot;,</span><br><span class="line"># &quot;db/MS_data/DNM/Michaelson_cases_DNM_with_allele_info.txt&quot;,</span><br><span class="line"># &quot;db/MS_data/DNM/Yuen_NM2015_cases_DNM_with_allele_info.txt&quot;,</span><br><span class="line"># &quot;db/MS_data/DNM/Wu_cases_DNM_with_allele_info.txt&quot;),    </span><br><span class="line">    </span><br><span class="line">window_file =  &apos;db/MS_data/windows_partition/cd_windows_with_div_score.bed&apos;,</span><br><span class="line">    </span><br><span class="line"># mutrate_scaling_files = c(&quot;/storage11_7T/fuy/TADA-A/cell_WES/cd_uniform_scaling_factors_1441_d_1692.txt&quot;),  </span><br><span class="line"># mutrate_scaling_files = c(&quot;/storage11_7T/fuy/TADA-A/cell_WES/cd_uniform_scaling_factors.txt&quot;),</span><br><span class="line"># mutrate_scaling_files = c(&quot;/storage11_7T/fuy/TADA-A/cell_WES/cd_uniform_scaling_factors_1441_d_1068.txt&quot;),</span><br><span class="line"># mutrate_scaling_files = c(&quot;/storage11_7T/fuy/TADA-A/cell_WES/cd_uniform_scaling_factors_1435_d_1052.txt&quot;),    </span><br><span class="line"># mutrate_scaling_files = c(&quot;/storage11_7T/fuy/TADA-A/cell_WES/cd_uniform_scaling_factors_1505_d_1692.txt&quot;),</span><br><span class="line">mutrate_scaling_files = c(&quot;/storage11_7T/fuy/TADA-A/cell_WES/cd_uniform_scaling_factors_1484_d_1630.txt&quot;),</span><br><span class="line">    </span><br><span class="line">sample_sizes = 6430,    </span><br><span class="line"># sample_sizes = 4059,</span><br><span class="line"># sample_sizes = 4000,</span><br><span class="line">    </span><br><span class="line"># mutrate_scaling_files = c(&quot;db/MS_data/results/Jiang_windows_mutrate_with_div_score_scaling_file_for_test_DNM.txt&quot;,</span><br><span class="line">#                          &quot;db/MS_data/results/Kong_windows_mutrate_with_div_score_scaling_file_for_test_DNM.txt&quot;,</span><br><span class="line">#                          &quot;db/MS_data/results/Michaelson_windows_mutrate_with_div_score_scaling_file_for_test_DNM.txt&quot;,</span><br><span class="line">#                          &quot;db/MS_data/results/Yuen_NM2015_windows_mutrate_with_div_score_scaling_file_for_test_DNM.txt&quot;,</span><br><span class="line">#                          &quot;db/MS_data/results/Wu_windows_mutrate_with_div_score_scaling_file_for_test_DNM.txt&quot;),    </span><br><span class="line">    </span><br><span class="line"># sample_sizes = c(32,78,10,162,32),    </span><br><span class="line">    </span><br><span class="line">    </span><br><span class="line">gene_prior_file = &quot;db/MS_data/prior/uniform_gene_prior.txt&quot;,</span><br><span class="line">    </span><br><span class="line">nonAS_noncoding_annotations = NA, #c( </span><br><span class="line">#                                  &quot;/storage11_7T/fuy/TADA-A/annotation/ccr/ccrs.allchrom.gt90.bed&quot;),</span><br><span class="line">                                #&quot;/storage11_7T/fuy/TADA-A/annotation/ccr/chr_ccr_gt95_syn_rm.bed&quot;),</span><br><span class="line"></span><br><span class="line">AS_noncoding_annotations = list(</span><br><span class="line"></span><br><span class="line">#                               c(&quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/autos.cd.window.all_cd_syn_alt_A.bed&quot;,</span><br><span class="line">#                                 &quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/autos.cd.window.all_cd_syn_alt_C.bed&quot;,</span><br><span class="line">#                                 &quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/autos.cd.window.all_cd_syn_alt_G.bed&quot;,</span><br><span class="line">#                                 &quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/autos.cd.window.all_cd_syn_alt_T.bed&quot;)</span><br><span class="line">    </span><br><span class="line">c(&quot;/storage11_7T/fuy/TADA-A/annotation/pLI/auto/auto.annovar_cd_window_pli995_altA.bed&quot;,</span><br><span class="line"> &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/auto/auto.annovar_cd_window_pli995_altC.bed&quot;,</span><br><span class="line"> &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/auto/auto.annovar_cd_window_pli995_altG.bed&quot;,</span><br><span class="line"> &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/auto/auto.annovar_cd_window_pli995_altT.bed&quot;) ,</span><br><span class="line">    </span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/pLI/annovar_cd_window_pli05-995_altA.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/annovar_cd_window_pli05-995_altC.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/annovar_cd_window_pli05-995_altG.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/annovar_cd_window_pli05-995_altT.bed&quot;) ,</span><br><span class="line">    </span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/pLI/annovar_cd_window_pli0-05_altA.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/annovar_cd_window_pli0-05_altC.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/annovar_cd_window_pli0-05_altG.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/annovar_cd_window_pli0-05_altT.bed&quot;)  ,</span><br><span class="line">    </span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/db/MS_data/annota/spidex_public_noncommercial_v1_0.tab_alt_A_lower10pct.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/db/MS_data/annota/spidex_public_noncommercial_v1_0.tab_alt_C_lower10pct.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/db/MS_data/annota/spidex_public_noncommercial_v1_0.tab_alt_G_lower10pct.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/db/MS_data/annota/spidex_public_noncommercial_v1_0.tab_alt_T_lower10pct.bed&quot;) ,</span><br><span class="line">    </span><br><span class="line">c(&quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/annovar/auto/auto.A.cd.MPC2.annovar.bed&quot;,</span><br><span class="line">&quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/annovar/auto/auto.C.cd.MPC2.annovar.bed&quot;,</span><br><span class="line"> &quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/annovar/auto/auto.G.cd.MPC2.annovar.bed&quot;,</span><br><span class="line"> &quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/annovar/auto/auto.T.cd.MPC2.annovar.bed&quot;) </span><br><span class="line">    </span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/annovar/A.cd.MPC12.annovar.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/annovar/C.cd.MPC12.annovar.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/annovar/G.cd.MPC12.annovar.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/annovar/T.cd.MPC12.annovar.bed&quot;) ,    </span><br><span class="line">    </span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/annovar/A.cd.MPC01.annovar.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/annovar/C.cd.MPC01.annovar.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/annovar/G.cd.MPC01.annovar.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/MPC_score/annovar/T.cd.MPC01.annovar.bed&quot;) ,</span><br><span class="line">    </span><br><span class="line">##############</span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/primateAI/chr_primateAI_exome_mutation_pathogen_rank_gt_80_alt_A.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/primateAI/chr_primateAI_exome_mutation_pathogen_rank_gt_80_alt_C.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/primateAI/chr_primateAI_exome_mutation_pathogen_rank_gt_80_alt_G.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/primateAI/chr_primateAI_exome_mutation_pathogen_rank_gt_80_alt_T.bed&quot;) ,</span><br><span class="line">    </span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/MVP/chr_MVP_all_rare_missense_pathogen_rank_gt_75_alt_A.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/MVP/chr_MVP_all_rare_missense_pathogen_rank_gt_75_alt_C.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/MVP/chr_MVP_all_rare_missense_pathogen_rank_gt_75_alt_G.bed&quot;,</span><br><span class="line"># &quot;/storage11_7T/fuy/TADA-A/annotation/MVP/chr_MVP_all_rare_missense_pathogen_rank_gt_75_alt_T.bed&quot;)</span><br><span class="line">    </span><br><span class="line">    </span><br><span class="line">),                    </span><br><span class="line"></span><br><span class="line">report_proportion = 18665/18665,</span><br><span class="line">#chunk_partition_num =1,</span><br><span class="line">chunk = 2,</span><br><span class="line">node_n = 2,</span><br><span class="line">mutrate_ref_files = c(&quot;db/MS_data/mutrate/Example_windows_extended_1bp_for_getting_base_level_mutrate.bed.fasta.tri.alt_A.mutrate.bw&quot;,</span><br><span class="line">                   &quot;db/MS_data/mutrate/Example_windows_extended_1bp_for_getting_base_level_mutrate.bed.fasta.tri.alt_C.mutrate.bw&quot;,</span><br><span class="line">                   &quot;db/MS_data/mutrate/Example_windows_extended_1bp_for_getting_base_level_mutrate.bed.fasta.tri.alt_G.mutrate.bw&quot;,</span><br><span class="line">                   &quot;db/MS_data/mutrate/Example_windows_extended_1bp_for_getting_base_level_mutrate.bed.fasta.tri.alt_T.mutrate.bw&quot;)</span><br><span class="line">    </span><br><span class="line"># mutrate_ref_files = c(&quot;/storage11_7T/data.for.yuwen/new_mutrate/window.hg19.genome.tri2.alt_A.uq.bedGraph.sort.bw&quot; ,</span><br><span class="line">#                       &quot;/storage11_7T/data.for.yuwen/new_mutrate/window.hg19.genome.tri2.alt_C.uq.bedGraph.sort.bw&quot; ,</span><br><span class="line">#                       &quot;/storage11_7T/data.for.yuwen/new_mutrate/window.hg19.genome.tri2.alt_G.uq.bedGraph.sort.bw&quot; ,</span><br><span class="line">#                       &quot;/storage11_7T/data.for.yuwen/new_mutrate/window.hg19.genome.tri2.alt_T.uq.bedGraph.sort.bw&quot; )</span><br><span class="line">                      </span><br><span class="line"></span><br><span class="line">))</span><br><span class="line"></span><br><span class="line">saveRDS(compact_data,paste0(&quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/&quot;,Sys.Date(),&quot;_12166DNM_6430_fam_auto_pli995_MPC2_1484_d_1630_compact.rds&quot;))</span><br></pre></td></tr></table></figure>


<pre><code> user   system  elapsed 
2.296    0.680 1552.936 </code></pre><h3 id="MPC2"><a href="#MPC2" class="headerlink" title="MPC2"></a>MPC2</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br></pre></td><td class="code"><pre><span class="line">RR = TADA_A_RR_estimate(data = compact_data$base_info, selected_annotations = c(2), </span><br><span class="line">#           gene_prior_file = &quot;db/MS_data/prior/Example_gene_prior.txt&quot;, </span><br><span class="line">        gene_prior_file = &quot;db/MS_data/prior/uniform_gene_prior.txt&quot;,</span><br><span class="line">                        optimization_iteration = 2000)</span><br><span class="line">RR$rr_report</span><br><span class="line"></span><br><span class="line">s = 0</span><br><span class="line">for(j in 1:length(compact_data$base_info))&#123;</span><br><span class="line">    </span><br><span class="line">lg = length(compact_data$base_info[[j]])</span><br><span class="line">for(i in 1:lg)&#123;</span><br><span class="line">    if(compact_data$base_info[[j]][[i]]$feature_vector[2] == 1)&#123;</span><br><span class="line">        s = s + compact_data$base_info[[j]][[i]]$sum_mut_rate</span><br><span class="line">    &#125;</span><br><span class="line">&#125;</span><br><span class="line">    &#125;</span><br><span class="line">s  #### background</span><br><span class="line"></span><br><span class="line">s = 0</span><br><span class="line">for(j in 1:length(compact_data$base_info))&#123;</span><br><span class="line">    </span><br><span class="line">lg = length(compact_data$base_info[[j]])</span><br><span class="line">for(i in 1:lg)&#123;</span><br><span class="line">    if(compact_data$base_info[[j]][[i]]$feature_vector[2] == 1)&#123;</span><br><span class="line">        s = s + compact_data$base_info[[j]][[i]]$sum_mut_count</span><br><span class="line">    &#125;</span><br><span class="line">&#125;</span><br><span class="line">    &#125;</span><br><span class="line">s  #### observed</span><br></pre></td></tr></table></figure>

<pre><code>[1] &quot;Read in gene prior file. Time consumed: 0.004s.&quot;
[1] &quot;Finished optimization. Time consumed: 94.947.s&quot;
[1] &quot;Got confidence intervals of RR estimates.&quot;
[1] &quot;Finished RR estimation!&quot;</code></pre><table>
<caption>A data.frame: 1 × 3</caption>
<thead>
    <tr><th scope="col">logRR</th><th scope="col">lower_bound</th><th scope="col">upper_bound</th></tr>
    <tr><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>2.311404</td><td>2.074534</td><td>2.548275</td></tr>
</tbody>
</table>




<p>179.871940314055</p>
<p>280</p>
<h3 id="pli995"><a href="#pli995" class="headerlink" title="pli995"></a>pli995</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br></pre></td><td class="code"><pre><span class="line">RR = TADA_A_RR_estimate(data = compact_data$base_info, selected_annotations = c(1), </span><br><span class="line">#           gene_prior_file = &quot;db/MS_data/prior/Example_gene_prior.txt&quot;, </span><br><span class="line">        gene_prior_file = &quot;db/MS_data/prior/uniform_gene_prior.txt&quot;,</span><br><span class="line">                        optimization_iteration = 2000)</span><br><span class="line">RR$rr_report</span><br><span class="line"></span><br><span class="line">s = 0</span><br><span class="line">for(j in 1:length(compact_data$base_info))&#123;</span><br><span class="line">    </span><br><span class="line">lg = length(compact_data$base_info[[j]])</span><br><span class="line">for(i in 1:lg)&#123;</span><br><span class="line">    if(compact_data$base_info[[j]][[i]]$feature_vector[1] == 1)&#123;</span><br><span class="line">        s = s + compact_data$base_info[[j]][[i]]$sum_mut_rate</span><br><span class="line">    &#125;</span><br><span class="line">&#125;</span><br><span class="line">    &#125;</span><br><span class="line">s  #### background</span><br><span class="line"></span><br><span class="line">s = 0</span><br><span class="line">for(j in 1:length(compact_data$base_info))&#123;</span><br><span class="line">    </span><br><span class="line">lg = length(compact_data$base_info[[j]])</span><br><span class="line">for(i in 1:lg)&#123;</span><br><span class="line">    if(compact_data$base_info[[j]][[i]]$feature_vector[1] == 1)&#123;</span><br><span class="line">        s = s + compact_data$base_info[[j]][[i]]$sum_mut_count</span><br><span class="line">    &#125;</span><br><span class="line">&#125;</span><br><span class="line">    &#125;</span><br><span class="line">s  #### observed</span><br></pre></td></tr></table></figure>

<pre><code>[1] &quot;Read in gene prior file. Time consumed: 0.004s.&quot;
[1] &quot;Finished optimization. Time consumed: 56.247.s&quot;
[1] &quot;Got confidence intervals of RR estimates.&quot;
[1] &quot;Finished RR estimation!&quot;</code></pre><table>
<caption>A data.frame: 1 × 3</caption>
<thead>
    <tr><th scope="col">logRR</th><th scope="col">lower_bound</th><th scope="col">upper_bound</th></tr>
    <tr><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>3.164477</td><td>2.894144</td><td>3.434811</td></tr>
</tbody>
</table>




<p>52.458127723218</p>
<p>153</p>
<h3 id="autos-cd-syn-calibr"><a href="#autos-cd-syn-calibr" class="headerlink" title="autos cd syn calibr"></a>autos cd syn calibr</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br></pre></td><td class="code"><pre><span class="line">RR = TADA_A_RR_estimate(data = compact_data$base_info, selected_annotations = c(1), </span><br><span class="line">#           gene_prior_file = &quot;db/MS_data/prior/Example_gene_prior.txt&quot;, </span><br><span class="line">        gene_prior_file = &quot;db/MS_data/prior/uniform_gene_prior.txt&quot;,</span><br><span class="line">                        optimization_iteration = 2000)</span><br><span class="line">RR$rr_report</span><br><span class="line"></span><br><span class="line">s = 0</span><br><span class="line">for(j in 1:length(compact_data$base_info))&#123;</span><br><span class="line">    </span><br><span class="line">lg = length(compact_data$base_info[[j]])</span><br><span class="line">for(i in 1:lg)&#123;</span><br><span class="line">    if(compact_data$base_info[[j]][[i]]$feature_vector[1] == 1)&#123;</span><br><span class="line">        s = s + compact_data$base_info[[j]][[i]]$sum_mut_rate</span><br><span class="line">    &#125;</span><br><span class="line">&#125;</span><br><span class="line">    &#125;</span><br><span class="line">s  #### background</span><br><span class="line"></span><br><span class="line">s = 0</span><br><span class="line">for(j in 1:length(compact_data$base_info))&#123;</span><br><span class="line">    </span><br><span class="line">lg = length(compact_data$base_info[[j]])</span><br><span class="line">for(i in 1:lg)&#123;</span><br><span class="line">    if(compact_data$base_info[[j]][[i]]$feature_vector[1] == 1)&#123;</span><br><span class="line">        s = s + compact_data$base_info[[j]][[i]]$sum_mut_count</span><br><span class="line">    &#125;</span><br><span class="line">&#125;</span><br><span class="line">    &#125;</span><br><span class="line">s  #### observed</span><br></pre></td></tr></table></figure>

<pre><code>[1] &quot;Read in gene prior file. Time consumed: 0.052s.&quot;
[1] &quot;Finished optimization. Time consumed: 343.387.s&quot;
[1] &quot;Got confidence intervals of RR estimates.&quot;
[1] &quot;Finished RR estimation!&quot;</code></pre><table>
<caption>A data.frame: 1 × 3</caption>
<thead>
    <tr><th scope="col">logRR</th><th scope="col">lower_bound</th><th scope="col">upper_bound</th></tr>
    <tr><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>0.01541753</td><td>-2.273329</td><td>2.304164</td></tr>
</tbody>
</table>




<p>1483.81264325268</p>
<p>1484</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">exp(.0154)</span><br></pre></td></tr></table></figure>


<p>1.01551919106144</p>
<h3 id="auto-cd-syn-uncalibr"><a href="#auto-cd-syn-uncalibr" class="headerlink" title="auto cd syn uncalibr"></a>auto cd syn uncalibr</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br></pre></td><td class="code"><pre><span class="line">RR = TADA_A_RR_estimate(data = compact_data$base_info, selected_annotations = c(1), </span><br><span class="line">#           gene_prior_file = &quot;db/MS_data/prior/Example_gene_prior.txt&quot;, </span><br><span class="line">        gene_prior_file = &quot;db/MS_data/prior/uniform_gene_prior.txt&quot;,</span><br><span class="line">                        optimization_iteration = 2000)</span><br><span class="line">RR$rr_report</span><br><span class="line"></span><br><span class="line">s = 0</span><br><span class="line">for(j in 1:length(compact_data$base_info))&#123;</span><br><span class="line">    </span><br><span class="line">lg = length(compact_data$base_info[[j]])</span><br><span class="line">for(i in 1:lg)&#123;</span><br><span class="line">    if(compact_data$base_info[[j]][[i]]$feature_vector[1] == 1)&#123;</span><br><span class="line">        s = s + compact_data$base_info[[j]][[i]]$sum_mut_rate</span><br><span class="line">    &#125;</span><br><span class="line">&#125;</span><br><span class="line">    &#125;</span><br><span class="line">s  #### background</span><br><span class="line"></span><br><span class="line">s = 0</span><br><span class="line">for(j in 1:length(compact_data$base_info))&#123;</span><br><span class="line">    </span><br><span class="line">lg = length(compact_data$base_info[[j]])</span><br><span class="line">for(i in 1:lg)&#123;</span><br><span class="line">    if(compact_data$base_info[[j]][[i]]$feature_vector[1] == 1)&#123;</span><br><span class="line">        s = s + compact_data$base_info[[j]][[i]]$sum_mut_count</span><br><span class="line">    &#125;</span><br><span class="line">&#125;</span><br><span class="line">    &#125;</span><br><span class="line">s  #### observed</span><br></pre></td></tr></table></figure>

<pre><code>[1] &quot;Read in gene prior file. Time consumed: 0.004s.&quot;
[1] &quot;Finished optimization. Time consumed: 668.575.s&quot;
[1] &quot;Got confidence intervals of RR estimates.&quot;
[1] &quot;Finished RR estimation!&quot;</code></pre><table>
<caption>A data.frame: 1 × 3</caption>
<thead>
    <tr><th scope="col">logRR</th><th scope="col">lower_bound</th><th scope="col">upper_bound</th></tr>
    <tr><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>-1</td><td>-2.10121</td><td>0.1012101</td></tr>
</tbody>
</table>




<p>1629.79421058078</p>
<p>1484</p>
<h3 id="MPC2-1"><a href="#MPC2-1" class="headerlink" title="MPC2"></a>MPC2</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br></pre></td><td class="code"><pre><span class="line">RR = TADA_A_RR_estimate(data = compact_data$base_info, selected_annotations = c(1), </span><br><span class="line">#           gene_prior_file = &quot;db/MS_data/prior/Example_gene_prior.txt&quot;, </span><br><span class="line">        gene_prior_file = &quot;db/MS_data/prior/uniform_gene_prior.txt&quot;,</span><br><span class="line">                        optimization_iteration = 2000)</span><br><span class="line">RR$rr_report</span><br><span class="line"></span><br><span class="line">s = 0</span><br><span class="line">for(j in 1:length(compact_data$base_info))&#123;</span><br><span class="line">    </span><br><span class="line">lg = length(compact_data$base_info[[j]])</span><br><span class="line">for(i in 1:lg)&#123;</span><br><span class="line">    if(compact_data$base_info[[j]][[i]]$feature_vector[1] == 1)&#123;</span><br><span class="line">        s = s + compact_data$base_info[[j]][[i]]$sum_mut_rate</span><br><span class="line">    &#125;</span><br><span class="line">&#125;</span><br><span class="line">    &#125;</span><br><span class="line">s  #### background</span><br><span class="line"></span><br><span class="line">s = 0</span><br><span class="line">for(j in 1:length(compact_data$base_info))&#123;</span><br><span class="line">    </span><br><span class="line">lg = length(compact_data$base_info[[j]])</span><br><span class="line">for(i in 1:lg)&#123;</span><br><span class="line">    if(compact_data$base_info[[j]][[i]]$feature_vector[1] == 1)&#123;</span><br><span class="line">        s = s + compact_data$base_info[[j]][[i]]$sum_mut_count</span><br><span class="line">    &#125;</span><br><span class="line">&#125;</span><br><span class="line">    &#125;</span><br><span class="line">s  #### observed</span><br></pre></td></tr></table></figure>

<h3 id="pli995-1"><a href="#pli995-1" class="headerlink" title="pli995"></a>pli995</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br></pre></td><td class="code"><pre><span class="line">RR = TADA_A_RR_estimate(data = compact_data$base_info, selected_annotations = c(1), </span><br><span class="line">#           gene_prior_file = &quot;db/MS_data/prior/Example_gene_prior.txt&quot;, </span><br><span class="line">        gene_prior_file = &quot;db/MS_data/prior/uniform_gene_prior.txt&quot;,</span><br><span class="line">                        optimization_iteration = 2000)</span><br><span class="line">RR$rr_report</span><br><span class="line"></span><br><span class="line">s = 0</span><br><span class="line">for(j in 1:length(compact_data$base_info))&#123;</span><br><span class="line">    </span><br><span class="line">lg = length(compact_data$base_info[[j]])</span><br><span class="line">for(i in 1:lg)&#123;</span><br><span class="line">    if(compact_data$base_info[[j]][[i]]$feature_vector[1] == 1)&#123;</span><br><span class="line">        s = s + compact_data$base_info[[j]][[i]]$sum_mut_rate</span><br><span class="line">    &#125;</span><br><span class="line">&#125;</span><br><span class="line">    &#125;</span><br><span class="line">s  #### background</span><br><span class="line"></span><br><span class="line">s = 0</span><br><span class="line">for(j in 1:length(compact_data$base_info))&#123;</span><br><span class="line">    </span><br><span class="line">lg = length(compact_data$base_info[[j]])</span><br><span class="line">for(i in 1:lg)&#123;</span><br><span class="line">    if(compact_data$base_info[[j]][[i]]$feature_vector[1] == 1)&#123;</span><br><span class="line">        s = s + compact_data$base_info[[j]][[i]]$sum_mut_count</span><br><span class="line">    &#125;</span><br><span class="line">&#125;</span><br><span class="line">    &#125;</span><br><span class="line">s  #### observed</span><br></pre></td></tr></table></figure>

<pre><code>[1] &quot;Read in gene prior file. Time consumed: 0.004s.&quot;
[1] &quot;Finished optimization. Time consumed: 16.582.s&quot;
[1] &quot;Got confidence intervals of RR estimates.&quot;
[1] &quot;Finished RR estimation!&quot;</code></pre><table>
<caption>A data.frame: 1 × 3</caption>
<thead>
    <tr><th scope="col">logRR</th><th scope="col">lower_bound</th><th scope="col">upper_bound</th></tr>
    <tr><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>3.165134</td><td>2.895437</td><td>3.434831</td></tr>
</tbody>
</table>




<p>54.1597245992531</p>
<p>154</p>
<h3 id="calib-syn"><a href="#calib-syn" class="headerlink" title="calib-syn"></a>calib-syn</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br></pre></td><td class="code"><pre><span class="line">RR = TADA_A_RR_estimate(data = compact_data$base_info, selected_annotations = c(1), </span><br><span class="line">#           gene_prior_file = &quot;db/MS_data/prior/Example_gene_prior.txt&quot;, </span><br><span class="line">        gene_prior_file = &quot;db/MS_data/prior/uniform_gene_prior.txt&quot;,</span><br><span class="line">                        optimization_iteration = 2000)</span><br><span class="line">RR$rr_report</span><br><span class="line"></span><br><span class="line">s = 0</span><br><span class="line">for(j in 1:length(compact_data$base_info))&#123;</span><br><span class="line">    </span><br><span class="line">lg = length(compact_data$base_info[[j]])</span><br><span class="line">for(i in 1:lg)&#123;</span><br><span class="line">    if(compact_data$base_info[[j]][[i]]$feature_vector[1] == 1)&#123;</span><br><span class="line">        s = s + compact_data$base_info[[j]][[i]]$sum_mut_rate</span><br><span class="line">    &#125;</span><br><span class="line">&#125;</span><br><span class="line">    &#125;</span><br><span class="line">s  #### background</span><br><span class="line"></span><br><span class="line">s = 0</span><br><span class="line">for(j in 1:length(compact_data$base_info))&#123;</span><br><span class="line">    </span><br><span class="line">lg = length(compact_data$base_info[[j]])</span><br><span class="line">for(i in 1:lg)&#123;</span><br><span class="line">    if(compact_data$base_info[[j]][[i]]$feature_vector[1] == 1)&#123;</span><br><span class="line">        s = s + compact_data$base_info[[j]][[i]]$sum_mut_count</span><br><span class="line">    &#125;</span><br><span class="line">&#125;</span><br><span class="line">    &#125;</span><br><span class="line">s  #### observed</span><br></pre></td></tr></table></figure>

<pre><code>[1] &quot;Read in gene prior file. Time consumed: 0.073s.&quot;
[1] &quot;Finished optimization. Time consumed: 303.606.s&quot;
[1] &quot;Got confidence intervals of RR estimates.&quot;
[1] &quot;Finished RR estimation!&quot;</code></pre><table>
<caption>A data.frame: 1 × 3</caption>
<thead>
    <tr><th scope="col">logRR</th><th scope="col">lower_bound</th><th scope="col">upper_bound</th></tr>
    <tr><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>0.03115438</td><td>-2.734808</td><td>2.797117</td></tr>
</tbody>
</table>




<p>1504.76043243318</p>
<p>1505</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">exp(0.03)</span><br></pre></td></tr></table></figure>


<p>1.03045453395352</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">RR = TADA_A_RR_estimate(data = compact_data$base_info, selected_annotations = c(1), </span><br><span class="line">#           gene_prior_file = &quot;db/MS_data/prior/Example_gene_prior.txt&quot;, </span><br><span class="line">        gene_prior_file = &quot;db/MS_data/prior/uniform_gene_prior.txt&quot;,</span><br><span class="line">                        optimization_iteration = 2000)</span><br><span class="line">RR</span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><span class="line">s = 0</span><br><span class="line">for(j in 1:length(compact_data$base_info))&#123;</span><br><span class="line">    </span><br><span class="line">lg = length(compact_data$base_info[[j]])</span><br><span class="line">for(i in 1:lg)&#123;</span><br><span class="line">    if(compact_data$base_info[[j]][[i]]$feature_vector[1] == 1)&#123;</span><br><span class="line">        s = s + compact_data$base_info[[j]][[i]]$sum_mut_rate</span><br><span class="line">    &#125;</span><br><span class="line">&#125;</span><br><span class="line">    &#125;</span><br><span class="line">s  #### background</span><br><span class="line"></span><br><span class="line">s = 0</span><br><span class="line">for(j in 1:length(compact_data$base_info))&#123;</span><br><span class="line">    </span><br><span class="line">lg = length(compact_data$base_info[[j]])</span><br><span class="line">for(i in 1:lg)&#123;</span><br><span class="line">    if(compact_data$base_info[[j]][[i]]$feature_vector[1] == 1)&#123;</span><br><span class="line">        s = s + compact_data$base_info[[j]][[i]]$sum_mut_count</span><br><span class="line">    &#125;</span><br><span class="line">&#125;</span><br><span class="line">    &#125;</span><br><span class="line">s  #### observed</span><br></pre></td></tr></table></figure>


<p>1691.73066556607</p>
<p>1505</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">tm = proc.time()</span><br><span class="line">dfn = c()</span><br><span class="line">for(i in 9:10)&#123;</span><br><span class="line">    </span><br><span class="line">RR = TADA_A_RR_estimate(data = compact_data$base_info, selected_annotations = c(i), </span><br><span class="line">#           gene_prior_file = &quot;db/MS_data/prior/Example_gene_prior.txt&quot;, </span><br><span class="line">        gene_prior_file = &quot;db/MS_data/prior/uniform_gene_prior.txt&quot;,</span><br><span class="line">                        optimization_iteration = 2000)</span><br><span class="line">    </span><br><span class="line">dfn = rbind(dfn,RR$rr_report)</span><br><span class="line">    &#125;</span><br><span class="line">proc.time() - tm</span><br><span class="line"># saveRDS(RR,paste0(&quot;/storage11_7T/fuy/TADA-A/annotation/results/&quot;,Sys.Date(),&quot;_wes_deepsea_brain_RR.rds&quot;))</span><br><span class="line"></span><br><span class="line">dfn</span><br></pre></td></tr></table></figure>

<pre><code>[1] &quot;Read in gene prior file. Time consumed: 0.081s.&quot;
[1] &quot;Finished optimization. Time consumed: 365.757.s&quot;
[1] &quot;Got confidence intervals of RR estimates.&quot;
[1] &quot;Finished RR estimation!&quot;
[1] &quot;Read in gene prior file. Time consumed: 0.005s.&quot;
[1] &quot;Finished optimization. Time consumed: 349.952.s&quot;
[1] &quot;Got confidence intervals of RR estimates.&quot;
[1] &quot;Finished RR estimation!&quot;



    user   system  elapsed 
3598.031    4.140  715.976 </code></pre><table>
<caption>A data.frame: 2 × 3</caption>
<thead>
    <tr><th scope="col">logRR</th><th scope="col">lower_bound</th><th scope="col">upper_bound</th></tr>
    <tr><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>1.592529</td><td>1.387401</td><td>1.797657</td></tr>
    <tr><td>1.776647</td><td>1.623609</td><td>1.929684</td></tr>
</tbody>
</table>




<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">RR = TADA_A_RR_estimate(data = compact_data$base_info, selected_annotations = c(1,2), </span><br><span class="line">#           gene_prior_file = &quot;db/MS_data/prior/Example_gene_prior.txt&quot;, </span><br><span class="line">        gene_prior_file = &quot;db/MS_data/prior/uniform_gene_prior.txt&quot;,</span><br><span class="line">                        optimization_iteration = 2000)</span><br></pre></td></tr></table></figure>

<pre><code>[1] &quot;Read in gene prior file. Time consumed: 0.017s.&quot;
[1] &quot;Finished optimization. Time consumed: 388.177.s&quot;
[1] &quot;Got confidence intervals of RR estimates.&quot;
[1] &quot;Finished RR estimation!&quot;</code></pre><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">RR$rr_report</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.frame: 2 × 3</caption>
<thead>
    <tr><th scope="col">logRR</th><th scope="col">lower_bound</th><th scope="col">upper_bound</th></tr>
    <tr><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>3.143814</td><td>2.878836</td><td>3.408791</td></tr>
    <tr><td>2.278504</td><td>2.050449</td><td>2.506558</td></tr>
</tbody>
</table>




<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line">options(scipen=200)</span><br><span class="line">tm = proc.time()</span><br><span class="line">BF = TADA_A_get_BFs(data = compact_data$base_info,</span><br><span class="line">               selected_annotations = c(1,2),</span><br><span class="line">               rr = RR$rr_report$logRR , </span><br><span class="line">#                additional_BF_file = c(&quot;/storage11_7T/fuy/TADA-A/annotation/cd_BF/Example_gene_coding_BF.txt&quot;), </span><br><span class="line">               additional_BF_file = c(&quot;/storage11_7T/fuy/TADA-A/annotation/cd_BF/uniform_cd_BFs.txt&quot;),</span><br><span class="line">               TADA_p0 = 0.94)</span><br><span class="line"></span><br><span class="line">g_BF = BF$gene_BF_table</span><br><span class="line">g_BF = g_BF[order(g_BF$FDR_all),]</span><br><span class="line">g_BF2 = g_BF[g_BF$FDR_all &lt;= 0.1,]</span><br><span class="line">proc.time() - tm</span><br><span class="line">nrow(g_BF2)</span><br><span class="line"></span><br><span class="line">g_BF2</span><br></pre></td></tr></table></figure>

<pre><code>[1] &quot;Read in additional BF file /storage11_7T/fuy/TADA-A/annotation/cd_BF/uniform_cd_BFs.txt.&quot;
[1] &quot;Flagged genes that don&apos;t have any bases with any informative annotation.&quot;
[1] &quot;Got genenames without bases that have informative rr features.&quot;
[1] &quot;Got logBF_noncoding.&quot;
[1] &quot;Added coding and non-coding logBF. Time consumed: 2.43s.&quot;
[1] &quot;Finished!&quot;



   user  system elapsed 
  4.680   0.000   3.961 </code></pre><p>38</p>
<table>
<caption>A data.table: 38 × 9</caption>
<thead>
    <tr><th scope="col">genename</th><th scope="col">logBF_noncoding</th><th scope="col">logBF_coding</th><th scope="col">logBF_all</th><th scope="col">BF_noncoding</th><th scope="col">BF_coding</th><th scope="col">BF_all</th><th scope="col">FDR_coding</th><th scope="col">FDR_all</th></tr>
    <tr><th scope="col">&lt;chr&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>SCN2A  </td><td>37.101469</td><td>0</td><td>37.101469</td><td>12970700178198198.00000</td><td>1</td><td>12970700178198198.00000</td><td>0.94</td><td>0.000000000000001332268</td></tr>
    <tr><td>SYNGAP1</td><td>15.352997</td><td>0</td><td>15.352997</td><td>          4652881.02203</td><td>1</td><td>          4652881.02203</td><td>0.94</td><td>0.000001683539064045725</td></tr>
    <tr><td>SLC6A1 </td><td>15.233880</td><td>0</td><td>15.233880</td><td>          4130378.88914</td><td>1</td><td>          4130378.88914</td><td>0.94</td><td>0.000002386699174961038</td></tr>
    <tr><td>CHD8   </td><td>13.099513</td><td>0</td><td>13.099513</td><td>           488704.38588</td><td>1</td><td>           488704.38588</td><td>0.94</td><td>0.000009804155674308923</td></tr>
    <tr><td>FOXP1  </td><td>12.164281</td><td>0</td><td>12.164281</td><td>           191813.91816</td><td>1</td><td>           191813.91816</td><td>0.94</td><td>0.000024177266648206518</td></tr>
    <tr><td>ARID1B </td><td>11.302438</td><td>0</td><td>11.302438</td><td>            81018.92716</td><td>1</td><td>            81018.92716</td><td>0.94</td><td>0.000052369900264119064</td></tr>
    <tr><td>GRIN2B </td><td>10.929415</td><td>0</td><td>10.929415</td><td>            55793.62976</td><td>1</td><td>            55793.62976</td><td>0.94</td><td>0.000084991038410975328</td></tr>
    <tr><td>KCNQ3  </td><td> 8.072257</td><td>0</td><td> 8.072257</td><td>             3204.32704</td><td>1</td><td>             3204.32704</td><td>0.94</td><td>0.000682546401555919613</td></tr>
    <tr><td>CHD2   </td><td> 7.866585</td><td>0</td><td> 7.866585</td><td>             2608.64299</td><td>1</td><td>             2608.64299</td><td>0.94</td><td>0.001270021684717373162</td></tr>
    <tr><td>ANK2   </td><td> 7.163774</td><td>0</td><td> 7.163774</td><td>             1291.77753</td><td>1</td><td>             1291.77753</td><td>0.94</td><td>0.002341286079239701754</td></tr>
    <tr><td>PPP2R5D</td><td> 6.559665</td><td>0</td><td> 6.559665</td><td>              706.03501</td><td>1</td><td>              706.03501</td><td>0.94</td><td>0.004101892228727416559</td></tr>
    <tr><td>ASH1L  </td><td> 5.764653</td><td>0</td><td> 5.764653</td><td>              318.82834</td><td>1</td><td>              318.82834</td><td>0.94</td><td>0.007663132321820855138</td></tr>
    <tr><td>ADNP   </td><td> 5.764525</td><td>0</td><td> 5.764525</td><td>              318.78764</td><td>1</td><td>              318.78764</td><td>0.94</td><td>0.010676927765345780794</td></tr>
    <tr><td>TCF4   </td><td> 5.706055</td><td>0</td><td> 5.706055</td><td>              300.68257</td><td>1</td><td>              300.68257</td><td>0.94</td><td>0.013451670548019965118</td></tr>
    <tr><td>ATP1B1 </td><td> 5.694572</td><td>0</td><td> 5.694572</td><td>              297.24956</td><td>1</td><td>              297.24956</td><td>0.94</td><td>0.015892669089122110626</td></tr>
    <tr><td>DPYSL2 </td><td> 5.681399</td><td>0</td><td> 5.681399</td><td>              293.35958</td><td>1</td><td>              293.35958</td><td>0.94</td><td>0.018067932326401005272</td></tr>
    <tr><td>NOL6   </td><td> 5.452025</td><td>0</td><td> 5.452025</td><td>              233.22991</td><td>1</td><td>              233.22991</td><td>0.94</td><td>0.020707729431314048085</td></tr>
    <tr><td>NAA15  </td><td> 5.360727</td><td>0</td><td> 5.360727</td><td>              212.87970</td><td>1</td><td>              212.87970</td><td>0.94</td><td>0.023365587900061015464</td></tr>
    <tr><td>SETD5  </td><td> 5.321888</td><td>0</td><td> 5.321888</td><td>              204.77006</td><td>1</td><td>              204.77006</td><td>0.94</td><td>0.025876400936694275018</td></tr>
    <tr><td>MYT1L  </td><td> 5.297465</td><td>0</td><td> 5.297465</td><td>              199.82964</td><td>1</td><td>              199.82964</td><td>0.94</td><td>0.028217600588264175887</td></tr>
    <tr><td>STXBP1 </td><td> 5.166411</td><td>0</td><td> 5.166411</td><td>              175.28467</td><td>1</td><td>              175.28467</td><td>0.94</td><td>0.030780826196886448831</td></tr>
    <tr><td>TRIP12 </td><td> 5.148896</td><td>0</td><td> 5.148896</td><td>              172.24122</td><td>1</td><td>              172.24122</td><td>0.94</td><td>0.033171433409178947838</td></tr>
    <tr><td>NRXN1  </td><td> 5.003762</td><td>0</td><td> 5.003762</td><td>              148.97257</td><td>1</td><td>              148.97257</td><td>0.94</td><td>0.035866482133073071537</td></tr>
    <tr><td>NF1    </td><td> 4.893366</td><td>0</td><td> 4.893366</td><td>              133.40180</td><td>1</td><td>              133.40180</td><td>0.94</td><td>0.038751092047992832801</td></tr>
    <tr><td>WDFY3  </td><td> 4.615404</td><td>0</td><td> 4.615404</td><td>              101.02861</td><td>1</td><td>              101.02861</td><td>0.94</td><td>0.042571160147908564120</td></tr>
    <tr><td>DEAF1  </td><td> 4.515042</td><td>0</td><td> 4.515042</td><td>               91.38144</td><td>1</td><td>               91.38144</td><td>0.94</td><td>0.046562717225392964704</td></tr>
    <tr><td>RORB   </td><td> 4.428253</td><td>0</td><td> 4.428253</td><td>               83.78488</td><td>1</td><td>               83.78488</td><td>0.94</td><td>0.050672640423468987003</td></tr>
    <tr><td>KDM5B  </td><td> 4.416790</td><td>0</td><td> 4.416790</td><td>               82.83001</td><td>1</td><td>               82.83001</td><td>0.94</td><td>0.054543539671785157275</td></tr>
    <tr><td>MED13L </td><td> 4.273211</td><td>0</td><td> 4.273211</td><td>               71.75168</td><td>1</td><td>               71.75168</td><td>0.94</td><td>0.058842550692057252193</td></tr>
    <tr><td>MKX    </td><td> 4.260844</td><td>0</td><td> 4.260844</td><td>               70.86975</td><td>1</td><td>               70.86975</td><td>0.94</td><td>0.062915843050658101787</td></tr>
    <tr><td>GNAI1  </td><td> 4.213925</td><td>0</td><td> 4.213925</td><td>               67.62140</td><td>1</td><td>               67.62140</td><td>0.94</td><td>0.066954111892309195464</td></tr>
    <tr><td>AP2S1  </td><td> 4.124807</td><td>0</td><td> 4.124807</td><td>               61.85586</td><td>1</td><td>               61.85586</td><td>0.94</td><td>0.071177164827275776759</td></tr>
    <tr><td>RFX3   </td><td> 4.091268</td><td>0</td><td> 4.091268</td><td>               59.81569</td><td>1</td><td>               59.81569</td><td>0.94</td><td>0.075309796636768094880</td></tr>
    <tr><td>SRPK2  </td><td> 4.005589</td><td>0</td><td> 4.005589</td><td>               54.90417</td><td>1</td><td>               54.90417</td><td>0.94</td><td>0.079624189959590058185</td></tr>
    <tr><td>GRIA2  </td><td> 3.918275</td><td>0</td><td> 3.918275</td><td>               50.31356</td><td>1</td><td>               50.31356</td><td>0.94</td><td>0.084133352626445542799</td></tr>
    <tr><td>ELAVL3 </td><td> 3.870546</td><td>0</td><td> 3.870546</td><td>               47.96856</td><td>1</td><td>               47.96856</td><td>0.94</td><td>0.088635061933261274913</td></tr>
    <tr><td>CORO1A </td><td> 3.707061</td><td>0</td><td> 3.707061</td><td>               40.73392</td><td>1</td><td>               40.73392</td><td>0.94</td><td>0.093746949348747790620</td></tr>
    <tr><td>TRAF7  </td><td> 3.694830</td><td>0</td><td> 3.694830</td><td>               40.23874</td><td>1</td><td>               40.23874</td><td>0.94</td><td>0.098654536725587771429</td></tr>
</tbody>
</table>




<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
</div><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">Author: </span><span class="post-copyright-info"><a href="mailto:undefined" target="_blank" rel="noopener"></a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">Link: </span><span class="post-copyright-info"><a href="https://yfu1116.github.io/project/1119-12166DNM-sf-6430-fam/">https://yfu1116.github.io/project/1119-12166DNM-sf-6430-fam/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">Copyright Notice: </span><span class="post-copyright-info">All articles in this blog are licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank" rel="noopener">CC BY-NC-SA 4.0</a> unless stating additionally.</span></div></div><div class="tag_share"><div class="post-meta__tag-list"></div><div class="post_share"><div class="social-share" data-image="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" data-sites="github,email,facebook,twitter,weixin,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js/dist/css/share.min.css"><script src="https://cdn.jsdelivr.net/npm/social-share.js/dist/js/social-share.min.js" defer></script></div></div><nav class="pagination-post" id="pagination"><div class="prev-post pull-left"><a href="/project/201126-tadaA-report/"><img class="prev-cover" data-lazy-src="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" onerror="onerror=null;src='/project/img/404.jpg'"><div class="pagination-info"><div class="label">Previous Post</div><div class="prev_info">201126-tadaA-report</div></div></a></div><div class="next-post pull-right"><a href="/project/201117-tadaA-report/"><img class="next-cover" data-lazy-src="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" onerror="onerror=null;src='/project/img/404.jpg'"><div class="pagination-info"><div class="label">Next Post</div><div class="next_info">201117-tadaA-report</div></div></a></div></nav></article></main><footer id="footer" data-type="color"><div id="footer-wrap"><div class="copyright">&copy;2020 By </div><div class="framework-info"><span>Framework </span><a href="https://hexo.io" target="_blank" rel="noopener">Hexo</a><span class="footer-separator">|</span><span>Theme </span><a href="https://github.com/jerryc127/hexo-theme-butterfly" target="_blank" rel="noopener">Butterfly</a></div></div></footer></div><section id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="Read Mode"><i class="fas fa-book-open"></i></button><button id="font_plus" type="button" title="Increase Font Size"><i class="fas fa-plus"></i></button><button id="font_minus" type="button" title="Decrease Font Size"><i class="fas fa-minus"></i></button><button id="darkmode" type="button" title="Switch Between Light And Dark Mode"><i class="fas fa-adjust"></i></button></div><div id="rightside-config-show"><button id="rightside_config" type="button" title="Setting"><i class="fas fa-cog"></i></button><button class="close" id="mobile-toc-button" type="button" title="Table Of Contents"><i class="fas fa-list-ul"></i></button><button id="go-up" type="button" title="Back To Top"><i class="fas fa-arrow-up"></i></button></div></section><div><script src="https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js"></script><script src="/project/js/utils.js"></script><script src="/project/js/main.js"></script><script src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.js"></script><script src="https://cdn.jsdelivr.net/npm/instant.page/instantpage.min.js" type="module" defer></script><script src="https://cdn.jsdelivr.net/npm/vanilla-lazyload/dist/lazyload.iife.min.js" async></script><div class="js-pjax"><script>if (!window.MathJax) {
  window.MathJax = {
    loader: {
      source: {
        '[tex]/amsCd': '[tex]/amscd'
      }
    },
    tex: {
      inlineMath: [ ['$','$'], ["\\(","\\)"]],
      tags: 'ams'
    },
    options: {
      renderActions: {
        findScript: [10, doc => {
          for (const node of document.querySelectorAll('script[type^="math/tex"]')) {
            const display = !!node.type.match(/; *mode=display/)
            const math = new doc.options.MathItem(node.textContent, doc.inputJax[0], display)
            const text = document.createTextNode('')
            node.parentNode.replaceChild(text, node)
            math.start = {node: text, delim: '', n: 0}
            math.end = {node: text, delim: '', n: 0}
            doc.math.push(math)
          }
        }, '']
      }
    }
  }
  
  var script = document.createElement('script')
  script.src = 'https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js'
  script.id = 'MathJax-script'
  script.async = true
  document.head.appendChild(script)
} else {
  MathJax.startup.document.state(0)
  MathJax.texReset()
  MathJax.typeset()
}</script><script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div></div></body></html>