<!DOCTYPE html><html lang="en" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1"><title>1104-tadaA-simulate-6430-rr036 | yang's blog</title><meta name="description" content="12345setwd(&quot;&#x2F;storage11_7T&#x2F;fuy&#x2F;TADA-A&#x2F;TADA-A&#x2F;lib&quot;)library(data.table)library(doParallel)library(foreach)library(data.table)  Loading required package: foreach  Loading required package: itera"><meta name="author" content=""><meta name="copyright" content=""><meta name="format-detection" content="telephone=no"><link rel="shortcut icon" href="/project/img/favicon.png"><link rel="canonical" href="https://yfu1116.github.io/project/1104-tadaA-simulate-6430-rr036/"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//fonts.googleapis.com" crossorigin="crossorigin"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><meta property="og:type" content="article"><meta property="og:title" content="1104-tadaA-simulate-6430-rr036"><meta property="og:url" content="https://yfu1116.github.io/project/1104-tadaA-simulate-6430-rr036/"><meta property="og:site_name" content="yang's blog"><meta property="og:description" content="12345setwd(&quot;&#x2F;storage11_7T&#x2F;fuy&#x2F;TADA-A&#x2F;TADA-A&#x2F;lib&quot;)library(data.table)library(doParallel)library(foreach)library(data.table)  Loading required package: foreach  Loading required package: itera"><meta property="og:image" content="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg"><meta property="article:published_time" content="2020-11-05T02:31:41.736Z"><meta property="article:modified_time" content="2020-11-05T02:32:33.303Z"><meta name="twitter:card" content="summary"><script>var activateDarkMode = function () {
  document.documentElement.setAttribute('data-theme', 'dark')
  if (document.querySelector('meta[name="theme-color"]') !== null) {
    document.querySelector('meta[name="theme-color"]').setAttribute('content', '#000')
  }
}
var activateLightMode = function () {
  document.documentElement.setAttribute('data-theme', 'light')
  if (document.querySelector('meta[name="theme-color"]') !== null) {
    document.querySelector('meta[name="theme-color"]').setAttribute('content', '#fff')
  }
}

var getCookies = function (name) {
  const value = `; ${document.cookie}`
  const parts = value.split(`; ${name}=`)
  if (parts.length === 2) return parts.pop().split(';').shift()
}

var autoChangeMode = 'false'
var t = getCookies('theme')
if (autoChangeMode === '1') {
  var isDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches
  var isLightMode = window.matchMedia('(prefers-color-scheme: light)').matches
  var isNotSpecified = window.matchMedia('(prefers-color-scheme: no-preference)').matches
  var hasNoSupport = !isDarkMode && !isLightMode && !isNotSpecified

  if (t === undefined) {
    if (isLightMode) activateLightMode()
    else if (isDarkMode) activateDarkMode()
    else if (isNotSpecified || hasNoSupport) {
      console.log('You specified no preference for a color scheme or your browser does not support it. I Schedule dark mode during night time.')
      var now = new Date()
      var hour = now.getHours()
      var isNight = hour <= 6 || hour >= 18
      isNight ? activateDarkMode() : activateLightMode()
    }
    window.matchMedia('(prefers-color-scheme: dark)').addListener(function (e) {
      if (Cookies.get('theme') === undefined) {
        e.matches ? activateDarkMode() : activateLightMode()
      }
    })
  } else if (t === 'light') activateLightMode()
  else activateDarkMode()
} else if (autoChangeMode === '2') {
  now = new Date()
  hour = now.getHours()
  isNight = hour <= 6 || hour >= 18
  if (t === undefined) isNight ? activateDarkMode() : activateLightMode()
  else if (t === 'light') activateLightMode()
  else activateDarkMode()
} else {
  if (t === 'dark') activateDarkMode()
  else if (t === 'light') activateLightMode()
}</script><link rel="stylesheet" href="/project/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.css"><link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Titillium+Web&amp;display=swap"><script>var GLOBAL_CONFIG = { 
  root: '/project/',
  hexoversion: '4.0.0',
  algolia: undefined,
  localSearch: undefined,
  translate: undefined,
  noticeOutdate: undefined,
  copy: {
    success: 'Copy successfully',
    error: 'Copy error',
    noSupport: 'The browser does not support'
  },
  bookmark: {
    message_prev: 'Press',
    message_next: 'to bookmark this page'
  },
  runtime_unit: 'days',
  runtime: false,
  copyright: undefined,
  ClickShowText: undefined,
  medium_zoom: false,
  fancybox: true,
  Snackbar: undefined,
  justifiedGallery: {
    js: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/js/jquery.justifiedGallery.min.js',
    css: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/css/justifiedGallery.min.css'
  },
  baiduPush: false,
  highlightCopy: true,
  highlightLang: true,
  isPhotoFigcaption: false,
  islazyload: true,
  isanchor: false    
}</script><script id="config_change">var GLOBAL_CONFIG_SITE = { 
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isSidebar: false,
  postUpdate: '2020-11-05 10:32:33'
}</script><noscript><style>
#nav {
  opacity: 1
}
.justified-gallery img {
  opacity: 1
}
</style></noscript></head><body><div id="mobile-sidebar"><div id="menu_mask"></div><div id="mobile-sidebar-menus"><div class="mobile_author_icon"><img class="avatar-img" data-lazy-src="/project/img/avatar.png" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="mobile_post_data"><div class="mobile_data_item is-center"><div class="mobile_data_link"><a href="/project/archives/"><div class="headline">Articles</div><div class="length_num">64</div></a></div></div></div><hr/></div></div><div id="body-wrap"><header class="post-bg" id="page-header" style="background-image: url(https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg)"><nav id="nav"><span class="pull-left" id="blog_name"><a class="blog_title" id="site-name" href="/project/">yang's blog</a></span><span class="pull-right menus"><span class="toggle-menu close"><a class="site-page"><i class="fas fa-bars fa-fw"></i></a></span></span></nav><div id="post-info"><div id="post-title"><div class="posttitle">1104-tadaA-simulate-6430-rr036</div></div><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">Created</span><time class="post-meta-date-created" datetime="2020-11-05T02:31:41.736Z" title="Created 2020-11-05 10:31:41">2020-11-05</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">Updated</span><time class="post-meta-date-updated" datetime="2020-11-05T02:32:33.303Z" title="Updated 2020-11-05 10:32:33">2020-11-05</time></span></div><div class="meta-secondline"> <span class="post-meta-separator">|</span><span class="post-meta-pv-cv"><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">Post View:</span><span id="busuanzi_value_page_pv"></span></span></div></div></div></header><main class="layout_post" id="content-inner"><article id="post"><div class="post-content" id="article-container"><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">setwd(&quot;/storage11_7T/fuy/TADA-A/TADA-A/lib&quot;)</span><br><span class="line">library(data.table)</span><br><span class="line">library(doParallel)</span><br><span class="line">library(foreach)</span><br><span class="line">library(data.table)</span><br></pre></td></tr></table></figure>

<pre><code>Loading required package: foreach

Loading required package: iterators

Loading required package: parallel</code></pre><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">f = fread(&quot;/storage11_7T/fuy/TADA-A/cell_WES/cd_mis_pli_syn.6954DNM_6430fam_scaling_factors.txt&quot;)</span><br><span class="line">colnames(f)[2] = &quot;z&quot;</span><br><span class="line">f$scaling_factor = rep(1,nrow(f))</span><br><span class="line">head(f,2)</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.table: 2 Ã— 3</caption>
<thead>
    <tr><th scope="col">site_index</th><th scope="col">z</th><th scope="col">scaling_factor</th></tr>
    <tr><th scope="col">&lt;int&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>6408983</td><td>1.300119</td><td>1</td></tr>
    <tr><td>6408984</td><td>1.463074</td><td>1</td></tr>
</tbody>
</table>




<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">fwrite(f[,c(1,3)],&quot;/storage11_7T/fuy/TADA-A/cell_WES/cd_uniform_scaling_factors.txt&quot;,sep=&quot;\t&quot;)</span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line">a = TADA_A_DNM_generator(window_file =  &apos;/storage11_7T/fuy/TADA-A/db/MS_data/windows_partition/cd_windows_with_div_score.bed&apos;,</span><br><span class="line">                                 mutrate_scaling_files = c(&quot;/storage11_7T/fuy/TADA-A/cell_WES/cd_uniform_scaling_factors.txt&quot;),</span><br><span class="line">                                 sample_sizes = 6430,</span><br><span class="line">                                 gene_prior_file = &quot;/storage11_7T/fuy/TADA-A/db/MS_data/prior/uniform_gene_prior.txt&quot;,</span><br><span class="line">                                 nonAS_noncoding_annotations = NA,</span><br><span class="line">                                AS_noncoding_annotations = list(</span><br><span class="line">                              c(&quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/all_cd_syn_alt_A.bed&quot;,</span><br><span class="line">                                &quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/all_cd_syn_alt_C.bed&quot;,</span><br><span class="line">                                &quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/all_cd_syn_alt_G.bed&quot;,</span><br><span class="line">                                &quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/all_cd_syn_alt_T.bed&quot;)),</span><br><span class="line"> </span><br><span class="line">                                 report_proportion = 1,</span><br><span class="line">                                 chunk_partition_num =2,</span><br><span class="line">                                 node_n = 2,</span><br><span class="line">                    mutrate_ref_files = c(</span><br><span class="line">       &quot;/storage11_7T/fuy/TADA-A/db/MS_data/mutrate/Example_windows_extended_1bp_for_getting_base_level_mutrate.bed.fasta.tri.alt_A.mutrate.bw&quot;,</span><br><span class="line">       &quot;/storage11_7T/fuy/TADA-A/db/MS_data/mutrate/Example_windows_extended_1bp_for_getting_base_level_mutrate.bed.fasta.tri.alt_C.mutrate.bw&quot;,</span><br><span class="line">       &quot;/storage11_7T/fuy/TADA-A/db/MS_data/mutrate/Example_windows_extended_1bp_for_getting_base_level_mutrate.bed.fasta.tri.alt_G.mutrate.bw&quot;,</span><br><span class="line">       &quot;/storage11_7T/fuy/TADA-A/db/MS_data/mutrate/Example_windows_extended_1bp_for_getting_base_level_mutrate.bed.fasta.tri.alt_T.mutrate.bw&quot;),                                 </span><br><span class="line">                                    rr = 0,</span><br><span class="line">                                 output_allele_info_files = &quot;2allele.bed&quot;,</span><br><span class="line">                                 output_bed_files = &quot;2.bed&quot;,</span><br><span class="line">                                 output_risk_genes_file = &quot;temp.riskgenes&quot;,</span><br><span class="line">                                 compact_mut_output = NA,</span><br><span class="line">                                 MPI = 1)</span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br><span class="line">155</span><br><span class="line">156</span><br><span class="line">157</span><br><span class="line">158</span><br><span class="line">159</span><br><span class="line">160</span><br><span class="line">161</span><br><span class="line">162</span><br><span class="line">163</span><br><span class="line">164</span><br><span class="line">165</span><br><span class="line">166</span><br><span class="line">167</span><br><span class="line">168</span><br><span class="line">169</span><br><span class="line">170</span><br><span class="line">171</span><br><span class="line">172</span><br><span class="line">173</span><br><span class="line">174</span><br><span class="line">175</span><br><span class="line">176</span><br><span class="line">177</span><br><span class="line">178</span><br><span class="line">179</span><br><span class="line">180</span><br><span class="line">181</span><br><span class="line">182</span><br><span class="line">183</span><br><span class="line">184</span><br><span class="line">185</span><br><span class="line">186</span><br><span class="line">187</span><br><span class="line">188</span><br><span class="line">189</span><br><span class="line">190</span><br><span class="line">191</span><br><span class="line">192</span><br><span class="line">193</span><br><span class="line">194</span><br><span class="line">195</span><br><span class="line">196</span><br><span class="line">197</span><br><span class="line">198</span><br><span class="line">199</span><br><span class="line">200</span><br><span class="line">201</span><br><span class="line">202</span><br><span class="line">203</span><br><span class="line">204</span><br><span class="line">205</span><br><span class="line">206</span><br><span class="line">207</span><br><span class="line">208</span><br><span class="line">209</span><br><span class="line">210</span><br><span class="line">211</span><br><span class="line">212</span><br><span class="line">213</span><br><span class="line">214</span><br><span class="line">215</span><br><span class="line">216</span><br><span class="line">217</span><br><span class="line">218</span><br><span class="line">219</span><br><span class="line">220</span><br><span class="line">221</span><br><span class="line">222</span><br><span class="line">223</span><br><span class="line">224</span><br><span class="line">225</span><br><span class="line">226</span><br><span class="line">227</span><br><span class="line">228</span><br><span class="line">229</span><br><span class="line">230</span><br><span class="line">231</span><br><span class="line">232</span><br><span class="line">233</span><br><span class="line">234</span><br><span class="line">235</span><br><span class="line">236</span><br><span class="line">237</span><br><span class="line">238</span><br><span class="line">239</span><br><span class="line">240</span><br><span class="line">241</span><br><span class="line">242</span><br><span class="line">243</span><br><span class="line">244</span><br><span class="line">245</span><br><span class="line">246</span><br><span class="line">247</span><br><span class="line">248</span><br><span class="line">249</span><br><span class="line">250</span><br><span class="line">251</span><br><span class="line">252</span><br><span class="line">253</span><br><span class="line">254</span><br><span class="line">255</span><br><span class="line">256</span><br><span class="line">257</span><br><span class="line">258</span><br><span class="line">259</span><br><span class="line">260</span><br><span class="line">261</span><br><span class="line">262</span><br><span class="line">263</span><br><span class="line">264</span><br><span class="line">265</span><br><span class="line">266</span><br><span class="line">267</span><br><span class="line">268</span><br><span class="line">269</span><br><span class="line">270</span><br><span class="line">271</span><br><span class="line">272</span><br><span class="line">273</span><br><span class="line">274</span><br><span class="line">275</span><br></pre></td><td class="code"><pre><span class="line">TADA_A_DNM_generator &lt;- function(window_file = &quot;../data/Example_windows.bed&quot;,</span><br><span class="line">                                 mutrate_scaling_files = c(&quot;../data/Example_windows_mutrate_scaling_file_for_Yuen_NM2015_cases_DNM.txt&quot;,&quot;../data/Example_windows_mutrate_scaling_file_for_Kong_cases_DNM.txt&quot;, &quot;../data/Example_windows_mutrate_scaling_file_for_Wu_cases_DNM.txt&quot;, &quot;../data/Example_windows_mutrate_scaling_file_for_Jiang_cases_DNM.txt&quot;, &quot;../data/Example_windows_mutrate_scaling_file_for_Michaelson_cases_DNM.txt&quot;),</span><br><span class="line">                                 sample_sizes = c(162, 78, 32, 32, 10),</span><br><span class="line">                                 gene_prior_file = &quot;../data/Example_gene_prior.txt&quot;,</span><br><span class="line">                                 nonAS_noncoding_annotations = c(&quot;../data/Noonan_brain_roadmap_union_within_10kb_and_promoter_no_utr.bed&quot;,&quot;../data/Epigenome_E081_E082_intersection__within_10kb_and_promoter_no_utr.bed&quot;,&quot;../data/Encode_DHS_union_within_10kb_and_promoter_no_utr.bed&quot;),</span><br><span class="line">                                 AS_noncoding_annotations = NA,</span><br><span class="line">                                 report_proportion = 100/18665,</span><br><span class="line">                                 chunk_partition_num =1 ,</span><br><span class="line">                                 node_n = 6,</span><br><span class="line">                                 mutrate_ref_files = c(&quot;../other_annotations/Mark_Daly_mutrate/Example_windows_extended_1bp_for_getting_base_level_mutrate.bed.fasta.tri.alt_A.mutrate.bw&quot;,</span><br><span class="line">                                                       &quot;../other_annotations/Mark_Daly_mutrate/Example_windows_extended_1bp_for_getting_base_level_mutrate.bed.fasta.tri.alt_C.mutrate.bw&quot;,</span><br><span class="line">                                                       &quot;../other_annotations/Mark_Daly_mutrate/Example_windows_extended_1bp_for_getting_base_level_mutrate.bed.fasta.tri.alt_G.mutrate.bw&quot;,</span><br><span class="line">                                                       &quot;../other_annotations/Mark_Daly_mutrate/Example_windows_extended_1bp_for_getting_base_level_mutrate.bed.fasta.tri.alt_T.mutrate.bw&quot;),</span><br><span class="line">                                 rr = c(1, 0, 0.5),</span><br><span class="line">                                 output_allele_info_files,</span><br><span class="line">                                 output_bed_files,</span><br><span class="line">                                 output_risk_genes_file = &quot;temp.riskgenes&quot;,</span><br><span class="line">                                 compact_mut_output = NA,</span><br><span class="line">                                 MPI = 1)&#123;</span><br><span class="line">  </span><br><span class="line">  prefix &lt;- prefix &lt;- as.integer((as.double(Sys.time())*1000+Sys.getpid()) %% 2^31) # prefix for temporary files that will be deleted at the end of the pipeline</span><br><span class="line">  prefix &lt;- paste(&quot;tmp/&quot;, prefix, MPI, sep = &quot;&quot;)</span><br><span class="line">  </span><br><span class="line">  # make a tmp folder for tmp files</span><br><span class="line">  system(&quot;mkdir -p tmp&quot;)</span><br><span class="line">  </span><br><span class="line">  #mut &lt;- fread(mut_file, header = FALSE, sep = &quot;\t&quot;, stringsAsFactors = FALSE)</span><br><span class="line">  windows &lt;- fread(window_file, header = TRUE, sep = &quot;\t&quot;, stringsAsFactors = FALSE)</span><br><span class="line">  coverage &lt;- windows</span><br><span class="line"></span><br><span class="line">  # the number of genomic windows in [mutrate_scaling] is less than the number of windows in [windows] because there are a few windows with mutration rate equal to 0, and thus removed.</span><br><span class="line">  for(i in 1:length(mutrate_scaling_files))&#123;</span><br><span class="line">    mutrate_scaling &lt;- fread(mutrate_scaling_files[i], header = TRUE, sep = &quot;\t&quot;, stringsAsFactors = FALSE)</span><br><span class="line">    system(paste(&quot;echo \&quot;Finished reading mutrate scaling file &quot;, mutrate_scaling_files[i], &quot;.\&quot;&quot;, sep = &quot;&quot;))</span><br><span class="line">    system(&quot;date&quot;)</span><br><span class="line">    coverage &lt;- coverage[mutrate_scaling, on = &quot;site_index&quot;]</span><br><span class="line">    coverage &lt;- coverage[complete.cases(coverage)] # release memory</span><br><span class="line">    colnames(coverage)[length(colnames(coverage))] &lt;- paste(&quot;scaling_factor_study_&quot;, i, sep = &quot;&quot;)</span><br><span class="line">    rm(mutrate_scaling) # release memory</span><br><span class="line">  &#125;</span><br><span class="line">  </span><br><span class="line">  gene_prior &lt;- fread(gene_prior_file, header = TRUE, sep = &quot;\t&quot;, stringsAsFactors = FALSE)</span><br><span class="line">  </span><br><span class="line">  # merge gene prior info</span><br><span class="line">  coverage &lt;- coverage[gene_prior, on = &quot;genename&quot;]</span><br><span class="line">  coverage &lt;-coverage[complete.cases(coverage)]</span><br><span class="line">  </span><br><span class="line">  # select genes based on TADA prior probability and [report_proportion]</span><br><span class="line">  genes_for_report_with_prior &lt;- gene_prior[order(gene_prior[,2],decreasing = TRUE),]</span><br><span class="line">  genes_for_report_with_prior &lt;- genes_for_report_with_prior[1:floor(nrow(genes_for_report_with_prior)*report_proportion),]</span><br><span class="line">  # randomly assign risk genes based on prior probability</span><br><span class="line">  genes_for_report_with_prior$risk &lt;- rbinom(nrow(genes_for_report_with_prior), 1, genes_for_report_with_prior$prior)</span><br><span class="line">  risk_genes &lt;- genes_for_report_with_prior[risk == 1,]$genename</span><br><span class="line">  nonrisk_genes &lt;- genes_for_report_with_prior[risk == 0,]$genename</span><br><span class="line">  coverage_risk &lt;- coverage[is.element(genename, risk_genes)]</span><br><span class="line">  coverage_risk$risk &lt;- 1</span><br><span class="line">  coverage_nonrisk &lt;- coverage[is.element(genename, nonrisk_genes)]</span><br><span class="line">  coverage_nonrisk$risk &lt;- 0 </span><br><span class="line">  </span><br><span class="line">  coverage &lt;- rbind(coverage_risk, coverage_nonrisk)</span><br><span class="line">  </span><br><span class="line">  # now need to extropolate window mutation file to base level mutation file</span><br><span class="line">  coverage &lt;-coverage[,c(&quot;chr&quot;,&quot;start&quot;,&quot;end&quot;,paste(&quot;scaling_factor_study_&quot;, seq(1,length(mutrate_scaling_files)), sep = &quot;&quot;),&quot;genename&quot;,&quot;risk&quot;),with = FALSE]</span><br><span class="line">  </span><br><span class="line">  coverage$ID &lt;- paste(coverage$genename, coverage$start, sep = &quot;_&quot;)</span><br><span class="line">  </span><br><span class="line">  total_rows &lt;- nrow(coverage)</span><br><span class="line">  interval &lt;- floor(total_rows/chunk_partition_num)</span><br><span class="line">  data_bins &lt;- c(rep(seq(1,chunk_partition_num), each = interval),rep(chunk_partition_num, total_rows -interval*chunk_partition_num))</span><br><span class="line">  </span><br><span class="line">  # split into 20 different chunks, then for each chunk split by genes, then by feature configuration. If split by gene at the first level, implementation would take too long</span><br><span class="line">  coverage &lt;- split(coverage, data_bins)</span><br><span class="line">  </span><br><span class="line">  #funtion to expand windows to bases</span><br><span class="line">  window_expansion &lt;- function(table_row)&#123;</span><br><span class="line">    start &lt;- seq(as.numeric(table_row[2]),as.numeric(table_row[3])-1)</span><br><span class="line">    data.frame(table_row[1], start, start+1, paste(table_row[&quot;genename&quot;],start,sep = &quot;_&quot;), table_row[&quot;ID&quot;])</span><br><span class="line">  &#125;</span><br><span class="line">  </span><br><span class="line">  </span><br><span class="line">  bed_extension &lt;- function(table_row)&#123;</span><br><span class="line">    data.frame(chr = rep(table_row[&quot;chr&quot;], table_row[&quot;mut_count&quot;]), start = rep(table_row[&quot;start&quot;], table_row[&quot;mut_count&quot;]), end = rep(table_row[&quot;end&quot;], table_row[&quot;mut_count&quot;]))</span><br><span class="line">  &#125;</span><br><span class="line">  </span><br><span class="line">  options(warn=-1)</span><br><span class="line">  if(node_n != 1)&#123;</span><br><span class="line">    cl &lt;- makeCluster(node_n)</span><br><span class="line">  &#125;</span><br><span class="line">  # use parallel computing and rbindlist to increase the speed by thousands of times. </span><br><span class="line">  environment(window_expansion) &lt;- .GlobalEnv</span><br><span class="line">  environment(bed_extension) &lt;- .GlobalEnv</span><br><span class="line">  # get nonAS feature number</span><br><span class="line">  if(is.na(nonAS_noncoding_annotations[1]))&#123;</span><br><span class="line">    nonAS_feature_number &lt;- 0 </span><br><span class="line">  &#125;else&#123;</span><br><span class="line">    nonAS_feature_number &lt;- length(nonAS_noncoding_annotations)</span><br><span class="line">  &#125;</span><br><span class="line">  # get AS feature number</span><br><span class="line">  if(is.na(AS_noncoding_annotations[1]))&#123;</span><br><span class="line">    AS_feature_number &lt;- 0 </span><br><span class="line">  &#125;else&#123;</span><br><span class="line">    AS_feature_number &lt;- length(AS_noncoding_annotations)</span><br><span class="line">  &#125;</span><br><span class="line">  </span><br><span class="line">  # get total feature number</span><br><span class="line">  feature_number = nonAS_feature_number + AS_feature_number</span><br><span class="line">  # function to get effective information of each element of partition_by_gene</span><br><span class="line">  # These information are those necessary to compute log-likelihood in the optimization function</span><br><span class="line">  partition_feature &lt;- function(pbg)&#123;</span><br><span class="line">    # input is one element of the list of partition_by_gene</span><br><span class="line">    pbg_split &lt;- split(pbg, pbg[,4:(4 + feature_number - 1)],drop = TRUE)</span><br><span class="line">    feature_combination_number &lt;- length(pbg_split)</span><br><span class="line">    # this function below is different from the function used in dealing with dataset without reading by chunk. Here, prior is not incoporated at this step.</span><br><span class="line">    info_for_each_feature &lt;- function(feature_set)&#123;</span><br><span class="line">      list(feature_vector = c(as.numeric(feature_set[1,4:(4 + feature_number - 1)])), sum_mut_rate_count = sum(feature_set$mut_count*log(feature_set$adjusted_base_mutrate)), sum_mut_rate = sum(feature_set$adjusted_base_mutrate), sum_mut_count = sum(feature_set$mut_count), log_fcount = sum(log(factorial(feature_set$mut_count))))</span><br><span class="line">    &#125;</span><br><span class="line">    sapply(pbg_split, info_for_each_feature,simplify = FALSE)</span><br><span class="line">  &#125;</span><br><span class="line">  </span><br><span class="line">  </span><br><span class="line">  alt_letters &lt;- c(&quot;A&quot;,&quot;C&quot;,&quot;G&quot;,&quot;T&quot;)</span><br><span class="line">  mut_output_list &lt;- list()</span><br><span class="line">  data_partition &lt;- list()</span><br><span class="line">  for(m in 1:length(mutrate_scaling_files))&#123;</span><br><span class="line">    mut_output_list[[m]] &lt;- data.table(chr = character(), start = integer(), end = integer(), ref = character(), alt = character())</span><br><span class="line">  &#125;</span><br><span class="line">  </span><br><span class="line">  # here only deals with the noncoding parts that are within 10kb of TSSs of genes. Will deal with </span><br><span class="line">  for(i in 1:chunk_partition_num)&#123;</span><br><span class="line">    # split coverage_noncoding to 10 parts, and for each part, generate feature table (which will be used for )</span><br><span class="line">    if(node_n != 1)&#123;</span><br><span class="line">      coverage_noncoding_for_base_mutrate &lt;- rbindlist(parApply(cl, coverage[[i]], 1, window_expansion))</span><br><span class="line">    &#125;else&#123;</span><br><span class="line">      coverage_noncoding_for_base_mutrate &lt;- rbindlist(apply(coverage[[i]], 1, window_expansion))</span><br><span class="line">    &#125;</span><br><span class="line">    system(paste(&quot;echo \&quot;Finished partitioning base-level coordinates data at Round &quot;, i, &quot;.\&quot;&quot;, sep = &quot;&quot;))</span><br><span class="line">    system(&quot;date&quot;)</span><br><span class="line">    colnames(coverage_noncoding_for_base_mutrate) &lt;- c(&quot;chr&quot;,&quot;start&quot;,&quot;end&quot;,&quot;base_ID&quot;,&quot;ID&quot;)</span><br><span class="line">    coverage_noncoding_for_base_mutrate &lt;- coverage_noncoding_for_base_mutrate[!duplicated(base_ID)]</span><br><span class="line">    coverage_noncoding_for_base_mutrate$start &lt;- as.integer(coverage_noncoding_for_base_mutrate$start)</span><br><span class="line">    coverage_noncoding_for_base_mutrate$end &lt;- as.integer(coverage_noncoding_for_base_mutrate$end)</span><br><span class="line">    </span><br><span class="line">    # write out a bed file to get base-level mutation rates</span><br><span class="line">    fwrite(coverage_noncoding_for_base_mutrate[,1:4],paste(prefix, &quot;_temp_for_mutrate.bed&quot;, sep = &quot;&quot;), col.names = FALSE, row.names = FALSE, sep = &quot;\t&quot;, quote = FALSE)</span><br><span class="line">    # read in allele-specific base-level mutation rate</span><br><span class="line">    for(j in 1:length(mutrate_ref_files))&#123;</span><br><span class="line">      command &lt;- paste(&quot;../external_tools/bigWigAverageOverBed &quot;, mutrate_ref_files[j], &quot; &quot;, paste(prefix, &quot;_temp_for_mutrate.bed&quot;, sep = &quot;&quot;), &quot; &quot;, paste(prefix, &quot;_temp_for_mutrate.bed.mutrate&quot;, sep = &quot;&quot; ), sep = &quot;&quot;)</span><br><span class="line">      system(command)</span><br><span class="line">      command &lt;- paste(&quot;awk &#123;&apos;print $1\&quot;\t\&quot;$4&apos;&#125; &quot;, paste(prefix, &quot;_temp_for_mutrate.bed.mutrate&quot;, sep = &quot;&quot;), &quot; &gt; &quot;, paste(prefix, &quot;_temp_for_mutrate.bed.mutrate.txt&quot;, sep = &quot;&quot;), sep = &quot;&quot;)</span><br><span class="line">      system(command)</span><br><span class="line">      coverage_noncoding_base_mutrate &lt;-fread(paste(prefix, &quot;_temp_for_mutrate.bed.mutrate.txt&quot;, sep = &quot;&quot;), header = FALSE, stringsAsFactors = FALSE, sep = &quot;\t&quot;)</span><br><span class="line">      colnames(coverage_noncoding_base_mutrate) &lt;- c(&quot;base_ID&quot;,paste(&quot;base_mutrate_alt_&quot;, alt_letters[j], sep = &quot;&quot;))</span><br><span class="line">      coverage_noncoding_for_base_mutrate &lt;- coverage_noncoding_for_base_mutrate[coverage_noncoding_base_mutrate, on = &quot;base_ID&quot;]</span><br><span class="line">      system(paste(&quot;echo \&quot;Finished obtaining base-level mutation rate for alt allele &quot;, alt_letters[j], &quot;.\&quot;&quot;, sep = &quot;&quot;))</span><br><span class="line">      system(&quot;date&quot;)</span><br><span class="line">    &#125;</span><br><span class="line">    </span><br><span class="line">    # read in non allele-specific epigenomic annotations</span><br><span class="line">    epi_ID = 1</span><br><span class="line">    if (!is.na(nonAS_noncoding_annotations)[1])&#123; # then epigenomic_marks must be a vector of epigenomic bed files that need to be compard with the mutation data</span><br><span class="line">      for(epi in nonAS_noncoding_annotations)&#123;</span><br><span class="line">        command &lt;- paste(&quot;../external_tools/bedtools-2.17.0/bin/bedtools coverage -a &quot;, epi, &quot; -b &quot;, paste(prefix, &quot;_temp_for_mutrate.bed&quot;, sep = &quot;&quot;),  &quot; &gt; &quot;, paste(prefix,&quot;_temp_for_mutrate_overlap_epi.bed&quot;, sep = &quot;&quot;),sep = &quot;&quot;)</span><br><span class="line">        system(command)</span><br><span class="line">        base_in_epi &lt;- fread(paste(prefix,&quot;_temp_for_mutrate_overlap_epi.bed&quot;, sep = &quot;&quot;), header = FALSE, sep = &quot;\t&quot;, stringsAsFactors = FALSE)</span><br><span class="line">        base_in_epi &lt;- base_in_epi[,c(&quot;V4&quot;,&quot;V5&quot;), with = FALSE]</span><br><span class="line">        colnames(base_in_epi) &lt;- c(&quot;base_ID&quot;, paste(&quot;Anno&quot;,epi_ID, sep = &quot;_&quot;))</span><br><span class="line">        coverage_noncoding_for_base_mutrate &lt;- coverage_noncoding_for_base_mutrate[base_in_epi, on = &quot;base_ID&quot;]</span><br><span class="line">        system(paste(&quot;echo \&quot;Finished reading non-allele specific noncoding annotations &quot;, epi_ID, &quot;.\&quot;&quot;, sep = &quot;&quot;))</span><br><span class="line">        system(&quot;date&quot;)</span><br><span class="line">        epi_ID &lt;- epi_ID + 1</span><br><span class="line">      &#125;</span><br><span class="line">    &#125;</span><br><span class="line">    </span><br><span class="line">    # read in allele-specific epigenomic annotations, now the base assignment for such noncoding annotations is based on the 50-bp windows. </span><br><span class="line">    # Should not be a big problem as distal introns are assigned based on refseq_ID for intron regions, should be consistent with assignment by spidex.</span><br><span class="line">    # though there is a small chance that a gene&apos;s distal intron, which is close to the promoter of another gene and within 10 kb from the TSSs of the latter gene, may be mis-assigned to the latter gene.</span><br><span class="line">    # To completely correct for this issue, need to allow epigenetic annotation to take its own gene assignment, which might be necessary in some situations under strict criteria, such as splicing annotaion. </span><br><span class="line">    </span><br><span class="line">    if (!is.na(AS_noncoding_annotations)[1])&#123; # then epigenomic_marks must be a vector of epigenomic bed files that need to be compard with the mutation data</span><br><span class="line">      for(epi in AS_noncoding_annotations)&#123;</span><br><span class="line">        for(k in 1:length(epi))&#123;</span><br><span class="line">          command &lt;- paste(&quot;../external_tools/bedtools-2.17.0/bin/bedtools coverage -a &quot;, epi[k], &quot; -b &quot;, paste(prefix, &quot;_temp_for_mutrate.bed&quot;, sep = &quot;&quot;),  &quot; &gt; &quot;, paste(prefix,&quot;_temp_for_mutrate_overlap_epi.bed&quot;, sep = &quot;&quot;),sep = &quot;&quot;)</span><br><span class="line">          system(command)</span><br><span class="line">          base_in_epi &lt;- fread(paste(prefix,&quot;_temp_for_mutrate_overlap_epi.bed&quot;, sep = &quot;&quot;), header = FALSE, sep = &quot;\t&quot;, stringsAsFactors = FALSE)</span><br><span class="line">          base_in_epi &lt;- base_in_epi[,c(&quot;V4&quot;,&quot;V5&quot;), with = FALSE]</span><br><span class="line">          colnames(base_in_epi) &lt;- c(&quot;base_ID&quot;, paste(&quot;Anno&quot;,epi_ID, alt_letters[k], sep = &quot;_&quot;))</span><br><span class="line">          coverage_noncoding_for_base_mutrate &lt;- coverage_noncoding_for_base_mutrate[base_in_epi, on = &quot;base_ID&quot;]</span><br><span class="line">        &#125;</span><br><span class="line">        system(paste(&quot;echo \&quot;Finished reading allele specific noncoding annotations &quot;, epi_ID, &quot;.\&quot;&quot;, sep = &quot;&quot;))</span><br><span class="line">        system(&quot;date&quot;)</span><br><span class="line">        epi_ID &lt;- epi_ID + 1</span><br><span class="line">      &#125;</span><br><span class="line">    &#125;</span><br><span class="line">    </span><br><span class="line">    # now for each sample size, generate mut data.</span><br><span class="line">    for(m in 1:length(mutrate_scaling_files))&#123;</span><br><span class="line">      mut_output_temp &lt;- data.table(chr = character(), start = integer(), end = integer(), ref = character(), alt = character())</span><br><span class="line">      for(letter in alt_letters)&#123;</span><br><span class="line">        if(!is.na(nonAS_noncoding_annotations)[1] &amp; !is.na(AS_noncoding_annotations)[1])&#123;</span><br><span class="line">          coverage_noncoding_for_base_mutrate_temp &lt;- coverage_noncoding_for_base_mutrate[, c(&quot;base_ID&quot;, &quot;ID&quot;,paste(&quot;base_mutrate_alt_&quot;, letter, sep = &quot;&quot;), paste(&quot;Anno_&quot;, seq(1, nonAS_feature_number), sep = &quot;&quot;), paste(&quot;Anno_&quot;, seq(nonAS_feature_number +1, nonAS_feature_number + AS_feature_number), &quot;_&quot;, letter ,sep = &quot;&quot;), &quot;chr&quot;, &quot;start&quot;, &quot;end&quot;), with = FALSE]</span><br><span class="line">        &#125;else if(!is.na(nonAS_noncoding_annotations)[1] &amp; is.na(AS_noncoding_annotations)[1])&#123;</span><br><span class="line">          coverage_noncoding_for_base_mutrate_temp &lt;- coverage_noncoding_for_base_mutrate[, c(&quot;base_ID&quot;, &quot;ID&quot;,paste(&quot;base_mutrate_alt_&quot;, letter, sep = &quot;&quot;), paste(&quot;Anno_&quot;, seq(1, nonAS_feature_number), sep = &quot;&quot;), &quot;chr&quot;, &quot;start&quot;, &quot;end&quot;), with = FALSE]</span><br><span class="line">        &#125;else if(is.na(nonAS_noncoding_annotations)[1] &amp; !is.na(AS_noncoding_annotations)[1])&#123;</span><br><span class="line">          coverage_noncoding_for_base_mutrate_temp &lt;- coverage_noncoding_for_base_mutrate[, c(&quot;base_ID&quot;, &quot;ID&quot;,paste(&quot;base_mutrate_alt_&quot;, letter, sep = &quot;&quot;), paste(&quot;Anno_&quot;, seq(nonAS_feature_number +1, nonAS_feature_number + AS_feature_number), &quot;_&quot;, letter ,sep = &quot;&quot;), &quot;chr&quot;, &quot;start&quot;, &quot;end&quot;), with = FALSE]</span><br><span class="line">        &#125;</span><br><span class="line">        # a very important step here is removing bases that have adjusted_mutrate_base 0. This happens when the allele of the mutrate we are using is just the reference allele. </span><br><span class="line">        # By doing this, we make the computation of likelihood valid, also we automatically removed bases with nonAS annotations but with mutant allele the same with ref allele at this step</span><br><span class="line">        coverage_noncoding_for_base_mutrate_temp &lt;- coverage_noncoding_for_base_mutrate_temp[coverage_noncoding_for_base_mutrate_temp[[paste(&quot;base_mutrate_alt_&quot;, letter, sep = &quot;&quot;)]] !=0]</span><br><span class="line">        coverage_noncoding_for_base_mutrate_temp &lt;- coverage_noncoding_for_base_mutrate_temp[coverage[[i]][,c(paste(&quot;scaling_factor_study_&quot;, m, sep = &quot;&quot;), &quot;genename&quot;,&quot;ID&quot;,&quot;risk&quot;), with = FALSE],on = &quot;ID&quot;]</span><br><span class="line">        # removing sites with allele-specific mutrate == 0 will result in incomplete rows after merging.</span><br><span class="line">        coverage_noncoding_for_base_mutrate_temp &lt;- coverage_noncoding_for_base_mutrate_temp[complete.cases(coverage_noncoding_for_base_mutrate_temp)]</span><br><span class="line">        # derive the adjusted mutation rate for bases of risk genes and nonrisk genes.</span><br><span class="line">        # first multiplies sample size and scaling factor, adjusted_base_mutrate will be used for generating data in a compact format.</span><br><span class="line">        coverage_noncoding_for_base_mutrate_temp$adjusted_base_mutrate &lt;- coverage_noncoding_for_base_mutrate_temp[, paste(&quot;base_mutrate_alt_&quot;, letter, sep = &quot;&quot;), with = FALSE] * coverage_noncoding_for_base_mutrate_temp[, paste(&quot;scaling_factor_study_&quot;, m, sep = &quot;&quot;), with = FALSE] * 2 * sample_sizes[m]</span><br><span class="line">        # Then multiplies effect size after taking exponential, to generate a mutrate that has taken into account RRs, will be used for simulating mutations</span><br><span class="line">        # </span><br><span class="line">        coverage_noncoding_for_base_mutrate_temp$adjusted_base_mutrate_rr &lt;- exp(as.matrix(coverage_noncoding_for_base_mutrate_temp[,4 : (3 + feature_number)]) %*% rr * coverage_noncoding_for_base_mutrate_temp$risk) * coverage_noncoding_for_base_mutrate_temp$adjusted_base_mutrate</span><br><span class="line">        # generate random allele-specific mutations based on adjusted mutation rate.</span><br><span class="line">        coverage_noncoding_for_base_mutrate_temp$mut_count &lt;- rpois(nrow(coverage_noncoding_for_base_mutrate_temp), coverage_noncoding_for_base_mutrate_temp$adjusted_base_mutrate_rr)</span><br><span class="line">        </span><br><span class="line">        if(!is.na(compact_mut_output))&#123;</span><br><span class="line">          # have to collpase data at this point, otherwise, the I will run out of RAM if process 1000 genes every time. </span><br><span class="line">          anno_count &lt;- rep(0, nrow(coverage_noncoding_for_base_mutrate_temp))</span><br><span class="line">          for(p in 1:feature_number)&#123;</span><br><span class="line">            anno_count &lt;- anno_count + coverage_noncoding_for_base_mutrate_temp[[(4 + p -1)]]</span><br><span class="line">          &#125;</span><br><span class="line">          # remove rows(bases) that don&apos;t have any non-coding features, this could save a lot of RAM, so I could use smaller partition number which would greatly accelerate speed when read in data for all genes.</span><br><span class="line">          coverage_noncoding_for_base_mutrate_temp1 &lt;- coverage_noncoding_for_base_mutrate_temp[anno_count &gt;0,]</span><br><span class="line">          </span><br><span class="line">          # first partition by gene for the current chunk</span><br><span class="line">          coverage_noncoding_for_base_mutrate_temp1&lt;- split(coverage_noncoding_for_base_mutrate_temp1, coverage_noncoding_for_base_mutrate_temp1$genename)</span><br><span class="line">          # then partition by feature configuration for each gene in the current chunk</span><br><span class="line">          coverage_noncoding_for_base_mutrate_temp1 &lt;- sapply(coverage_noncoding_for_base_mutrate_temp1, partition_feature, simplify = FALSE)</span><br><span class="line">          # add compact data</span><br><span class="line">          data_partition &lt;- append(data_partition, coverage_noncoding_for_base_mutrate_temp1)</span><br><span class="line">          rm(coverage_noncoding_for_base_mutrate_temp1) # release memory</span><br><span class="line">          system(paste(&quot;echo \&quot;Finished read in mutation data and make them into the compact format for Study &quot;, m, &quot; and allele &quot;, letter, &quot;.\&quot;&quot;, sep = &quot;&quot;))</span><br><span class="line">          system(&quot;date&quot;)</span><br><span class="line">        &#125;</span><br><span class="line">        # then generate bed file format, this is a pseudobed file as I will put N at the 4th column as the ref allele is not relevant and mutant allele at the last column.</span><br><span class="line">        # remove rows(bases) that don&apos;t have any DNMs</span><br><span class="line">        coverage_noncoding_for_base_mutrate_temp &lt;- coverage_noncoding_for_base_mutrate_temp[mut_count &gt;0,]</span><br><span class="line">        if(node_n != 1)&#123;</span><br><span class="line">          mut_bed &lt;- rbindlist(parApply(cl, coverage_noncoding_for_base_mutrate_temp, 1, bed_extension))</span><br><span class="line">        &#125;else&#123;</span><br><span class="line">          mut_bed &lt;- rbindlist(apply(coverage_noncoding_for_base_mutrate_temp, 1, bed_extension))</span><br><span class="line">        &#125;</span><br><span class="line">        mut_bed$ref &lt;- &quot;N&quot;</span><br><span class="line">        mut_bed$alt &lt;- letter</span><br><span class="line">        mut_bed$start &lt;- as.integer(as.character(mut_bed$start))</span><br><span class="line">        mut_bed$end &lt;- as.integer(as.character(mut_bed$end))</span><br><span class="line">        mut_output_temp &lt;- rbind(mut_output_temp, mut_bed)</span><br><span class="line">      &#125;</span><br><span class="line">      mut_output_list[[m]] &lt;- rbind(mut_output_list[[m]], mut_output_temp)</span><br><span class="line">    &#125;</span><br><span class="line">  &#125;</span><br><span class="line">  </span><br><span class="line">  if(node_n != 1)&#123;</span><br><span class="line">    stopCluster(cl)</span><br><span class="line">  &#125;</span><br><span class="line">  options(warn = 0)</span><br><span class="line">  # remove temporary files</span><br><span class="line">  system(paste(&quot;rm &quot;, prefix, &quot;_temp*&quot;, sep = &quot;&quot;))</span><br><span class="line">  print(paste(&quot;echo \&quot;Temp files cleaned and mutation generation finished finished!\&quot;&quot;))</span><br><span class="line">  system(&quot;date&quot;)</span><br><span class="line">  # now write mutation file and mutation with allele info file</span><br><span class="line">  for(m in 1:length(mutrate_scaling_files))&#123;</span><br><span class="line">    fwrite(mut_output_list[[m]], output_allele_info_files[m], col.names = FALSE, row.names = FALSE, sep = &quot;\t&quot;, quote = FALSE)</span><br><span class="line">    fwrite(mut_output_list[[m]][,1:3], output_bed_files[m], col.names = FALSE, row.names = FALSE, sep = &quot;\t&quot;, quote = FALSE)</span><br><span class="line">  &#125;</span><br><span class="line">  write.table(risk_genes, output_risk_genes_file, col.names = FALSE, row.names = FALSE, sep = &quot;\t&quot;, quote = FALSE)</span><br><span class="line">  if(!is.na(compact_mut_output))&#123;</span><br><span class="line">    saveRDS(data_partition, compact_mut_output)</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">library(tadaA)</span><br><span class="line">setwd(&quot;/storage11_7T/fuy/TADA-A&quot;)</span><br></pre></td></tr></table></figure>


<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">TADA_A_adjust_mutrate(window_file = &apos;db/MS_data/windows_partition/cd_windows_with_div_score.bed&apos;,</span><br><span class="line">                      mut_file = &apos;/storage11_7T/fuy/TADA-A/TADA-A/lib/sim.DNM.bed&apos;,</span><br><span class="line">                      genes = &quot;all&quot;,</span><br><span class="line">                      sample_size = 6430,</span><br><span class="line">                      mutrate_mode = &quot;regular&quot;,</span><br><span class="line">                      scaling_file_name = &quot;/storage11_7T/fuy/TADA-A/TADA-A/lib/bed_sf.txt&quot;,</span><br><span class="line">                      scale_features = c(&quot;GC_content&quot;, &quot;div_score&quot;))</span><br></pre></td></tr></table></figure>

<pre><code>[1] &quot;Made a tmp folder for tmp files.&quot;
[1] &quot;Finished calculating mutation count within per genomic window. Time consumed: 1.452s&quot;
[1] &quot;Finished fitting mutation rate model and calculating scaling factors. Time consumed: 4.186s&quot;
[1] &quot;Removed temporary files.&quot;</code></pre><table>
<caption>A matrix: 3 Ã— 4 of type dbl</caption>
<thead>
    <tr><th></th><th scope="col">Estimate</th><th scope="col">Std. Error</th><th scope="col">z value</th><th scope="col">Pr(&gt;|z|)</th></tr>
</thead>
<tbody>
    <tr><th scope="row">(Intercept)</th><td> 0.05323104</td><td>0.01289846</td><td>  4.126929</td><td>3.676403e-05</td></tr>
    <tr><th scope="row">GC_content</th><td>-0.20127824</td><td>0.01364125</td><td>-14.755119</td><td>2.851895e-49</td></tr>
    <tr><th scope="row">div_score</th><td> 0.04558964</td><td>0.01242493</td><td>  3.669206</td><td>2.433050e-04</td></tr>
</tbody>
</table>




<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br></pre></td><td class="code"><pre><span class="line">system.time(compact_data &lt;- TADA_A_read_info_by_chunks(</span><br><span class="line">mut_files = &quot;/storage11_7T/fuy/TADA-A/TADA-A/lib/allele.bed&quot;,</span><br><span class="line">    </span><br><span class="line"># c(&quot;db/MS_data/DNM/Jiang_cases_DNM_with_allele_info.txt&quot;,</span><br><span class="line"># &quot;db/MS_data/DNM/Kong_cases_DNM_with_allele_info.txt&quot;,</span><br><span class="line"># &quot;db/MS_data/DNM/Michaelson_cases_DNM_with_allele_info.txt&quot;,</span><br><span class="line"># &quot;db/MS_data/DNM/Yuen_NM2015_cases_DNM_with_allele_info.txt&quot;,</span><br><span class="line"># &quot;db/MS_data/DNM/Wu_cases_DNM_with_allele_info.txt&quot;),    </span><br><span class="line">    </span><br><span class="line">window_file =  &apos;db/MS_data/windows_partition/cd_windows_with_div_score.bed&apos;,</span><br><span class="line">    </span><br><span class="line">mutrate_scaling_files = c(&quot;/storage11_7T/fuy/TADA-A/TADA-A/lib/bed_sf.txt&quot;),  </span><br><span class="line">    </span><br><span class="line"># sample_sizes = 4265,</span><br><span class="line">sample_sizes = 6430,    </span><br><span class="line">    </span><br><span class="line"># mutrate_scaling_files = c(&quot;db/MS_data/results/Jiang_windows_mutrate_with_div_score_scaling_file_for_test_DNM.txt&quot;,</span><br><span class="line">#                          &quot;db/MS_data/results/Kong_windows_mutrate_with_div_score_scaling_file_for_test_DNM.txt&quot;,</span><br><span class="line">#                          &quot;db/MS_data/results/Michaelson_windows_mutrate_with_div_score_scaling_file_for_test_DNM.txt&quot;,</span><br><span class="line">#                          &quot;db/MS_data/results/Yuen_NM2015_windows_mutrate_with_div_score_scaling_file_for_test_DNM.txt&quot;,</span><br><span class="line">#                          &quot;db/MS_data/results/Wu_windows_mutrate_with_div_score_scaling_file_for_test_DNM.txt&quot;),    </span><br><span class="line">    </span><br><span class="line"># sample_sizes = c(32,78,10,162,32),    </span><br><span class="line">    </span><br><span class="line">    </span><br><span class="line">gene_prior_file = &quot;db/MS_data/prior/uniform_gene_prior.txt&quot;,</span><br><span class="line">    </span><br><span class="line">nonAS_noncoding_annotations = NA, #c( </span><br><span class="line">#                                  &quot;/storage11_7T/fuy/TADA-A/annotation/ccr/ccrs.allchrom.gt90.bed&quot;),</span><br><span class="line">                                #&quot;/storage11_7T/fuy/TADA-A/annotation/ccr/chr_ccr_gt95_syn_rm.bed&quot;),</span><br><span class="line"></span><br><span class="line">AS_noncoding_annotations = list(</span><br><span class="line"># c(&quot;/storage11_7T/fuy/TADA-A/annotation/pLI/pLI_nlt_995_gnomad.v2.1.1.all_lofs_snv_alt_A.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/pLI_nlt_995_gnomad.v2.1.1.all_lofs_snv_alt_C.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/pLI_nlt_995_gnomad.v2.1.1.all_lofs_snv_alt_G.bed&quot;,</span><br><span class="line">#  &quot;/storage11_7T/fuy/TADA-A/annotation/pLI/pLI_nlt_995_gnomad.v2.1.1.all_lofs_snv_alt_T.bed&quot;)</span><br><span class="line">    </span><br><span class="line">                              c(&quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/all_cd_syn_alt_A.bed&quot;,</span><br><span class="line">                                &quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/all_cd_syn_alt_C.bed&quot;,</span><br><span class="line">                                &quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/all_cd_syn_alt_G.bed&quot;,</span><br><span class="line">                                &quot;/storage11_7T/fuy/TADA-A/annotation/driverMAPS/all_cd_syn_alt_T.bed&quot;)</span><br><span class="line">),                    </span><br><span class="line"></span><br><span class="line">report_proportion = 18665/18665,</span><br><span class="line">#chunk_partition_num =1,</span><br><span class="line">chunk = 2,</span><br><span class="line">node_n = 2,</span><br><span class="line">mutrate_ref_files = c(&quot;db/MS_data/mutrate/Example_windows_extended_1bp_for_getting_base_level_mutrate.bed.fasta.tri.alt_A.mutrate.bw&quot;,</span><br><span class="line">                   &quot;db/MS_data/mutrate/Example_windows_extended_1bp_for_getting_base_level_mutrate.bed.fasta.tri.alt_C.mutrate.bw&quot;,</span><br><span class="line">                   &quot;db/MS_data/mutrate/Example_windows_extended_1bp_for_getting_base_level_mutrate.bed.fasta.tri.alt_G.mutrate.bw&quot;,</span><br><span class="line">                   &quot;db/MS_data/mutrate/Example_windows_extended_1bp_for_getting_base_level_mutrate.bed.fasta.tri.alt_T.mutrate.bw&quot;)</span><br><span class="line">    </span><br><span class="line"># mutrate_ref_files = c(&quot;/storage11_7T/data.for.yuwen/new_mutrate/window.hg19.genome.tri2.alt_A.uq.bedGraph.sort.bw&quot; ,</span><br><span class="line">#                       &quot;/storage11_7T/data.for.yuwen/new_mutrate/window.hg19.genome.tri2.alt_C.uq.bedGraph.sort.bw&quot; ,</span><br><span class="line">#                       &quot;/storage11_7T/data.for.yuwen/new_mutrate/window.hg19.genome.tri2.alt_G.uq.bedGraph.sort.bw&quot; ,</span><br><span class="line">#                       &quot;/storage11_7T/data.for.yuwen/new_mutrate/window.hg19.genome.tri2.alt_T.uq.bedGraph.sort.bw&quot; )</span><br><span class="line">                      </span><br><span class="line"></span><br><span class="line">))</span><br><span class="line"></span><br><span class="line">saveRDS(compact_data,paste0(&quot;/storage11_7T/fuy/TADA-A/TADA-A/lib/&quot;,Sys.Date(),&quot;_sim_syn_compact.rds&quot;))</span><br></pre></td></tr></table></figure>


<pre><code> user   system  elapsed 
1.865    0.486 1344.063 </code></pre><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">tm = proc.time()</span><br><span class="line">dfn = c()</span><br><span class="line">for(i in 1)&#123;</span><br><span class="line">    </span><br><span class="line">RR = TADA_A_RR_estimate(data = compact_data$base_info, selected_annotations = c(i), </span><br><span class="line">#           gene_prior_file = &quot;db/MS_data/prior/Example_gene_prior.txt&quot;, </span><br><span class="line">        gene_prior_file = &quot;db/MS_data/prior/uniform_gene_prior.txt&quot;,</span><br><span class="line">                        optimization_iteration = 2000)</span><br><span class="line">    </span><br><span class="line">dfn = rbind(dfn,RR$rr_report)</span><br><span class="line">    &#125;</span><br><span class="line">proc.time() - tm</span><br><span class="line"># saveRDS(RR,paste0(&quot;/storage11_7T/fuy/TADA-A/annotation/results/&quot;,Sys.Date(),&quot;_wes_deepsea_brain_RR.rds&quot;))</span><br><span class="line"></span><br><span class="line">dfn</span><br></pre></td></tr></table></figure>

<pre><code>[1] &quot;Read in gene prior file. Time consumed: 0.033s.&quot;
[1] &quot;Finished optimization. Time consumed: 259.537.s&quot;
[1] &quot;Got confidence intervals of RR estimates.&quot;
[1] &quot;Finished RR estimation!&quot;



    user   system  elapsed 
1709.807    1.949  259.580 </code></pre><table>
<caption>A data.frame: 1 Ã— 3</caption>
<thead>
    <tr><th scope="col">logRR</th><th scope="col">lower_bound</th><th scope="col">upper_bound</th></tr>
    <tr><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>0.6742976</td><td>0.3087695</td><td>1.039826</td></tr>
</tbody>
</table>




<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">dfn$RR = exp(dfn$logRR)</span><br><span class="line">dfn$exp = 1689/6130</span><br><span class="line">dfn$obs = 1810/6130</span><br><span class="line">dfn$n_DNM_org = 6954</span><br><span class="line">dfn$n_DNM_sim = 6130</span><br><span class="line">dfn$oe_ratio = dfn$obs / dfn$exp</span><br><span class="line">dfn</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.frame: 1 Ã— 9</caption>
<thead>
    <tr><th scope="col">logRR</th><th scope="col">lower_bound</th><th scope="col">upper_bound</th><th scope="col">RR</th><th scope="col">exp</th><th scope="col">obs</th><th scope="col">oe_ratio</th><th scope="col">n_DNM_org</th><th scope="col">n_DNM_sim</th></tr>
    <tr><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>0.6742976</td><td>0.3087695</td><td>1.039826</td><td>1.962654</td><td>0.2755302</td><td>0.2952692</td><td>1.07164</td><td>6954</td><td>6130</td></tr>
</tbody>
</table>




<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">1640/6954/(1756/6954)</span><br></pre></td></tr></table></figure>


<p>0.933940774487472</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">l = length(compact_data$base_info)</span><br><span class="line">l</span><br><span class="line"></span><br><span class="line">s = 0</span><br><span class="line">for(i in 1:l)&#123;</span><br><span class="line">    s = s + compact_data$base_info[[i]]$`1`$sum_mut_rate</span><br><span class="line">&#125;</span><br><span class="line">s</span><br></pre></td></tr></table></figure>


<p>67106</p>
<p>1689.09746890996</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">cd /storage11_7T/fuy/TADA-A/TADA-A/lib</span><br><span class="line">wc -l allele.bed</span><br></pre></td></tr></table></figure>

<pre><code>6130 allele.bed</code></pre></div><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">Author: </span><span class="post-copyright-info"><a href="mailto:undefined" target="_blank" rel="noopener"></a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">Link: </span><span class="post-copyright-info"><a href="https://yfu1116.github.io/project/1104-tadaA-simulate-6430-rr036/">https://yfu1116.github.io/project/1104-tadaA-simulate-6430-rr036/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">Copyright Notice: </span><span class="post-copyright-info">All articles in this blog are licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank" rel="noopener">CC BY-NC-SA 4.0</a> unless stating additionally.</span></div></div><div class="tag_share"><div class="post-meta__tag-list"></div><div class="post_share"><div class="social-share" data-image="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" data-sites="github,email,facebook,twitter,weixin,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js/dist/css/share.min.css"><script src="https://cdn.jsdelivr.net/npm/social-share.js/dist/js/social-share.min.js" defer></script></div></div><nav class="pagination-post" id="pagination"><div class="prev-post pull-left"><a href="/project/1105-tadaA-simulate-rr0-uniform-scale-6166/"><img class="prev-cover" data-lazy-src="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" onerror="onerror=null;src='/project/img/404.jpg'"><div class="pagination-info"><div class="label">Previous Post</div><div class="prev_info">1105-tadaA-simulate-rr0-uniform-scale-6166</div></div></a></div><div class="next-post pull-right"><a href="/project/201023-tadaA-res-allinfo/"><img class="next-cover" data-lazy-src="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" onerror="onerror=null;src='/project/img/404.jpg'"><div class="pagination-info"><div class="label">Next Post</div><div class="next_info">201023-tadaA-res-allinfo</div></div></a></div></nav></article></main><footer id="footer" data-type="color"><div id="footer-wrap"><div class="copyright">&copy;2020 - 2021 By </div><div class="framework-info"><span>Framework </span><a href="https://hexo.io" target="_blank" rel="noopener">Hexo</a><span class="footer-separator">|</span><span>Theme </span><a href="https://github.com/jerryc127/hexo-theme-butterfly" target="_blank" rel="noopener">Butterfly</a></div></div></footer></div><section id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="Read Mode"><i class="fas fa-book-open"></i></button><button id="font_plus" type="button" title="Increase Font Size"><i class="fas fa-plus"></i></button><button id="font_minus" type="button" title="Decrease Font Size"><i class="fas fa-minus"></i></button><button id="darkmode" type="button" title="Switch Between Light And Dark Mode"><i class="fas fa-adjust"></i></button></div><div id="rightside-config-show"><button id="rightside_config" type="button" title="Setting"><i class="fas fa-cog"></i></button><button id="go-up" type="button" title="Back To Top"><i class="fas fa-arrow-up"></i></button></div></section><div><script src="https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js"></script><script src="/project/js/utils.js"></script><script src="/project/js/main.js"></script><script src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.js"></script><script src="https://cdn.jsdelivr.net/npm/instant.page/instantpage.min.js" type="module" defer></script><script src="https://cdn.jsdelivr.net/npm/vanilla-lazyload/dist/lazyload.iife.min.js" async></script><div class="js-pjax"><script>if (!window.MathJax) {
  window.MathJax = {
    loader: {
      source: {
        '[tex]/amsCd': '[tex]/amscd'
      }
    },
    tex: {
      inlineMath: [ ['$','$'], ["\\(","\\)"]],
      tags: 'ams'
    },
    options: {
      renderActions: {
        findScript: [10, doc => {
          for (const node of document.querySelectorAll('script[type^="math/tex"]')) {
            const display = !!node.type.match(/; *mode=display/)
            const math = new doc.options.MathItem(node.textContent, doc.inputJax[0], display)
            const text = document.createTextNode('')
            node.parentNode.replaceChild(text, node)
            math.start = {node: text, delim: '', n: 0}
            math.end = {node: text, delim: '', n: 0}
            doc.math.push(math)
          }
        }, '']
      }
    }
  }
  
  var script = document.createElement('script')
  script.src = 'https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js'
  script.id = 'MathJax-script'
  script.async = true
  document.head.appendChild(script)
} else {
  MathJax.startup.document.state(0)
  MathJax.texReset()
  MathJax.typeset()
}</script><script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div></div></body></html>