<!DOCTYPE html><html lang="en" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1"><title>2021-08-10-tadaA-report | yang's blog</title><meta name="description" content="Check RR, prior, and FDR by simulationparameters: 3 annotations (Lof: ptv &gt; 0.995, union of spidex_low3 and spliceai, without ptv, mpc&gt;2) , log-RR = 3, 2,1 pi = 0.05,   Sample size N  = 6000,"><meta name="author" content=""><meta name="copyright" content=""><meta name="format-detection" content="telephone=no"><link rel="shortcut icon" href="/project/img/favicon.png"><link rel="canonical" href="https://yfu1116.github.io/project/2021-08-10-tadaA-report/"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//fonts.googleapis.com" crossorigin="crossorigin"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><meta property="og:type" content="article"><meta property="og:title" content="2021-08-10-tadaA-report"><meta property="og:url" content="https://yfu1116.github.io/project/2021-08-10-tadaA-report/"><meta property="og:site_name" content="yang's blog"><meta property="og:description" content="Check RR, prior, and FDR by simulationparameters: 3 annotations (Lof: ptv &gt; 0.995, union of spidex_low3 and spliceai, without ptv, mpc&gt;2) , log-RR = 3, 2,1 pi = 0.05,   Sample size N  = 6000,"><meta property="og:image" content="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg"><meta property="article:published_time" content="2021-08-13T10:04:35.793Z"><meta property="article:modified_time" content="2021-08-13T10:04:45.114Z"><meta name="twitter:card" content="summary"><script>var activateDarkMode = function () {
  document.documentElement.setAttribute('data-theme', 'dark')
  if (document.querySelector('meta[name="theme-color"]') !== null) {
    document.querySelector('meta[name="theme-color"]').setAttribute('content', '#000')
  }
}
var activateLightMode = function () {
  document.documentElement.setAttribute('data-theme', 'light')
  if (document.querySelector('meta[name="theme-color"]') !== null) {
    document.querySelector('meta[name="theme-color"]').setAttribute('content', '#fff')
  }
}

var getCookies = function (name) {
  const value = `; ${document.cookie}`
  const parts = value.split(`; ${name}=`)
  if (parts.length === 2) return parts.pop().split(';').shift()
}

var autoChangeMode = 'false'
var t = getCookies('theme')
if (autoChangeMode === '1') {
  var isDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches
  var isLightMode = window.matchMedia('(prefers-color-scheme: light)').matches
  var isNotSpecified = window.matchMedia('(prefers-color-scheme: no-preference)').matches
  var hasNoSupport = !isDarkMode && !isLightMode && !isNotSpecified

  if (t === undefined) {
    if (isLightMode) activateLightMode()
    else if (isDarkMode) activateDarkMode()
    else if (isNotSpecified || hasNoSupport) {
      console.log('You specified no preference for a color scheme or your browser does not support it. I Schedule dark mode during night time.')
      var now = new Date()
      var hour = now.getHours()
      var isNight = hour <= 6 || hour >= 18
      isNight ? activateDarkMode() : activateLightMode()
    }
    window.matchMedia('(prefers-color-scheme: dark)').addListener(function (e) {
      if (Cookies.get('theme') === undefined) {
        e.matches ? activateDarkMode() : activateLightMode()
      }
    })
  } else if (t === 'light') activateLightMode()
  else activateDarkMode()
} else if (autoChangeMode === '2') {
  now = new Date()
  hour = now.getHours()
  isNight = hour <= 6 || hour >= 18
  if (t === undefined) isNight ? activateDarkMode() : activateLightMode()
  else if (t === 'light') activateLightMode()
  else activateDarkMode()
} else {
  if (t === 'dark') activateDarkMode()
  else if (t === 'light') activateLightMode()
}</script><link rel="stylesheet" href="/project/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.css"><link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Titillium+Web&amp;display=swap"><script>var GLOBAL_CONFIG = { 
  root: '/project/',
  hexoversion: '4.0.0',
  algolia: undefined,
  localSearch: undefined,
  translate: undefined,
  noticeOutdate: undefined,
  copy: {
    success: 'Copy successfully',
    error: 'Copy error',
    noSupport: 'The browser does not support'
  },
  bookmark: {
    message_prev: 'Press',
    message_next: 'to bookmark this page'
  },
  runtime_unit: 'days',
  runtime: false,
  copyright: undefined,
  ClickShowText: undefined,
  medium_zoom: false,
  fancybox: true,
  Snackbar: undefined,
  justifiedGallery: {
    js: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/js/jquery.justifiedGallery.min.js',
    css: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/css/justifiedGallery.min.css'
  },
  baiduPush: false,
  highlightCopy: true,
  highlightLang: true,
  isPhotoFigcaption: false,
  islazyload: true,
  isanchor: false    
}</script><script id="config_change">var GLOBAL_CONFIG_SITE = { 
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isSidebar: true,
  postUpdate: '2021-08-13 18:04:45'
}</script><noscript><style>
#nav {
  opacity: 1
}
.justified-gallery img {
  opacity: 1
}
</style></noscript></head><body><div id="mobile-sidebar"><div id="menu_mask"></div><div id="mobile-sidebar-menus"><div class="mobile_author_icon"><img class="avatar-img" data-lazy-src="/project/img/avatar.png" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="mobile_post_data"><div class="mobile_data_item is-center"><div class="mobile_data_link"><a href="/project/archives/"><div class="headline">Articles</div><div class="length_num">71</div></a></div></div></div><hr/></div></div><div id="body-wrap"><div id="sidebar"><i class="fas fa-arrow-right on" id="toggle-sidebar"></i><div class="sidebar-toc"><div class="sidebar-toc__title">Catalog</div><div class="sidebar-toc__progress"><span class="progress-notice">You've read</span><span class="progress-num">0</span><span class="progress-percentage">%</span><div class="sidebar-toc__progress-bar">     </div></div><div class="sidebar-toc__content"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#Check-RR-prior-and-FDR-by-simulation"><span class="toc-number">1.</span> <span class="toc-text">Check RR, prior, and FDR by simulation</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#logRR-and-pi-（averaged-over-100-rounds-of-simulation）"><span class="toc-number">1.1.</span> <span class="toc-text">logRR and pi （averaged over 100 rounds of simulation）</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#FDR-averaged-over-100-rounds-of-simulation"><span class="toc-number">1.2.</span> <span class="toc-text">FDR (averaged over 100 rounds of simulation)</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#two-simulated-dataset-‘Rdata’-files-for-mutation-rate-and-mutation-count"><span class="toc-number">1.3.</span> <span class="toc-text">two simulated dataset (‘Rdata’ files for mutation rate and mutation count)</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#ASD"><span class="toc-number">2.</span> <span class="toc-text">ASD</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#RR-of-14-functional-annotations"><span class="toc-number">2.1.</span> <span class="toc-text">RR of 14 functional annotations</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#effect-size-of-frameshift"><span class="toc-number">2.2.</span> <span class="toc-text">effect size of frameshift</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#method-1"><span class="toc-number">2.2.1.</span> <span class="toc-text">method 1</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#method-2"><span class="toc-number">2.2.2.</span> <span class="toc-text">method 2</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#novel-genes-method-1"><span class="toc-number">2.3.</span> <span class="toc-text">novel genes (method 1)</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#Enrichment-analysis"><span class="toc-number">2.3.1.</span> <span class="toc-text">Enrichment analysis</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#potentially-damaging-mutations"><span class="toc-number">2.4.</span> <span class="toc-text">potentially damaging mutations</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#CHD"><span class="toc-number">3.</span> <span class="toc-text">CHD</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#effect-size-of-13-SNV-annotations-pi-0-022"><span class="toc-number">3.1.</span> <span class="toc-text">effect size of 13 SNV annotations, pi=0.022</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#effect-size-of-frameshift-1"><span class="toc-number">3.2.</span> <span class="toc-text">effect size of frameshift</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#method-1-1"><span class="toc-number">3.2.1.</span> <span class="toc-text">method 1</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#method-2-1"><span class="toc-number">3.2.2.</span> <span class="toc-text">method 2</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#novel-genes-method-1-1"><span class="toc-number">3.3.</span> <span class="toc-text">novel genes (method 1)</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Enrichment-analysis-1"><span class="toc-number">3.4.</span> <span class="toc-text">Enrichment analysis</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#GWAS-signals"><span class="toc-number">3.5.</span> <span class="toc-text">GWAS signals</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#potentially-damaging-variants"><span class="toc-number">3.6.</span> <span class="toc-text">potentially damaging variants</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#new-DD-dataset-to-be-explored"><span class="toc-number">4.</span> <span class="toc-text">new DD dataset to be explored</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#Question"><span class="toc-number">5.</span> <span class="toc-text">Question</span></a></li></ol></div></div></div><header class="post-bg" id="page-header" style="background-image: url(https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg)"><nav id="nav"><span class="pull-left" id="blog_name"><a class="blog_title" id="site-name" href="/project/">yang's blog</a></span><span class="pull-right menus"><span class="toggle-menu close"><a class="site-page"><i class="fas fa-bars fa-fw"></i></a></span></span></nav><div id="post-info"><div id="post-title"><div class="posttitle">2021-08-10-tadaA-report</div></div><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">Created</span><time class="post-meta-date-created" datetime="2021-08-13T10:04:35.793Z" title="Created 2021-08-13 18:04:35">2021-08-13</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">Updated</span><time class="post-meta-date-updated" datetime="2021-08-13T10:04:45.114Z" title="Updated 2021-08-13 18:04:45">2021-08-13</time></span></div><div class="meta-secondline"> <span class="post-meta-separator">|</span><span class="post-meta-pv-cv"><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">Post View:</span><span id="busuanzi_value_page_pv"></span></span></div></div></div></header><main class="layout_post" id="content-inner"><article id="post"><div class="post-content" id="article-container"><h1 id="Check-RR-prior-and-FDR-by-simulation"><a href="#Check-RR-prior-and-FDR-by-simulation" class="headerlink" title="Check RR, prior, and FDR by simulation"></a>Check RR, prior, and FDR by simulation</h1><p><strong>parameters:</strong> 3 annotations (<code>Lof: ptv &gt; 0.995</code>, <code>union of spidex_low3 and spliceai, without ptv</code>, <code>mpc&gt;2</code>) , log-RR = 3, 2,1</p>
<p><code>pi</code> = 0.05,   <code>Sample size N</code>  = 6000,    <code>num_of_genes</code> = 5000</p>
<h2 id="logRR-and-pi-（averaged-over-100-rounds-of-simulation）"><a href="#logRR-and-pi-（averaged-over-100-rounds-of-simulation）" class="headerlink" title="logRR and pi （averaged over 100 rounds of simulation）"></a>logRR and pi （averaged over 100 rounds of simulation）</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">readRDS(&quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/simulation/org.rr.mean.rds&quot;)</span><br></pre></td></tr></table></figure>


<table>
<caption>A matrix: 3 × 4 of type chr</caption>
<tbody>
    <tr><th scope="row">annota</th><td>Lof: ptv &gt; 0.995</td><td>union of spidex_low3 and spliceai, without ptv</td><td>mpc&gt;2 </td><td>prior </td></tr>
    <tr><th scope="row">sim_par</th><td>3.00            </td><td>2.00                                          </td><td>1.00  </td><td>0.05  </td></tr>
    <tr><th scope="row">joint_estim_average</th><td>2.7197          </td><td>1.7714                                        </td><td>0.6039</td><td>0.0926</td></tr>
</tbody>
</table>



<ul>
<li>estimated parameter大体上接近simulation的设定值，但有时会估出CI (Confidence Interval) 特别大的结果，如：</li>
</ul>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">readRDS(&quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/simulation/dt19_allrrinfo.rds&quot;)</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.frame: 4 × 9</caption>
<thead>
    <tr><th scope="col">annota</th><th scope="col">joint_estim</th><th scope="col">joint_estim_CI</th><th scope="col">fix_pi0.05_joint</th><th scope="col">fix_pi0.05_sep</th><th scope="col">fix_pi0.05_sep_CI</th><th scope="col">obs_DNM</th><th scope="col">bg_DNM</th><th scope="col">burden(obs/bg)</th></tr>
    <tr><th scope="col">&lt;chr&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;chr&gt;</th><th scope="col">&lt;chr&gt;</th><th scope="col">&lt;chr&gt;</th><th scope="col">&lt;chr&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td><span style="white-space:pre-wrap">Lof: ptv &gt; 0.995                              </span></td><td> 1.2852623</td><td><span style="white-space:pre-wrap">[-0.103, 2.674]  </span></td><td><span style="white-space:pre-wrap">2.256291        </span></td><td><span style="white-space:pre-wrap">2.469966        </span></td><td><span style="white-space:pre-wrap">[1.275, 3.665]  </span></td><td>15</td><td>10</td><td>1.5000000</td></tr>
    <tr><td>union of spidex_low3 and spliceai, without ptv</td><td> 1.1044304</td><td>[0.16, 2.049]    </td><td>1.986388        </td><td>2.044128        </td><td>[1.361, 2.727]  </td><td>81</td><td>58</td><td>1.3965517</td></tr>
    <tr><td><span style="white-space:pre-wrap">mpc&gt;2                                         </span></td><td>-3.4387649</td><td>[-25.695, 18.818]</td><td><span style="white-space:pre-wrap">-4.936559       </span></td><td><span style="white-space:pre-wrap">-1              </span></td><td>[-5.296, 3.296] </td><td>49</td><td>61</td><td>0.8032787</td></tr>
    <tr><td>prior                                         </td><td> 0.1991692</td><td>[-0.057, 0.455]  </td><td>fix prior = 0.05</td><td>fix prior = 0.05</td><td>fix prior = 0.05</td><td>NA</td><td>NA</td><td>       NA</td></tr>
</tbody>
</table>



<h2 id="FDR-averaged-over-100-rounds-of-simulation"><a href="#FDR-averaged-over-100-rounds-of-simulation" class="headerlink" title="FDR (averaged over 100 rounds of simulation)"></a>FDR (averaged over 100 rounds of simulation)</h2><ul>
<li>it seems that 0.1 is feasible.</li>
</ul>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">readRDS(&quot;/storage11_7T/fuy/TADA-A/cell_WES/DNM/simulation/org.fdr.check.res.rds&quot;)</span><br></pre></td></tr></table></figure>


<table>
<caption>A data.frame: 8 × 2</caption>
<thead>
    <tr><th scope="col">cutoff.lst</th><th scope="col">fdr.average</th></tr>
    <tr><th scope="col">&lt;dbl&gt;</th><th scope="col">&lt;dbl&gt;</th></tr>
</thead>
<tbody>
    <tr><td>0.02</td><td>0.02165</td></tr>
    <tr><td>0.05</td><td>0.04120</td></tr>
    <tr><td>0.08</td><td>0.08554</td></tr>
    <tr><td>0.10</td><td>0.10238</td></tr>
    <tr><td>0.15</td><td>0.17765</td></tr>
    <tr><td>0.20</td><td>0.24224</td></tr>
    <tr><td>0.30</td><td>0.31463</td></tr>
    <tr><td>0.50</td><td>0.53028</td></tr>
</tbody>
</table>



<h2 id="two-simulated-dataset-‘Rdata’-files-for-mutation-rate-and-mutation-count"><a href="#two-simulated-dataset-‘Rdata’-files-for-mutation-rate-and-mutation-count" class="headerlink" title="two simulated dataset (‘Rdata’ files for mutation rate and mutation count)"></a>two simulated dataset (‘Rdata’ files for mutation rate and mutation count)</h2><ul>
<li>dt9 (estimated RRs are normal)</li>
<li>dt19(estimated RRs are abnormal)</li>
</ul>
<p>download: <a href="https://github.com/yfu1116/project/tree/master/files/RData" target="_blank" rel="noopener">https://github.com/yfu1116/project/tree/master/files/RData</a></p>
<h1 id="ASD"><a href="#ASD" class="headerlink" title="ASD"></a>ASD</h1><h2 id="RR-of-14-functional-annotations"><a href="#RR-of-14-functional-annotations" class="headerlink" title="RR of 14 functional annotations"></a>RR of 14 functional annotations</h2><ul>
<li><p>SNV rate was calibrated using the number of synonymous mutations.</p>
</li>
<li><p>We fixed the prior as 0.05 in the initial feature selection,and took all significant ones into joint estimation, where pi was also regarded as a parameter.</p>
</li>
<li><p>pi = 0.12 in joint estimation</p>
</li>
<li><p>details: <a href="https://yfu1116.github.io/project/2021-05-20-fix-pi-VS-estim-pi-effect-size-num-enrich/">https://yfu1116.github.io/project/2021-05-20-fix-pi-VS-estim-pi-effect-size-num-enrich/</a></p>
</li>
</ul>
<h2 id="effect-size-of-frameshift"><a href="#effect-size-of-frameshift" class="headerlink" title="effect size of frameshift"></a>effect size of frameshift</h2><h3 id="method-1"><a href="#method-1" class="headerlink" title="method 1"></a>method 1</h3><ul>
<li>derived inframe rate using GLM</li>
</ul>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">glm(formula = inframe count from the affected ~ calibrated_SNV_rate * 2N + GC_content, family = poisson(link = &quot;log&quot;), data = df)</span><br></pre></td></tr></table></figure>

<ul>
<li><p>frameshift rate = (frameshift count / inframe count from the unaffected) * inframe rate</p>
</li>
<li><p>estimated relative risk of frameshift using EM, <strong>RR = 17.7,    logRR = 2.9</strong></p>
</li>
<li><p>details: <a href="https://yfu1116.github.io/project/2021-08-10-nonfs-rate-fs-gama-report/">https://yfu1116.github.io/project/2021-08-10-nonfs-rate-fs-gama-report/</a></p>
</li>
</ul>
<h3 id="method-2"><a href="#method-2" class="headerlink" title="method 2"></a>method 2</h3><ul>
<li>took the RR of nonsense as the RR of frameshift (fix prior = 0.05), <strong>logRR = 1.952505, CI=[1.782561, 2.122449]</strong></li>
</ul>
<p>There is a big difference of results between these two methods.</p>
<h2 id="novel-genes-method-1"><a href="#novel-genes-method-1" class="headerlink" title="novel genes (method 1)"></a>novel genes (method 1)</h2><ul>
<li>FDR = 0.1</li>
<li>158 risk genes were identified, including <code>105 novel genes</code> with overlapping genes excluded</li>
</ul>
<p><strong>For overlapping genes:</strong></p>
<p>window中保留了overlap的区间，因此有些与risk genes重叠较多的基因，如UGT系列，也会被模型找出，目前采取的办法是只保留文章里DNM对应的那个</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">UGT1A3, UGT1A4, UGT1A5, UGT1A7, UGT1A10, UGT1A9, UGT1A1, UGT1A6</span><br></pre></td></tr></table></figure>

<ul>
<li>comparison between the results of baseline (MPC+PTV) and the annotations we used<br><a href="https://yfu1116.github.io/project/2021-08-10-ASD-CHD-res-comparison/">https://yfu1116.github.io/project/2021-08-10-ASD-CHD-res-comparison/</a></li>
</ul>
<h3 id="Enrichment-analysis"><a href="#Enrichment-analysis" class="headerlink" title="Enrichment analysis"></a>Enrichment analysis</h3><p>105 novel ASD risk genes (<code>frameshift + snv</code> vs. <code>baseline</code>; remove the redundant)</p>
<p><strong>GO</strong></p>
<p><img src= "/project/img/loading.gif" data-lazy-src="https://z3.ax1x.com/2021/08/10/fYpoPx.png" alt="fYpoPx.png"></p>
<p><strong>DisGeNET (gene-disease association)</strong></p>
<p><img src= "/project/img/loading.gif" data-lazy-src="https://z3.ax1x.com/2021/08/10/fY9iQS.png" alt="fY9iQS.png"></p>
<h2 id="potentially-damaging-mutations"><a href="#potentially-damaging-mutations" class="headerlink" title="potentially damaging mutations"></a>potentially damaging mutations</h2><ul>
<li><p>Some of the synonymous DNMs might be pathogenic by disrupting splicing or RNA-binding. For instance, a synonymous mutation in KCNMA1 gene, might contribute to the etiology of ASD through regulation of argonaute protein (AGO) binding.<strong>(to be further investigated)</strong></p>
</li>
<li><p>all info: <a href="https://github.com/yfu1116/project/blob/master/files/filtered.indel%2Bsnv-baseline_ASD_GO_DisGeNET.xlsx" target="_blank" rel="noopener">https://github.com/yfu1116/project/blob/master/files/filtered.indel%2Bsnv-baseline_ASD_GO_DisGeNET.xlsx</a></p>
</li>
</ul>
<h1 id="CHD"><a href="#CHD" class="headerlink" title="CHD"></a>CHD</h1><h2 id="effect-size-of-13-SNV-annotations-pi-0-022"><a href="#effect-size-of-13-SNV-annotations-pi-0-022" class="headerlink" title="effect size of 13 SNV annotations, pi=0.022"></a>effect size of 13 SNV annotations, pi=0.022</h2><p><a href="https://yfu1116.github.io/project/2021-06-01-CHD-selected13-all-rr-info/">https://yfu1116.github.io/project/2021-06-01-CHD-selected13-all-rr-info/</a></p>
<h2 id="effect-size-of-frameshift-1"><a href="#effect-size-of-frameshift-1" class="headerlink" title="effect size of frameshift"></a>effect size of frameshift</h2><h3 id="method-1-1"><a href="#method-1-1" class="headerlink" title="method 1"></a>method 1</h3><ul>
<li>RR =52.85 , logRR=3.97</li>
</ul>
<h3 id="method-2-1"><a href="#method-2-1" class="headerlink" title="method 2"></a>method 2</h3><ul>
<li>RR 105.6, = logRR = 4.66</li>
</ul>
<h2 id="novel-genes-method-1-1"><a href="#novel-genes-method-1-1" class="headerlink" title="novel genes (method 1)"></a>novel genes (method 1)</h2><ul>
<li>FDR = 0.1</li>
<li>32 risk genes were identified, including <code>24 novel genes</code> (compared to the baseline (MPC+PTV)) with overlapping genes excluded</li>
<li>results comparison between baseline and other annotations<br><a href="https://yfu1116.github.io/project/2021-08-10-ASD-CHD-res-comparison/">https://yfu1116.github.io/project/2021-08-10-ASD-CHD-res-comparison/</a></li>
</ul>
<h2 id="Enrichment-analysis-1"><a href="#Enrichment-analysis-1" class="headerlink" title="Enrichment analysis"></a>Enrichment analysis</h2><p>32 risk genes</p>
<p><strong>GO</strong><br><img src= "/project/img/loading.gif" data-lazy-src="https://z3.ax1x.com/2021/08/11/fULVJK.png" alt="fULVJK.png"></p>
<p><strong>DisGeNET (gene-disease association)</strong></p>
<p><img src= "/project/img/loading.gif" data-lazy-src="https://z3.ax1x.com/2021/08/11/fUL2y4.png" alt="fUL2y4.png"></p>
<h2 id="GWAS-signals"><a href="#GWAS-signals" class="headerlink" title="GWAS signals"></a>GWAS signals</h2><p>We do not find any CHD GWAS signals associated with these risk genes.</p>
<h2 id="potentially-damaging-variants"><a href="#potentially-damaging-variants" class="headerlink" title="potentially damaging variants"></a>potentially damaging variants</h2><p>all info: <a href="https://github.com/yfu1116/project/blob/master/files/filtered.indel%2Bsnv_CHD_GO_DisGeNET.xlsx" target="_blank" rel="noopener">https://github.com/yfu1116/project/blob/master/files/filtered.indel%2Bsnv_CHD_GO_DisGeNET.xlsx</a></p>
<h1 id="new-DD-dataset-to-be-explored"><a href="#new-DD-dataset-to-be-explored" class="headerlink" title="new DD dataset to be explored"></a>new DD dataset to be explored</h1><ul>
<li><a href="https://www.nature.com/articles/s41586-020-2832-5#Sec2" target="_blank" rel="noopener">https://www.nature.com/articles/s41586-020-2832-5#Sec2</a></li>
<li>45,221 coding and splicing DNMs in 31,058 individuals </li>
</ul>
<h1 id="Question"><a href="#Question" class="headerlink" title="Question"></a>Question</h1><ul>
<li>How to explain the outliers in RRs (account for 20%) estimated using simulated data?</li>
<li>There is a big difference of frameshift RR between method 1 and method 2. I’m confused why the RR of nonsense could be a substitution for the RR of frameshift.</li>
</ul>
</div><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">Author: </span><span class="post-copyright-info"><a href="mailto:undefined" target="_blank" rel="noopener"></a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">Link: </span><span class="post-copyright-info"><a href="https://yfu1116.github.io/project/2021-08-10-tadaA-report/">https://yfu1116.github.io/project/2021-08-10-tadaA-report/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">Copyright Notice: </span><span class="post-copyright-info">All articles in this blog are licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank" rel="noopener">CC BY-NC-SA 4.0</a> unless stating additionally.</span></div></div><div class="tag_share"><div class="post-meta__tag-list"></div><div class="post_share"><div class="social-share" data-image="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" data-sites="github,email,facebook,twitter,weixin,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js/dist/css/share.min.css"><script src="https://cdn.jsdelivr.net/npm/social-share.js/dist/js/social-share.min.js" defer></script></div></div><nav class="pagination-post" id="pagination"><div class="prev-post pull-left"><a href="/project/2021-08-19-tadaA-report/"><img class="prev-cover" data-lazy-src="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" onerror="onerror=null;src='/project/img/404.jpg'"><div class="pagination-info"><div class="label">Previous Post</div><div class="prev_info">2021-08-19-tadaA-report</div></div></a></div><div class="next-post pull-right"><a href="/project/2021-08-10-ASD-CHD-res-comparison/"><img class="next-cover" data-lazy-src="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" onerror="onerror=null;src='/project/img/404.jpg'"><div class="pagination-info"><div class="label">Next Post</div><div class="next_info">2021-08-10-ASD-CHD-res-comparison</div></div></a></div></nav></article></main><footer id="footer" data-type="color"><div id="footer-wrap"><div class="copyright">&copy;2020 - 2021 By </div><div class="framework-info"><span>Framework </span><a href="https://hexo.io" target="_blank" rel="noopener">Hexo</a><span class="footer-separator">|</span><span>Theme </span><a href="https://github.com/jerryc127/hexo-theme-butterfly" target="_blank" rel="noopener">Butterfly</a></div></div></footer></div><section id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="Read Mode"><i class="fas fa-book-open"></i></button><button id="font_plus" type="button" title="Increase Font Size"><i class="fas fa-plus"></i></button><button id="font_minus" type="button" title="Decrease Font Size"><i class="fas fa-minus"></i></button><button id="darkmode" type="button" title="Switch Between Light And Dark Mode"><i class="fas fa-adjust"></i></button></div><div id="rightside-config-show"><button id="rightside_config" type="button" title="Setting"><i class="fas fa-cog"></i></button><button class="close" id="mobile-toc-button" type="button" title="Table Of Contents"><i class="fas fa-list-ul"></i></button><button id="go-up" type="button" title="Back To Top"><i class="fas fa-arrow-up"></i></button></div></section><div><script src="https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js"></script><script src="/project/js/utils.js"></script><script src="/project/js/main.js"></script><script src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.js"></script><script src="https://cdn.jsdelivr.net/npm/instant.page/instantpage.min.js" type="module" defer></script><script src="https://cdn.jsdelivr.net/npm/vanilla-lazyload/dist/lazyload.iife.min.js" async></script><div class="js-pjax"><script>if (!window.MathJax) {
  window.MathJax = {
    loader: {
      source: {
        '[tex]/amsCd': '[tex]/amscd'
      }
    },
    tex: {
      inlineMath: [ ['$','$'], ["\\(","\\)"]],
      tags: 'ams'
    },
    options: {
      renderActions: {
        findScript: [10, doc => {
          for (const node of document.querySelectorAll('script[type^="math/tex"]')) {
            const display = !!node.type.match(/; *mode=display/)
            const math = new doc.options.MathItem(node.textContent, doc.inputJax[0], display)
            const text = document.createTextNode('')
            node.parentNode.replaceChild(text, node)
            math.start = {node: text, delim: '', n: 0}
            math.end = {node: text, delim: '', n: 0}
            doc.math.push(math)
          }
        }, '']
      }
    }
  }
  
  var script = document.createElement('script')
  script.src = 'https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js'
  script.id = 'MathJax-script'
  script.async = true
  document.head.appendChild(script)
} else {
  MathJax.startup.document.state(0)
  MathJax.texReset()
  MathJax.typeset()
}</script><script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div></div></body></html>